{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "37c1c3e0-d1b3-44e9-af4b-069e45a9726c",
   "metadata": {
    "tags": []
   },
   "source": [
    "# <b> Neutrophil Lymphangioleiomyomatosis (LAM) Analysis - for Sasha-Gaye Richards, PhD from the Johnson Lab <b>\n",
    "This is the notebook for the analysis of different Neutrophil phenotypes using online available datasets of single cell RNA Seq data.\n",
    "- Single-Cell Transcriptomic Analysis Identifies a Unique Pulmonary Lymphangioleiomyomatosis Cell\n",
    "    - DOI: https://doi.org/10.1164/rccm.201912-2445OC\n",
    "    - GEO: GSE135851\n",
    "        - Samples:\n",
    "            - GSE135851_LAM1\n",
    "            - GSE135851_LAM2\n",
    "            - GSE135851_LAM3\n",
    "            - GSE135851_LAM4\n",
    "- Midkine expression by stem-like tumor cells drives persistence to mTOR inhibition and an immune-suppressive microenvironment\n",
    "    - DOI: https://doi.org/10.1038/s41467-022-32673-7\n",
    "    - GEO: GSE190260\n",
    "        - Samples:\n",
    "            - GSE190260_LAM1\n",
    "            - GSE190260_LAM2\n",
    "            - GSE190260_LAM3\n",
    "For a reference dataset the following publication was used:\n",
    "- Neutrophil single-cell analysis identifies a type II interferon-related subset for predicting relapse of autoimmune small vessel vasculitis.\n",
    "    - DOI: https://doi.org/10.1038/s41467-025-58550-7\n",
    "    - Online: https://ddbj.nig.ac.jp/public/ddbj_database/gea/experiment/E-GEAD-000/E-GEAD-867/\n",
    "        - Samples:\n",
    "            - HD-1\n",
    "            - HD-2\n",
    "            - HD-3\n",
    "            - HD-4\n",
    "            - HD-5\n",
    "            - HD-6\n",
    "            - HD-7\n",
    "            - MPA-1\n",
    "            - MPA-2\n",
    "            - MPA-3\n",
    "            - MPA-4\n",
    "            - MPA-5\n",
    "            - MPA-6  \n",
    "\n",
    "## <b>Analysis<b>\n",
    "The analysis was done by first isolating the Neutrophil using *MPO* as a marker.\n",
    "\n",
    "Datasets were then merged by using the *harmonizepy* package.\n",
    "\n",
    "After this was done different subsets of Neutrophils were marked as **N1** and **N2**, depending on a set of gene expression markers.\n",
    "\n",
    "| **N1 Neutrophils** | **N2 Neutrophils** |\n",
    "| --- | --- |\n",
    "| CD11b (ITGAM) | CD11b (ITGAM) |\n",
    "| CD16 (FCGR3B) | CD16 (FCGR3B) |\n",
    "| CD66 (CEACAM1) | CD66 (CEACAM1) |\n",
    "| CD177 (GPI) | CD177 (GPI) |\n",
    "| ICAM1 | CD62L (SELL/L-selectin)|\n",
    "| CXCL1 | CD206 (MRC1) |\n",
    "| CXCL2 | |\n",
    "| CXCL3 | |\n",
    "| CXCL10 | |\n",
    "| TNF | |\n",
    "| MYD88 | |"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "34e175a1-a83c-472e-bc40-b5d8fef75d88",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "import random\n",
    "import scanpy as sc\n",
    "import scanpy.external as sce\n",
    "import anndata as adata\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import scipy\n",
    "import os\n",
    "import scipy.io as sio\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.cm as cm\n",
    "import matplotlib.colors as mcolors\n",
    "from matplotlib.colors import Normalize\n",
    "import seaborn as sns\n",
    "import gseapy\n",
    "import scvi\n",
    "import tempfile\n",
    "import numba\n",
    "import pooch\n",
    "import pynndescent\n",
    "import torch\n",
    "import scipy.sparse as sp\n",
    "import gdown\n",
    "import gzip\n",
    "import shutil\n",
    "import urllib.request\n",
    "from gprofiler import GProfiler\n",
    "import celltypist\n",
    "from celltypist import models\n",
    "import warnings\n",
    "import logging\n",
    "from IPython.display import display, HTML\n",
    "from sc_plot_and_embed import sc_plot_and_embed\n",
    "from onedrive_share import onedrive_direct_link \n",
    "\n",
    "# Initialize the g:Profiler object\n",
    "gp = GProfiler(return_dataframe=True)\n",
    "print(gp.profile)\n",
    "\n",
    "data_path = '/gpfs01/home/mzxcs8/Extra/Neutrophil LAM Phenotype - Johnson Lab/Analysis_LAM/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "40ec4874-24d0-4dea-9b23-d77bdfc6e265",
   "metadata": {
    "editable": true,
    "slideshow": {
     "slide_type": ""
    },
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "random.seed(1325)\n",
    "sc.settings.verbosity = 0 # verbosity: errors (0), warnings (1), info (2), hints (3)\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "logging.getLogger().setLevel(logging.ERROR)\n",
    "sc.settings.set_figure_params(dpi=100, fontsize=10, dpi_save=100, figsize=(6,5), format='tiff')\n",
    "sc.settings.figdir = '/gpfs01/home/mzxcs8/Extra/Neutrophil LAM Phenotype - Johnson Lab/Analysis_LAM/_static/Figures'\n",
    "OneDrive = 'https://uniofnottm.sharepoint.com/:f:/r/sites/MacrophagePaper/Shared%20Documents/General/Bioinformatics/Analysis%20for%20Publication/scRNASeq/LCC_Healthy/Figures?csf=1&web=1&e=BeWfn7'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f73233fd-4858-4778-802a-8e4ef6273520",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.normalize_total(HD1_AD, target_sum=1e4)\n",
    "sc.pp.log1p(HD1_AD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "98490fb4-240d-4f56-a06a-b68beeb71551",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.neighbors(HD1_AD)\n",
    "sc.tl.umap(HD1_AD)\n",
    "sc.tl.leiden(HD1_AD,flavor = \"igraph\", resolution=0.5, n_iterations = 2)\n",
    "sc.tl.dendrogram(HD1_AD, groupby = \"leiden\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4ef57b51-9a5d-45f1-abb5-dccc83a90bfb",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(HD1_AD,color=['leiden'], legend_fontsize=8)\n",
    "sc.pl.umap(HD1_AD,color=['dataset'], legend_fontsize=8)\n",
    "sc.pl.umap(HD1_AD,color=['sample'], legend_fontsize=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fef82007-9b59-431b-b2cc-5a023192e887",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(HD1_AD,color=['MPO','ELANE','SLPI','ITGAM','SELL','ICAM1','CXCL10','CXCL1','CXCL2','CEACAM1'], legend_fontsize=8, cmap ='Reds' )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5c4de871-d386-4bce-835c-5d4413c76ebf",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    HD1_AD,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"leiden\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    label = 'Dotplot of Neutrophil specific markers - LAM 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e607157e-4ac9-43f0-9eb8-42fa59f150ba",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "HD1_AD.obs[\"Cell_Type\"] = \"Other\"\n",
    "HD1_AD.obs[\"Cell_Type\"] = HD1_AD.obs[\"Cell_Type\"].astype(\"category\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "091a5078-ac7c-4211-867a-6d9748525950",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "HD1_AD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dca6bbd7-5b52-4c92-827c-e61162b0b298",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "# Utility function to safely extract gene expression values\n",
    "# Step 1: Subset epithelial only\n",
    "HD1_AD_Neutrophil = HD1_AD.copy()\n",
    "\n",
    "def get_gene_expression(adata, gene_name):\n",
    "    \"\"\"Extract expression values for a given gene safely, accounting for sparse matrices.\"\"\"\n",
    "    if gene_name in adata.var_names:\n",
    "        gene_idx = adata.var_names.get_loc(gene_name)\n",
    "        values = adata.X[:, gene_idx].toarray() if sp.issparse(adata.X) else adata.X[:, gene_idx]\n",
    "        return values.flatten()\n",
    "    else:\n",
    "        print(f\"Warning: Gene {gene_name} not found in dataset.\")\n",
    "        return np.zeros(adata.shape[0])\n",
    "\n",
    "#Neutrophil\n",
    "MPO_values = get_gene_expression(HD1_AD_Neutrophil, 'MPO')\n",
    "ELANE_values = get_gene_expression(HD1_AD_Neutrophil, 'ELANE')\n",
    "\n",
    "if \"Neutrophil MPO+\" not in HD1_AD_Neutrophil.obs[\"Cell_Type\"].cat.categories:\n",
    "    HD1_AD_Neutrophil.obs[\"Cell_Type\"] = HD1_AD_Neutrophil.obs[\"Cell_Type\"].cat.add_categories([\"Neutrophil MPO+\"]) \n",
    "\n",
    "neutrophil_mask = (MPO_values > 0.001)\n",
    "HD1_AD_Neutrophil.obs.loc[neutrophil_mask, 'Cell_Type'] = \"Neutrophil MPO+\"\n",
    "\n",
    "HD1_AD_Neutrophil.obs[\"Cell_Type\"] = HD1_AD_Neutrophil.obs[\"Cell_Type\"].astype(\"category\")\n",
    "\n",
    "print(HD1_AD_Neutrophil.obs['Cell_Type'].value_counts())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3039b6d2-de41-4f99-820e-55034d2fe48a",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "HD1_AD_Neutrophil.obs['NP_Only'] = HD1_AD_Neutrophil.obs['Cell_Type'] == 'Neutrophil MPO+'\n",
    "sc.pl.umap(HD1_AD_Neutrophil, color='NP_Only', palette = 'Reds', size = 30, title = 'Neutrophil MPO+')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "15f1d326-b596-427a-9e01-4f49516b5bdf",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    HD1_AD_Neutrophil,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"Cell_Type\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    title = 'Dotplot of Neutrophil specific markers - Lam 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e9cc4c72-4608-4c2e-bdc2-8ba8a2beb10a",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.normalize_total(HD2_AD, target_sum=1e4)\n",
    "sc.pp.log1p(HD2_AD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "19bd8ce4-3a45-43d1-9795-23a207ef0f4d",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.neighbors(HD2_AD)\n",
    "sc.tl.umap(HD2_AD)\n",
    "sc.tl.leiden(HD2_AD,flavor = \"igraph\", resolution=0.5, n_iterations = 2)\n",
    "sc.tl.dendrogram(HD2_AD, groupby = \"leiden\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4140b87b-4294-4bcd-a044-6138f41ac4a7",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(HD2_AD,color=['leiden'], legend_fontsize=8)\n",
    "sc.pl.umap(HD2_AD,color=['dataset'], legend_fontsize=8)\n",
    "sc.pl.umap(HD2_AD,color=['sample'], legend_fontsize=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0e3f5d34-9a96-42bd-a812-2ad77fb3b064",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(HD2_AD,color=['MPO','ELANE','SLPI','ITGAM','SELL','ICAM1','CXCL10','CXCL1','CXCL2','CEACAM1'], legend_fontsize=8, cmap ='Reds' )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3956b124-89f6-4936-bebc-6b8c54c55e15",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    HD2_AD,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"leiden\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    label = 'Dotplot of Neutrophil specific markers - LAM 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d709dd05-0dc7-4112-b30c-c9f734279a42",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "HD2_AD.obs[\"Cell_Type\"] = \"Other\"\n",
    "HD2_AD.obs[\"Cell_Type\"] = HD2_AD.obs[\"Cell_Type\"].astype(\"category\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cbb9efa2-8b76-4736-bab5-a33a3e592a22",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "HD2_AD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "27dd07a2-4519-43bb-b3ef-02b76d9b0f71",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "# Utility function to safely extract gene expression values\n",
    "# Step 1: Subset epithelial only\n",
    "HD2_AD_Neutrophil = HD2_AD.copy()\n",
    "\n",
    "def get_gene_expression(adata, gene_name):\n",
    "    \"\"\"Extract expression values for a given gene safely, accounting for sparse matrices.\"\"\"\n",
    "    if gene_name in adata.var_names:\n",
    "        gene_idx = adata.var_names.get_loc(gene_name)\n",
    "        values = adata.X[:, gene_idx].toarray() if sp.issparse(adata.X) else adata.X[:, gene_idx]\n",
    "        return values.flatten()\n",
    "    else:\n",
    "        print(f\"Warning: Gene {gene_name} not found in dataset.\")\n",
    "        return np.zeros(adata.shape[0])\n",
    "\n",
    "#Neutrophil\n",
    "MPO_values = get_gene_expression(HD2_AD_Neutrophil, 'MPO')\n",
    "ELANE_values = get_gene_expression(HD2_AD_Neutrophil, 'ELANE')\n",
    "\n",
    "if \"Neutrophil MPO+\" not in HD2_AD_Neutrophil.obs[\"Cell_Type\"].cat.categories:\n",
    "    HD2_AD_Neutrophil.obs[\"Cell_Type\"] = HD2_AD_Neutrophil.obs[\"Cell_Type\"].cat.add_categories([\"Neutrophil MPO+\"]) \n",
    "\n",
    "neutrophil_mask = (MPO_values > 0.001)\n",
    "HD2_AD_Neutrophil.obs.loc[neutrophil_mask, 'Cell_Type'] = \"Neutrophil MPO+\"\n",
    "\n",
    "HD2_AD_Neutrophil.obs[\"Cell_Type\"] = HD2_AD_Neutrophil.obs[\"Cell_Type\"].astype(\"category\")\n",
    "\n",
    "print(HD2_AD_Neutrophil.obs['Cell_Type'].value_counts())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1a25450e-583e-49ff-84b3-196f632b8cae",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "HD2_AD_Neutrophil.obs['NP_Only'] = HD2_AD_Neutrophil.obs['Cell_Type'] == 'Neutrophil MPO+'\n",
    "sc.pl.umap(HD2_AD_Neutrophil, color='NP_Only', palette = 'Reds', size = 30, title = 'Neutrophil MPO+')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e04f02af-15f3-40b7-acff-d88311a9d991",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    HD2_AD_Neutrophil,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"Cell_Type\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    title = 'Dotplot of Neutrophil specific markers - Lam 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "19aba19b-94ed-4244-a111-8afbee5386ca",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.normalize_total(HD3_AD, target_sum=1e4)\n",
    "sc.pp.log1p(HD3_AD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a3754252-1000-4748-bd25-f59290a9b44d",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.neighbors(HD3_AD)\n",
    "sc.tl.umap(HD3_AD)\n",
    "sc.tl.leiden(HD3_AD,flavor = \"igraph\", resolution=0.5, n_iterations = 2)\n",
    "sc.tl.dendrogram(HD3_AD, groupby = \"leiden\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7da68c15-f45c-4c2e-bbc9-31155d456da1",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(HD3_AD,color=['leiden'], legend_fontsize=8)\n",
    "sc.pl.umap(HD3_AD,color=['dataset'], legend_fontsize=8)\n",
    "sc.pl.umap(HD3_AD,color=['sample'], legend_fontsize=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c225b1d-d1f4-4dd8-951d-690978a0a478",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(HD3_AD,color=['MPO','ELANE','SLPI','ITGAM','SELL','ICAM1','CXCL10','CXCL1','CXCL2','CEACAM1'], legend_fontsize=8, cmap ='Reds' )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bdc57a61-7fba-4da5-9c65-88cb7d6217b7",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    HD3_AD,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"leiden\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    label = 'Dotplot of Neutrophil specific markers - LAM 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "74a0275c-aa10-4d40-a279-50850dc0a625",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "HD3_AD.obs[\"Cell_Type\"] = \"Other\"\n",
    "HD3_AD.obs[\"Cell_Type\"] = HD3_AD.obs[\"Cell_Type\"].astype(\"category\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "84c2fdb7-8b1c-46cb-bfa6-5fcca86911ed",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "HD3_AD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9805a096-dc15-4c21-9abf-ebdcc3803f85",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "# Utility function to safely extract gene expression values\n",
    "# Step 1: Subset epithelial only\n",
    "HD3_AD_Neutrophil = HD3_AD.copy()\n",
    "\n",
    "def get_gene_expression(adata, gene_name):\n",
    "    \"\"\"Extract expression values for a given gene safely, accounting for sparse matrices.\"\"\"\n",
    "    if gene_name in adata.var_names:\n",
    "        gene_idx = adata.var_names.get_loc(gene_name)\n",
    "        values = adata.X[:, gene_idx].toarray() if sp.issparse(adata.X) else adata.X[:, gene_idx]\n",
    "        return values.flatten()\n",
    "    else:\n",
    "        print(f\"Warning: Gene {gene_name} not found in dataset.\")\n",
    "        return np.zeros(adata.shape[0])\n",
    "\n",
    "#Neutrophil\n",
    "MPO_values = get_gene_expression(HD3_AD_Neutrophil, 'MPO')\n",
    "ELANE_values = get_gene_expression(HD3_AD_Neutrophil, 'ELANE')\n",
    "\n",
    "if \"Neutrophil MPO+\" not in HD3_AD_Neutrophil.obs[\"Cell_Type\"].cat.categories:\n",
    "    HD3_AD_Neutrophil.obs[\"Cell_Type\"] = HD3_AD_Neutrophil.obs[\"Cell_Type\"].cat.add_categories([\"Neutrophil MPO+\"]) \n",
    "\n",
    "neutrophil_mask = (MPO_values > 0.001)\n",
    "HD3_AD_Neutrophil.obs.loc[neutrophil_mask, 'Cell_Type'] = \"Neutrophil MPO+\"\n",
    "\n",
    "HD3_AD_Neutrophil.obs[\"Cell_Type\"] = HD3_AD_Neutrophil.obs[\"Cell_Type\"].astype(\"category\")\n",
    "\n",
    "print(HD3_AD_Neutrophil.obs['Cell_Type'].value_counts())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2cddfcdc-3a30-45a3-96c0-8a50ea4d3914",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "HD3_AD_Neutrophil.obs['NP_Only'] = HD3_AD_Neutrophil.obs['Cell_Type'] == 'Neutrophil MPO+'\n",
    "sc.pl.umap(HD3_AD_Neutrophil, color='NP_Only', palette = 'Reds', size = 30, title = 'Neutrophil MPO+')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa829925-c093-4309-8fee-ca803470f0f0",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    HD3_AD_Neutrophil,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"Cell_Type\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    title = 'Dotplot of Neutrophil specific markers - Lam 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a1611e83-2362-4d92-8552-c124cc8156e5",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.normalize_total(HD4_AD, target_sum=1e4)\n",
    "sc.pp.log1p(HD4_AD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2d7b8f9b-d8b7-4c5b-ac39-3c2d1873c130",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.neighbors(HD4_AD)\n",
    "sc.tl.umap(HD4_AD)\n",
    "sc.tl.leiden(HD4_AD,flavor = \"igraph\", resolution=0.5, n_iterations = 2)\n",
    "sc.tl.dendrogram(HD4_AD, groupby = \"leiden\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0b9aeebc-88ff-4a0d-ba69-47941f70f7e7",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(HD4_AD,color=['leiden'], legend_fontsize=8)\n",
    "sc.pl.umap(HD4_AD,color=['dataset'], legend_fontsize=8)\n",
    "sc.pl.umap(HD4_AD,color=['sample'], legend_fontsize=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ade2ab5f-b467-43a1-97ff-6ba866d520e8",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(HD4_AD,color=['MPO','ELANE','SLPI','ITGAM','SELL','ICAM1','CXCL10','CXCL1','CXCL2','CEACAM1'], legend_fontsize=8, cmap ='Reds' )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7154bc40-b426-4ab2-ad64-461ac1dc2bf8",
   "metadata": {
    "scrolled": true,
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    HD4_AD,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"leiden\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    label = 'Dotplot of Neutrophil specific markers - LAM 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fec420a7-7179-47ad-9688-2d5afdcb695a",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "HD4_AD.obs[\"Cell_Type\"] = \"Other\"\n",
    "HD4_AD.obs[\"Cell_Type\"] = HD4_AD.obs[\"Cell_Type\"].astype(\"category\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7bf03f49-66a5-415c-98b3-be7dd6f2ac0b",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "HD4_AD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e4abc8f8-f6ef-4138-bec2-3548b824609b",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "# Utility function to safely extract gene expression values\n",
    "# Step 1: Subset epithelial only\n",
    "HD4_AD_Neutrophil = HD4_AD.copy()\n",
    "\n",
    "def get_gene_expression(adata, gene_name):\n",
    "    \"\"\"Extract expression values for a given gene safely, accounting for sparse matrices.\"\"\"\n",
    "    if gene_name in adata.var_names:\n",
    "        gene_idx = adata.var_names.get_loc(gene_name)\n",
    "        values = adata.X[:, gene_idx].toarray() if sp.issparse(adata.X) else adata.X[:, gene_idx]\n",
    "        return values.flatten()\n",
    "    else:\n",
    "        print(f\"Warning: Gene {gene_name} not found in dataset.\")\n",
    "        return np.zeros(adata.shape[0])\n",
    "\n",
    "#Neutrophil\n",
    "MPO_values = get_gene_expression(HD4_AD_Neutrophil, 'MPO')\n",
    "ELANE_values = get_gene_expression(HD4_AD_Neutrophil, 'ELANE')\n",
    "\n",
    "if \"Neutrophil MPO+\" not in HD4_AD_Neutrophil.obs[\"Cell_Type\"].cat.categories:\n",
    "    HD4_AD_Neutrophil.obs[\"Cell_Type\"] = HD4_AD_Neutrophil.obs[\"Cell_Type\"].cat.add_categories([\"Neutrophil MPO+\"]) \n",
    "\n",
    "neutrophil_mask = (MPO_values > 0.001)\n",
    "HD4_AD_Neutrophil.obs.loc[neutrophil_mask, 'Cell_Type'] = \"Neutrophil MPO+\"\n",
    "\n",
    "HD4_AD_Neutrophil.obs[\"Cell_Type\"] = HD4_AD_Neutrophil.obs[\"Cell_Type\"].astype(\"category\")\n",
    "\n",
    "print(HD4_AD_Neutrophil.obs['Cell_Type'].value_counts())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ae5ed023-b5a7-4c71-9447-fc2ffd489bf9",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "HD4_AD_Neutrophil.obs['NP_Only'] = HD4_AD_Neutrophil.obs['Cell_Type'] == 'Neutrophil MPO+'\n",
    "sc.pl.umap(HD4_AD_Neutrophil, color='NP_Only', palette = 'Reds', size = 30, title = 'Neutrophil MPO+')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "27270b40-3129-482e-8883-cbdb55624a39",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    HD4_AD_Neutrophil,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"Cell_Type\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    title = 'Dotplot of Neutrophil specific markers - Lam 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d215d2f1-8712-4b2d-9512-00a4b6b2be6e",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.normalize_total(HD5_AD, target_sum=1e4)\n",
    "sc.pp.log1p(HD5_AD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d8fa0786-cc26-4501-9efa-bdb175674a6a",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.neighbors(HD5_AD)\n",
    "sc.tl.umap(HD5_AD)\n",
    "sc.tl.leiden(HD5_AD,flavor = \"igraph\", resolution=0.5, n_iterations = 2)\n",
    "sc.tl.dendrogram(HD5_AD, groupby = \"leiden\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "25c35959-b0a0-42d6-8152-9c6f438c949d",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(HD5_AD,color=['leiden'], legend_fontsize=8)\n",
    "sc.pl.umap(HD5_AD,color=['dataset'], legend_fontsize=8)\n",
    "sc.pl.umap(HD5_AD,color=['sample'], legend_fontsize=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "825c02f9-5e1f-4736-a36f-247aaa063ca3",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(HD5_AD,color=['MPO','ELANE','SLPI','ITGAM','SELL','ICAM1','CXCL10','CXCL1','CXCL2','CEACAM1'], legend_fontsize=8, cmap ='Reds' )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e196b7d2-0eaa-4d1f-9a68-b5a21524e69a",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    HD5_AD,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"leiden\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    label = 'Dotplot of Neutrophil specific markers - LAM 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a48c7454-15ad-4892-8bd2-8193296bdec6",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "HD5_AD.obs[\"Cell_Type\"] = \"Other\"\n",
    "HD5_AD.obs[\"Cell_Type\"] = HD5_AD.obs[\"Cell_Type\"].astype(\"category\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4f466918-26c2-43b3-b6c6-6440b47e79f9",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "HD5_AD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bff39002-c307-44c7-9060-e1f2e637c793",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "# Utility function to safely extract gene expression values\n",
    "# Step 1: Subset epithelial only\n",
    "HD5_AD_Neutrophil = HD5_AD.copy()\n",
    "\n",
    "def get_gene_expression(adata, gene_name):\n",
    "    \"\"\"Extract expression values for a given gene safely, accounting for sparse matrices.\"\"\"\n",
    "    if gene_name in adata.var_names:\n",
    "        gene_idx = adata.var_names.get_loc(gene_name)\n",
    "        values = adata.X[:, gene_idx].toarray() if sp.issparse(adata.X) else adata.X[:, gene_idx]\n",
    "        return values.flatten()\n",
    "    else:\n",
    "        print(f\"Warning: Gene {gene_name} not found in dataset.\")\n",
    "        return np.zeros(adata.shape[0])\n",
    "\n",
    "#Neutrophil\n",
    "MPO_values = get_gene_expression(HD5_AD_Neutrophil, 'MPO')\n",
    "ELANE_values = get_gene_expression(HD5_AD_Neutrophil, 'ELANE')\n",
    "\n",
    "if \"Neutrophil MPO+\" not in HD5_AD_Neutrophil.obs[\"Cell_Type\"].cat.categories:\n",
    "    HD5_AD_Neutrophil.obs[\"Cell_Type\"] = HD5_AD_Neutrophil.obs[\"Cell_Type\"].cat.add_categories([\"Neutrophil MPO+\"]) \n",
    "\n",
    "neutrophil_mask = (MPO_values > 0.001)\n",
    "HD5_AD_Neutrophil.obs.loc[neutrophil_mask, 'Cell_Type'] = \"Neutrophil MPO+\"\n",
    "\n",
    "HD5_AD_Neutrophil.obs[\"Cell_Type\"] = HD5_AD_Neutrophil.obs[\"Cell_Type\"].astype(\"category\")\n",
    "\n",
    "print(HD5_AD_Neutrophil.obs['Cell_Type'].value_counts())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1dcf77e1-cfc2-4c26-9dc6-b665ad67d9b7",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "HD5_AD_Neutrophil.obs['NP_Only'] = HD5_AD_Neutrophil.obs['Cell_Type'] == 'Neutrophil MPO+'\n",
    "sc.pl.umap(HD5_AD_Neutrophil, color='NP_Only', palette = 'Reds', size = 30, title = 'Neutrophil MPO+')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7a5859da-c9b8-4f1e-a638-10b9d550d6e0",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    HD5_AD_Neutrophil,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"Cell_Type\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    title = 'Dotplot of Neutrophil specific markers - Lam 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3cbf50c5-b3af-41eb-92de-16a3807325c1",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.normalize_total(HD6_AD, target_sum=1e4)\n",
    "sc.pp.log1p(HD6_AD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1e357575-a079-4dd6-90b0-6359f42a2693",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.neighbors(HD6_AD)\n",
    "sc.tl.umap(HD6_AD)\n",
    "sc.tl.leiden(HD6_AD,flavor = \"igraph\", resolution=0.5, n_iterations = 2)\n",
    "sc.tl.dendrogram(HD6_AD, groupby = \"leiden\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "84323c39-bdae-43d4-a99e-f5e28215984f",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(HD6_AD,color=['leiden'], legend_fontsize=8)\n",
    "sc.pl.umap(HD6_AD,color=['dataset'], legend_fontsize=8)\n",
    "sc.pl.umap(HD6_AD,color=['sample'], legend_fontsize=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "baacb40e-0a61-4cbd-ac5b-b761cf10dc3d",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(HD6_AD,color=['MPO','ELANE','SLPI','ITGAM','SELL','ICAM1','CXCL10','CXCL1','CXCL2','CEACAM1'], legend_fontsize=8, cmap ='Reds' )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "23dc4e64-d7a9-4327-9483-fe9d4fc33439",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    HD6_AD,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"leiden\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    label = 'Dotplot of Neutrophil specific markers - LAM 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bc78398c-4396-4c57-8f65-64217f08467c",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "HD6_AD.obs[\"Cell_Type\"] = \"Other\"\n",
    "HD6_AD.obs[\"Cell_Type\"] = HD6_AD.obs[\"Cell_Type\"].astype(\"category\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9f25670a-bfc6-46d7-9335-060b38c1e116",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "HD6_AD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dce7cca6-a691-43aa-8a5e-c70d4de150b6",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "# Utility function to safely extract gene expression values\n",
    "# Step 1: Subset epithelial only\n",
    "HD6_AD_Neutrophil = HD6_AD.copy()\n",
    "\n",
    "def get_gene_expression(adata, gene_name):\n",
    "    \"\"\"Extract expression values for a given gene safely, accounting for sparse matrices.\"\"\"\n",
    "    if gene_name in adata.var_names:\n",
    "        gene_idx = adata.var_names.get_loc(gene_name)\n",
    "        values = adata.X[:, gene_idx].toarray() if sp.issparse(adata.X) else adata.X[:, gene_idx]\n",
    "        return values.flatten()\n",
    "    else:\n",
    "        print(f\"Warning: Gene {gene_name} not found in dataset.\")\n",
    "        return np.zeros(adata.shape[0])\n",
    "\n",
    "#Neutrophil\n",
    "MPO_values = get_gene_expression(HD6_AD_Neutrophil, 'MPO')\n",
    "ELANE_values = get_gene_expression(HD6_AD_Neutrophil, 'ELANE')\n",
    "\n",
    "if \"Neutrophil MPO+\" not in HD6_AD_Neutrophil.obs[\"Cell_Type\"].cat.categories:\n",
    "    HD6_AD_Neutrophil.obs[\"Cell_Type\"] = HD6_AD_Neutrophil.obs[\"Cell_Type\"].cat.add_categories([\"Neutrophil MPO+\"]) \n",
    "\n",
    "neutrophil_mask = (MPO_values > 0.001)\n",
    "HD6_AD_Neutrophil.obs.loc[neutrophil_mask, 'Cell_Type'] = \"Neutrophil MPO+\"\n",
    "\n",
    "HD6_AD_Neutrophil.obs[\"Cell_Type\"] = HD6_AD_Neutrophil.obs[\"Cell_Type\"].astype(\"category\")\n",
    "\n",
    "print(HD6_AD_Neutrophil.obs['Cell_Type'].value_counts())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0dadfc69-513a-43d4-96af-e0108a09b5ad",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "HD6_AD_Neutrophil.obs['NP_Only'] = HD6_AD_Neutrophil.obs['Cell_Type'] == 'Neutrophil MPO+'\n",
    "sc.pl.umap(HD6_AD_Neutrophil, color='NP_Only', palette = 'Reds', size = 30, title = 'Neutrophil MPO+')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e988fbbf-ab2c-4fe8-aa06-df667e8303b4",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    HD6_AD_Neutrophil,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"Cell_Type\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    title = 'Dotplot of Neutrophil specific markers - Lam 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5f7413f1-63df-4170-afa3-ce2b9cbcb4ed",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.normalize_total(HD7_AD, target_sum=1e4)\n",
    "sc.pp.log1p(HD7_AD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2dda2ef6-7fe9-4e8c-b335-d3bac3a75044",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.neighbors(HD7_AD)\n",
    "sc.tl.umap(HD7_AD)\n",
    "sc.tl.leiden(HD7_AD,flavor = \"igraph\", resolution=0.5, n_iterations = 2)\n",
    "sc.tl.dendrogram(HD7_AD, groupby = \"leiden\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "15b851fd-add5-4d9e-b24e-8e5bbd4b7fc1",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(HD7_AD,color=['leiden'], legend_fontsize=8)\n",
    "sc.pl.umap(HD7_AD,color=['dataset'], legend_fontsize=8)\n",
    "sc.pl.umap(HD7_AD,color=['sample'], legend_fontsize=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5e7f6342-2737-4234-9198-c1cece71c0a5",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(HD7_AD,color=['MPO','ELANE','SLPI','ITGAM','SELL','ICAM1','CXCL10','CXCL1','CXCL2','CEACAM1'], legend_fontsize=8, cmap ='Reds' )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c86d140c-c871-489a-bf4b-507d79c54e9b",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    HD7_AD,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"leiden\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    label = 'Dotplot of Neutrophil specific markers - LAM 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a1bdab70-694e-4b59-aba3-282b31436b5f",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "HD7_AD.obs[\"Cell_Type\"] = \"Other\"\n",
    "HD7_AD.obs[\"Cell_Type\"] = HD7_AD.obs[\"Cell_Type\"].astype(\"category\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "997af58f-60c4-45d0-8d83-84a3e8f81a98",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "HD7_AD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ed35418-6684-4932-9b5d-8407203d655a",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "# Utility function to safely extract gene expression values\n",
    "# Step 1: Subset epithelial only\n",
    "HD7_AD_Neutrophil = HD7_AD.copy()\n",
    "\n",
    "def get_gene_expression(adata, gene_name):\n",
    "    \"\"\"Extract expression values for a given gene safely, accounting for sparse matrices.\"\"\"\n",
    "    if gene_name in adata.var_names:\n",
    "        gene_idx = adata.var_names.get_loc(gene_name)\n",
    "        values = adata.X[:, gene_idx].toarray() if sp.issparse(adata.X) else adata.X[:, gene_idx]\n",
    "        return values.flatten()\n",
    "    else:\n",
    "        print(f\"Warning: Gene {gene_name} not found in dataset.\")\n",
    "        return np.zeros(adata.shape[0])\n",
    "\n",
    "#Neutrophil\n",
    "MPO_values = get_gene_expression(HD7_AD_Neutrophil, 'MPO')\n",
    "ELANE_values = get_gene_expression(HD7_AD_Neutrophil, 'ELANE')\n",
    "\n",
    "if \"Neutrophil MPO+\" not in HD7_AD_Neutrophil.obs[\"Cell_Type\"].cat.categories:\n",
    "    HD7_AD_Neutrophil.obs[\"Cell_Type\"] = HD7_AD_Neutrophil.obs[\"Cell_Type\"].cat.add_categories([\"Neutrophil MPO+\"]) \n",
    "\n",
    "neutrophil_mask = (MPO_values > 0.001)\n",
    "HD7_AD_Neutrophil.obs.loc[neutrophil_mask, 'Cell_Type'] = \"Neutrophil MPO+\"\n",
    "\n",
    "HD7_AD_Neutrophil.obs[\"Cell_Type\"] = HD7_AD_Neutrophil.obs[\"Cell_Type\"].astype(\"category\")\n",
    "\n",
    "print(HD7_AD_Neutrophil.obs['Cell_Type'].value_counts())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "44312c05-3113-41d7-bc3e-ab1674b3816b",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "HD7_AD_Neutrophil.obs['NP_Only'] = HD7_AD_Neutrophil.obs['Cell_Type'] == 'Neutrophil MPO+'\n",
    "sc.pl.umap(HD7_AD_Neutrophil, color='NP_Only', palette = 'Reds', size = 30, title = 'Neutrophil MPO+')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5c3b1608-d33e-449c-baa1-e5abe7a86434",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    HD7_AD_Neutrophil,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"Cell_Type\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    title = 'Dotplot of Neutrophil specific markers - Lam 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b0e6750a-16a4-44e8-a481-c7998364992c",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.normalize_total(MPA1_AD, target_sum=1e4)\n",
    "sc.pp.log1p(MPA1_AD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6967e11f-beed-43a0-9841-ffe4a3588951",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.neighbors(MPA1_AD)\n",
    "sc.tl.umap(MPA1_AD)\n",
    "sc.tl.leiden(MPA1_AD,flavor = \"igraph\", resolution=0.5, n_iterations = 2)\n",
    "sc.tl.dendrogram(MPA1_AD, groupby = \"leiden\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "30dca340-3a65-4661-9f86-003dc5e64aa0",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(MPA1_AD,color=['leiden'], legend_fontsize=8)\n",
    "sc.pl.umap(MPA1_AD,color=['dataset'], legend_fontsize=8)\n",
    "sc.pl.umap(MPA1_AD,color=['sample'], legend_fontsize=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b4e4fab5-aa28-4205-bafc-b2250d00b704",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(MPA1_AD,color=['MPO','ELANE','SLPI','ITGAM','SELL','ICAM1','CXCL10','CXCL1','CXCL2','CEACAM1'], legend_fontsize=8, cmap ='Reds' )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "10694f5f-2c1e-4e74-ba92-8dbcac084324",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    MPA1_AD,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"leiden\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    label = 'Dotplot of Neutrophil specific markers - LAM 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6603a488-1bb8-4940-a7b5-c65bf93feafa",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "MPA1_AD.obs[\"Cell_Type\"] = \"Other\"\n",
    "MPA1_AD.obs[\"Cell_Type\"] = MPA1_AD.obs[\"Cell_Type\"].astype(\"category\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0f1180ff-3629-4056-b672-62062845bc7d",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "MPA1_AD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "49d4d226-86a7-40ea-a81f-f48f6b5651ea",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "# Utility function to safely extract gene expression values\n",
    "# Step 1: Subset epithelial only\n",
    "MPA1_AD_Neutrophil = MPA1_AD.copy()\n",
    "\n",
    "def get_gene_expression(adata, gene_name):\n",
    "    \"\"\"Extract expression values for a given gene safely, accounting for sparse matrices.\"\"\"\n",
    "    if gene_name in adata.var_names:\n",
    "        gene_idx = adata.var_names.get_loc(gene_name)\n",
    "        values = adata.X[:, gene_idx].toarray() if sp.issparse(adata.X) else adata.X[:, gene_idx]\n",
    "        return values.flatten()\n",
    "    else:\n",
    "        print(f\"Warning: Gene {gene_name} not found in dataset.\")\n",
    "        return np.zeros(adata.shape[0])\n",
    "\n",
    "#Neutrophil\n",
    "MPO_values = get_gene_expression(MPA1_AD_Neutrophil, 'MPO')\n",
    "ELANE_values = get_gene_expression(MPA1_AD_Neutrophil, 'ELANE')\n",
    "\n",
    "if \"Neutrophil MPO+\" not in MPA1_AD_Neutrophil.obs[\"Cell_Type\"].cat.categories:\n",
    "    MPA1_AD_Neutrophil.obs[\"Cell_Type\"] = MPA1_AD_Neutrophil.obs[\"Cell_Type\"].cat.add_categories([\"Neutrophil MPO+\"]) \n",
    "\n",
    "neutrophil_mask = (MPO_values > 0.001)\n",
    "MPA1_AD_Neutrophil.obs.loc[neutrophil_mask, 'Cell_Type'] = \"Neutrophil MPO+\"\n",
    "\n",
    "MPA1_AD_Neutrophil.obs[\"Cell_Type\"] = MPA1_AD_Neutrophil.obs[\"Cell_Type\"].astype(\"category\")\n",
    "\n",
    "print(MPA1_AD_Neutrophil.obs['Cell_Type'].value_counts())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0603b092-a9b0-4a57-b4ff-ab751c02426e",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "MPA1_AD_Neutrophil.obs['NP_Only'] = MPA1_AD_Neutrophil.obs['Cell_Type'] == 'Neutrophil MPO+'\n",
    "sc.pl.umap(MPA1_AD_Neutrophil, color='NP_Only', palette = 'Reds', size = 30, title = 'Neutrophil MPO+')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b1afd498-b70e-4847-a7f6-aff3ffe01886",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    MPA1_AD_Neutrophil,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"Cell_Type\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    title = 'Dotplot of Neutrophil specific markers - Lam 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "01d3023e-91b3-43af-9f1c-6ac078d497f1",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.normalize_total(MPA2_AD, target_sum=1e4)\n",
    "sc.pp.log1p(MPA2_AD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f1ee09ca-b508-42a5-879a-34e234a27394",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.neighbors(MPA2_AD)\n",
    "sc.tl.umap(MPA2_AD)\n",
    "sc.tl.leiden(MPA2_AD,flavor = \"igraph\", resolution=0.5, n_iterations = 2)\n",
    "sc.tl.dendrogram(MPA2_AD, groupby = \"leiden\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "424595e2-9f0a-4182-8114-82ad19e26b09",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(MPA2_AD,color=['leiden'], legend_fontsize=8)\n",
    "sc.pl.umap(MPA2_AD,color=['dataset'], legend_fontsize=8)\n",
    "sc.pl.umap(MPA2_AD,color=['sample'], legend_fontsize=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a12cf5f8-0868-4e8d-b817-af6f657ca254",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(MPA2_AD,color=['MPO','ELANE','SLPI','ITGAM','SELL','ICAM1','CXCL1','CXCL2','CEACAM1'], legend_fontsize=8, cmap ='Reds' )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "98a4098d-8a9e-49f0-9bee-c198d40fbba0",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    MPA2_AD,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"leiden\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    label = 'Dotplot of Neutrophil specific markers - LAM 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "720e083a-1c06-4c17-8a38-86ba738d3099",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "MPA2_AD.obs[\"Cell_Type\"] = \"Other\"\n",
    "MPA2_AD.obs[\"Cell_Type\"] = MPA2_AD.obs[\"Cell_Type\"].astype(\"category\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d3a46951-5535-4444-8085-b2ea480b4682",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "MPA2_AD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a47a5d54-7682-45aa-b40a-583864525ac1",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "# Utility function to safely extract gene expression values\n",
    "# Step 1: Subset epithelial only\n",
    "MPA2_AD_Neutrophil = MPA2_AD.copy()\n",
    "\n",
    "def get_gene_expression(adata, gene_name):\n",
    "    \"\"\"Extract expression values for a given gene safely, accounting for sparse matrices.\"\"\"\n",
    "    if gene_name in adata.var_names:\n",
    "        gene_idx = adata.var_names.get_loc(gene_name)\n",
    "        values = adata.X[:, gene_idx].toarray() if sp.issparse(adata.X) else adata.X[:, gene_idx]\n",
    "        return values.flatten()\n",
    "    else:\n",
    "        print(f\"Warning: Gene {gene_name} not found in dataset.\")\n",
    "        return np.zeros(adata.shape[0])\n",
    "\n",
    "#Neutrophil\n",
    "MPO_values = get_gene_expression(MPA2_AD_Neutrophil, 'MPO')\n",
    "ELANE_values = get_gene_expression(MPA2_AD_Neutrophil, 'ELANE')\n",
    "\n",
    "if \"Neutrophil MPO+\" not in MPA2_AD_Neutrophil.obs[\"Cell_Type\"].cat.categories:\n",
    "    MPA2_AD_Neutrophil.obs[\"Cell_Type\"] = MPA2_AD_Neutrophil.obs[\"Cell_Type\"].cat.add_categories([\"Neutrophil MPO+\"]) \n",
    "\n",
    "neutrophil_mask = (MPO_values > 0.001)\n",
    "MPA2_AD_Neutrophil.obs.loc[neutrophil_mask, 'Cell_Type'] = \"Neutrophil MPO+\"\n",
    "\n",
    "MPA2_AD_Neutrophil.obs[\"Cell_Type\"] = MPA2_AD_Neutrophil.obs[\"Cell_Type\"].astype(\"category\")\n",
    "\n",
    "print(MPA2_AD_Neutrophil.obs['Cell_Type'].value_counts())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d3304968-0e7e-49d3-9138-2bf596c79ad3",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "MPA2_AD_Neutrophil.obs['NP_Only'] = MPA2_AD_Neutrophil.obs['Cell_Type'] == 'Neutrophil MPO+'\n",
    "sc.pl.umap(MPA2_AD_Neutrophil, color='NP_Only', palette = 'Reds', size = 30, title = 'Neutrophil MPO+')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1bbb3754-63b6-4617-a666-c1e1dd32f354",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    MPA2_AD_Neutrophil,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"Cell_Type\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    title = 'Dotplot of Neutrophil specific markers - Lam 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "25ae2d1d-e10e-47e5-8a4d-0028ef7d9dda",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.normalize_total(MPA3_AD, target_sum=1e4)\n",
    "sc.pp.log1p(MPA3_AD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "93eb8d3a-069c-4cb7-a29b-8c7bc0a5118f",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.neighbors(MPA3_AD)\n",
    "sc.tl.umap(MPA3_AD)\n",
    "sc.tl.leiden(MPA3_AD,flavor = \"igraph\", resolution=0.5, n_iterations = 2)\n",
    "sc.tl.dendrogram(MPA3_AD, groupby = \"leiden\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c111f671-32cb-4ad6-9ad9-31aca2505b9c",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(MPA3_AD,color=['leiden'], legend_fontsize=8)\n",
    "sc.pl.umap(MPA3_AD,color=['dataset'], legend_fontsize=8)\n",
    "sc.pl.umap(MPA3_AD,color=['sample'], legend_fontsize=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ca7879cc-52cd-4334-9fae-5509c7b374cc",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(MPA3_AD,color=['MPO','ELANE','SLPI','ITGAM','SELL','ICAM1','CXCL10','CXCL1','CXCL2','CEACAM1'], legend_fontsize=8, cmap ='Reds' )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a2ba84cb-172c-4419-afc5-8cc47cb0b2aa",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    MPA3_AD,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"leiden\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    label = 'Dotplot of Neutrophil specific markers - LAM 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d1d6347a-967a-47b7-bbee-5e371167550a",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "MPA3_AD.obs[\"Cell_Type\"] = \"Other\"\n",
    "MPA3_AD.obs[\"Cell_Type\"] = MPA3_AD.obs[\"Cell_Type\"].astype(\"category\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d0f0037f-37be-483b-8d84-b9051e27dbe9",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "MPA3_AD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ec608253-ceae-4d15-b2aa-08b0e25bb9c2",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "# Utility function to safely extract gene expression values\n",
    "# Step 1: Subset epithelial only\n",
    "MPA3_AD_Neutrophil = MPA3_AD.copy()\n",
    "\n",
    "def get_gene_expression(adata, gene_name):\n",
    "    \"\"\"Extract expression values for a given gene safely, accounting for sparse matrices.\"\"\"\n",
    "    if gene_name in adata.var_names:\n",
    "        gene_idx = adata.var_names.get_loc(gene_name)\n",
    "        values = adata.X[:, gene_idx].toarray() if sp.issparse(adata.X) else adata.X[:, gene_idx]\n",
    "        return values.flatten()\n",
    "    else:\n",
    "        print(f\"Warning: Gene {gene_name} not found in dataset.\")\n",
    "        return np.zeros(adata.shape[0])\n",
    "\n",
    "#Neutrophil\n",
    "MPO_values = get_gene_expression(MPA3_AD_Neutrophil, 'MPO')\n",
    "ELANE_values = get_gene_expression(MPA3_AD_Neutrophil, 'ELANE')\n",
    "\n",
    "if \"Neutrophil MPO+\" not in MPA3_AD_Neutrophil.obs[\"Cell_Type\"].cat.categories:\n",
    "    MPA3_AD_Neutrophil.obs[\"Cell_Type\"] = MPA3_AD_Neutrophil.obs[\"Cell_Type\"].cat.add_categories([\"Neutrophil MPO+\"]) \n",
    "\n",
    "neutrophil_mask = (MPO_values > 0.001)\n",
    "MPA3_AD_Neutrophil.obs.loc[neutrophil_mask, 'Cell_Type'] = \"Neutrophil MPO+\"\n",
    "\n",
    "MPA3_AD_Neutrophil.obs[\"Cell_Type\"] = MPA3_AD_Neutrophil.obs[\"Cell_Type\"].astype(\"category\")\n",
    "\n",
    "print(MPA3_AD_Neutrophil.obs['Cell_Type'].value_counts())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c2eeb68b-2b38-4c1b-b4d1-7d6abcdea61d",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "MPA3_AD_Neutrophil.obs['NP_Only'] = MPA3_AD_Neutrophil.obs['Cell_Type'] == 'Neutrophil MPO+'\n",
    "sc.pl.umap(MPA3_AD_Neutrophil, color='NP_Only', palette = 'Reds', size = 30, title = 'Neutrophil MPO+')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "450c9825-b394-40bd-86be-1a917b463eb7",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    MPA3_AD_Neutrophil,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"Cell_Type\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    title = 'Dotplot of Neutrophil specific markers - Lam 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b22830f9-b124-4fc0-ac5f-f32ef125275a",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.normalize_total(MPA4_AD, target_sum=1e4)\n",
    "sc.pp.log1p(MPA4_AD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3e89684c-ddd8-4074-af2d-158cc1969ed4",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.neighbors(MPA4_AD)\n",
    "sc.tl.umap(MPA4_AD)\n",
    "sc.tl.leiden(MPA4_AD,flavor = \"igraph\", resolution=0.5, n_iterations = 2)\n",
    "sc.tl.dendrogram(MPA4_AD, groupby = \"leiden\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7c57d2a1-b0d9-434d-9321-b4a092f53655",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(MPA4_AD,color=['leiden'], legend_fontsize=8)\n",
    "sc.pl.umap(MPA4_AD,color=['dataset'], legend_fontsize=8)\n",
    "sc.pl.umap(MPA4_AD,color=['sample'], legend_fontsize=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7088b0ab-54bf-4f9f-a1f2-e7e3acdae028",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(MPA4_AD,color=['MPO','ELANE','SLPI','ITGAM','SELL','ICAM1','CXCL10','CXCL1','CXCL2','CEACAM1'], legend_fontsize=8, cmap ='Reds' )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bc15716d-2949-4c7b-9aa5-b952dbbffbca",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    MPA4_AD,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"leiden\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    label = 'Dotplot of Neutrophil specific markers - LAM 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "305a620d-c10d-4519-bf40-8f36cb0e578a",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "MPA4_AD.obs[\"Cell_Type\"] = \"Other\"\n",
    "MPA4_AD.obs[\"Cell_Type\"] = MPA4_AD.obs[\"Cell_Type\"].astype(\"category\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "332b0711-86b5-4e20-bc22-c9e5d630bd84",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "MPA4_AD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "589e8894-c1b6-46c7-8075-f30567294540",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "# Utility function to safely extract gene expression values\n",
    "# Step 1: Subset epithelial only\n",
    "MPA4_AD_Neutrophil = MPA4_AD.copy()\n",
    "\n",
    "def get_gene_expression(adata, gene_name):\n",
    "    \"\"\"Extract expression values for a given gene safely, accounting for sparse matrices.\"\"\"\n",
    "    if gene_name in adata.var_names:\n",
    "        gene_idx = adata.var_names.get_loc(gene_name)\n",
    "        values = adata.X[:, gene_idx].toarray() if sp.issparse(adata.X) else adata.X[:, gene_idx]\n",
    "        return values.flatten()\n",
    "    else:\n",
    "        print(f\"Warning: Gene {gene_name} not found in dataset.\")\n",
    "        return np.zeros(adata.shape[0])\n",
    "\n",
    "#Neutrophil\n",
    "MPO_values = get_gene_expression(MPA4_AD_Neutrophil, 'MPO')\n",
    "ELANE_values = get_gene_expression(MPA4_AD_Neutrophil, 'ELANE')\n",
    "\n",
    "if \"Neutrophil MPO+\" not in MPA4_AD_Neutrophil.obs[\"Cell_Type\"].cat.categories:\n",
    "    MPA4_AD_Neutrophil.obs[\"Cell_Type\"] = MPA4_AD_Neutrophil.obs[\"Cell_Type\"].cat.add_categories([\"Neutrophil MPO+\"]) \n",
    "\n",
    "neutrophil_mask = (MPO_values > 0.001)\n",
    "MPA4_AD_Neutrophil.obs.loc[neutrophil_mask, 'Cell_Type'] = \"Neutrophil MPO+\"\n",
    "\n",
    "MPA4_AD_Neutrophil.obs[\"Cell_Type\"] = MPA4_AD_Neutrophil.obs[\"Cell_Type\"].astype(\"category\")\n",
    "\n",
    "print(MPA4_AD_Neutrophil.obs['Cell_Type'].value_counts())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "29ace795-8772-4151-8b3e-ac8723f7f134",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "MPA4_AD_Neutrophil.obs['NP_Only'] = MPA4_AD_Neutrophil.obs['Cell_Type'] == 'Neutrophil MPO+'\n",
    "sc.pl.umap(MPA4_AD_Neutrophil, color='NP_Only', palette = 'Reds', size = 30, title = 'Neutrophil MPO+')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a588cd55-e837-4104-9fdc-a90be827808f",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    MPA4_AD_Neutrophil,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"Cell_Type\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    title = 'Dotplot of Neutrophil specific markers - Lam 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "54e7c116-772a-4b64-86ef-11452e42022b",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.normalize_total(MPA5_AD, target_sum=1e4)\n",
    "sc.pp.log1p(MPA5_AD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65d6cddc-763f-4cc0-9784-8e60290aa61f",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.neighbors(MPA5_AD)\n",
    "sc.tl.umap(MPA5_AD)\n",
    "sc.tl.leiden(MPA5_AD,flavor = \"igraph\", resolution=0.5, n_iterations = 2)\n",
    "sc.tl.dendrogram(MPA5_AD, groupby = \"leiden\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa270f69-f767-44c5-988e-09f0ac2a2293",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(MPA5_AD,color=['leiden'], legend_fontsize=8)\n",
    "sc.pl.umap(MPA5_AD,color=['dataset'], legend_fontsize=8)\n",
    "sc.pl.umap(MPA5_AD,color=['sample'], legend_fontsize=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6f4ab0ca-569c-4077-8ebb-95658d1d8a59",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(MPA5_AD,color=['MPO','ELANE','SLPI','ITGAM','SELL','ICAM1','CXCL10','CXCL1','CXCL2','CEACAM1'], legend_fontsize=8, cmap ='Reds' )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "df234010-5898-4998-a083-0c90d078caf3",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    MPA5_AD,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"leiden\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    label = 'Dotplot of Neutrophil specific markers - LAM 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa1dfc16-5a96-48a9-b989-b3e1bc1e6a85",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "MPA5_AD.obs[\"Cell_Type\"] = \"Other\"\n",
    "MPA5_AD.obs[\"Cell_Type\"] = MPA5_AD.obs[\"Cell_Type\"].astype(\"category\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9c24c390-d3b8-4505-8d40-b51b76780184",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "MPA5_AD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4818e0dc-2f5d-47ef-9a44-549e2db7b6cd",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "# Utility function to safely extract gene expression values\n",
    "# Step 1: Subset epithelial only\n",
    "MPA5_AD_Neutrophil = MPA5_AD.copy()\n",
    "\n",
    "def get_gene_expression(adata, gene_name):\n",
    "    \"\"\"Extract expression values for a given gene safely, accounting for sparse matrices.\"\"\"\n",
    "    if gene_name in adata.var_names:\n",
    "        gene_idx = adata.var_names.get_loc(gene_name)\n",
    "        values = adata.X[:, gene_idx].toarray() if sp.issparse(adata.X) else adata.X[:, gene_idx]\n",
    "        return values.flatten()\n",
    "    else:\n",
    "        print(f\"Warning: Gene {gene_name} not found in dataset.\")\n",
    "        return np.zeros(adata.shape[0])\n",
    "\n",
    "#Neutrophil\n",
    "MPO_values = get_gene_expression(MPA5_AD_Neutrophil, 'MPO')\n",
    "ELANE_values = get_gene_expression(MPA5_AD_Neutrophil, 'ELANE')\n",
    "\n",
    "if \"Neutrophil MPO+\" not in MPA5_AD_Neutrophil.obs[\"Cell_Type\"].cat.categories:\n",
    "    MPA5_AD_Neutrophil.obs[\"Cell_Type\"] = MPA5_AD_Neutrophil.obs[\"Cell_Type\"].cat.add_categories([\"Neutrophil MPO+\"]) \n",
    "\n",
    "neutrophil_mask = (MPO_values > 0.001)\n",
    "MPA5_AD_Neutrophil.obs.loc[neutrophil_mask, 'Cell_Type'] = \"Neutrophil MPO+\"\n",
    "\n",
    "MPA5_AD_Neutrophil.obs[\"Cell_Type\"] = MPA5_AD_Neutrophil.obs[\"Cell_Type\"].astype(\"category\")\n",
    "\n",
    "print(MPA5_AD_Neutrophil.obs['Cell_Type'].value_counts())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b6ea58f5-dc77-4b3a-91da-393c7ec0882a",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "MPA5_AD_Neutrophil.obs['NP_Only'] = MPA5_AD_Neutrophil.obs['Cell_Type'] == 'Neutrophil MPO+'\n",
    "sc.pl.umap(MPA5_AD_Neutrophil, color='NP_Only', palette = 'Reds', size = 30, title = 'Neutrophil MPO+')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "52e3157b-ed79-4f27-aff3-ec4823558703",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    MPA5_AD_Neutrophil,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"Cell_Type\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    title = 'Dotplot of Neutrophil specific markers - Lam 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "225c570e-5c27-48f6-b437-258cf8f23f65",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Joining All_LAM with reference Neutrophil Dataset (E-GEAD-867)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2dd9f297-d36e-4c6b-915f-496017cee487",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "All_LAM = sc.read_h5ad('/gpfs01/home/mzxcs8/Extra/Neutrophil LAM Phenotype - Johnson Lab/All_LAM.h5ad')\n",
    "EGEAD867_LAM = sc.read_h5ad('/gpfs01/home/mzxcs8/Extra/Neutrophil LAM Phenotype - Johnson Lab/EGEAD867_LAM.h5ad')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b8842348-55bd-4395-bd8a-5be1db774f31",
   "metadata": {
    "jupyter": {
     "source_hidden": true
    },
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "# fixed_pipeline.py\n",
    "import mygene\n",
    "import pandas as pd\n",
    "import scanpy as sc\n",
    "import numpy as np\n",
    "\n",
    "mg = mygene.MyGeneInfo()\n",
    "\n",
    "# =====================================\n",
    "# Step 0: Collect datasets\n",
    "# =====================================\n",
    "datasets = {\n",
    "    'All_LAM': All_LAM,\n",
    "    'EGEAD867_LAM': EGEAD867_LAM   \n",
    "}\n",
    "\n",
    "# Build reference gene set as union (keep any gene present in any dataset)\n",
    "reference_genes = set().union(*(ad.var_names for ad in datasets.values()))\n",
    "print(f\"Reference gene set (union): {len(reference_genes)} genes\")\n",
    "\n",
    "# =====================================\n",
    "# Step 1: Gene mapping\n",
    "# =====================================\n",
    "def build_gene_mapping(gene_list, reference_set, dataset_name):\n",
    "    \"\"\"\n",
    "    Map Ensembl IDs, symbols, or aliases to HGNC symbols,\n",
    "    prioritizing reference_set membership.\n",
    "\n",
    "    Returns:\n",
    "        mapping (dict): query -> mapped ID\n",
    "        unmapped_df (pd.DataFrame): queries that could not be mapped\n",
    "    \"\"\"\n",
    "    # Query mygene (ask for both 'ensembl.gene' and 'ensembl' to capture both formats)\n",
    "    res = mg.querymany(\n",
    "        list(gene_list),\n",
    "        scopes=[\"ensembl.gene\", \"symbol\", \"alias\", \"ensembl\"],\n",
    "        fields=\"symbol,ensembl.gene,ensembl,alias\",\n",
    "        species=\"human\",\n",
    "        as_dataframe=True,\n",
    "    )\n",
    "\n",
    "    mapping = {}\n",
    "    unmapped_records = []  # store info for dataframe\n",
    "\n",
    "    # iterate rows (each query)\n",
    "    for query, row in res.iterrows():\n",
    "        # mygene returns a 'notfound' column for missing queries\n",
    "        if isinstance(row, pd.Series) and row.get(\"notfound\"):\n",
    "            unmapped_records.append(\n",
    "                {\"dataset\": dataset_name, \"query\": query, \"reason\": \"notfound\"}\n",
    "            )\n",
    "            continue\n",
    "\n",
    "        # Collect all possible identifiers\n",
    "        candidates = set()\n",
    "\n",
    "        # symbol\n",
    "        symbol = row.get(\"symbol\")\n",
    "        if pd.notna(symbol):\n",
    "            candidates.add(symbol)\n",
    "\n",
    "        # ensembl.gene (string or list)\n",
    "        ens_gene = row.get(\"ensembl.gene\")\n",
    "        if pd.notna(ens_gene):\n",
    "            if isinstance(ens_gene, list):\n",
    "                candidates.update(ens_gene)\n",
    "            else:\n",
    "                candidates.add(ens_gene)\n",
    "\n",
    "        # ensembl (list of dicts or single dict) -> extract 'gene' keys\n",
    "        ens_full = row.get(\"ensembl\")\n",
    "        if pd.notna(ens_full):\n",
    "            if isinstance(ens_full, dict) and \"gene\" in ens_full:\n",
    "                candidates.add(ens_full[\"gene\"])\n",
    "            elif isinstance(ens_full, list):\n",
    "                for entry in ens_full:\n",
    "                    if isinstance(entry, dict) and \"gene\" in entry:\n",
    "                        candidates.add(entry[\"gene\"])\n",
    "\n",
    "        # alias (string or list)\n",
    "        aliases = row.get(\"alias\")\n",
    "        if pd.notna(aliases):\n",
    "            if isinstance(aliases, list):\n",
    "                candidates.update(aliases)\n",
    "            else:\n",
    "                candidates.add(aliases)\n",
    "\n",
    "        # Try to match against reference set (priority)\n",
    "        match = next((c for c in candidates if c in reference_set), None)\n",
    "\n",
    "        if match:\n",
    "            mapping[query] = match\n",
    "        elif symbol:  # fallback: use symbol if present\n",
    "            mapping[query] = symbol\n",
    "        else:\n",
    "            unmapped_records.append(\n",
    "                {\"dataset\": dataset_name, \"query\": query, \"reason\": \"no_match_in_reference\"}\n",
    "            )\n",
    "\n",
    "    unmapped_df = pd.DataFrame(unmapped_records)\n",
    "    # ensure consistent columns even if empty\n",
    "    if unmapped_df.empty:\n",
    "        unmapped_df = pd.DataFrame(columns=[\"dataset\", \"query\", \"reason\"])\n",
    "    return mapping, unmapped_df\n",
    "\n",
    "\n",
    "# =====================================\n",
    "# Step 2: Harmonize + merge duplicates\n",
    "# =====================================\n",
    "def harmonize_and_merge(adata, reference_set, dataset_name, merge_method=\"sum\", convert_sparse=True):\n",
    "    \"\"\"\n",
    "    Rename genes to reference symbols, merge duplicates.\n",
    "    Returns (adata_harmonized, unmapped_df).\n",
    "    Parameters:\n",
    "      - merge_method: \"sum\" (default) or \"mean\"\n",
    "      - convert_sparse: if True, transiently convert sparse matrix to dense for grouping;\n",
    "                        if False, will try a memory-friendly sparse approach (not implemented here).\n",
    "    \"\"\"\n",
    "    print(f\"\\n--- Harmonizing {dataset_name} ---\")\n",
    "    mapping, unmapped_df = build_gene_mapping(adata.var_names, reference_set, dataset_name)\n",
    "\n",
    "    # Keep a copy of original var names\n",
    "    adata.var[\"old_names\"] = adata.var_names\n",
    "\n",
    "    # Apply mapping (map queries to mapped name or keep original)\n",
    "    new_names = adata.var_names.to_series().map(lambda g: mapping.get(g, g))\n",
    "    adata.var_names = new_names\n",
    "\n",
    "    # If duplicates exist, merge them\n",
    "    if adata.var_names.has_duplicates:\n",
    "        ndup = adata.var_names.duplicated().sum()\n",
    "        print(f\"{dataset_name}: {ndup} duplicate gene names detected  merging ({merge_method})\")\n",
    "\n",
    "        # Convert to dense carefully  only if requested and if size allows\n",
    "        if convert_sparse and hasattr(adata.X, \"toarray\"):\n",
    "            X = adata.X.T.toarray()  # genes x cells\n",
    "        else:\n",
    "            # fallback to numpy conversion (may still be large)\n",
    "            X = np.asarray(adata.X.T)\n",
    "\n",
    "        # Create DataFrame with genes x cells\n",
    "        df = pd.DataFrame(X, index=adata.var_names, columns=adata.obs_names)\n",
    "\n",
    "        # Group by gene name and aggregate\n",
    "        if merge_method == \"sum\":\n",
    "            df2 = df.groupby(df.index).sum()\n",
    "        elif merge_method == \"mean\":\n",
    "            df2 = df.groupby(df.index).mean()\n",
    "        else:\n",
    "            raise ValueError(\"merge_method must be 'sum' or 'mean'\")\n",
    "\n",
    "        # Re-create AnnData (cells x genes)\n",
    "        adata = sc.AnnData(df2.T, obs=adata.obs.copy())\n",
    "        adata.var_names = df2.index\n",
    "        adata.var_names.name = None\n",
    "\n",
    "    # Final sanity-check: ensure var_names are strings\n",
    "    adata.var_names = adata.var_names.astype(str)\n",
    "\n",
    "    print(f\"{dataset_name}: {adata.n_vars} genes after harmonization\")\n",
    "    return adata, unmapped_df\n",
    "\n",
    "\n",
    "# =====================================\n",
    "# Step 3: Apply harmonization & track unmapped\n",
    "# =====================================\n",
    "all_unmapped_list = []\n",
    "harmonized_datasets = {}\n",
    "\n",
    "for name, ad in datasets.items():\n",
    "    adata_harmonized, unmapped_df = harmonize_and_merge(ad.copy(), reference_genes, name)\n",
    "    harmonized_datasets[name] = adata_harmonized\n",
    "    all_unmapped_list.append(unmapped_df)\n",
    "\n",
    "# Concatenate unmapped into one DataFrame\n",
    "all_unmapped = pd.concat(all_unmapped_list, ignore_index=True)\n",
    "\n",
    "# =====================================\n",
    "# Step 4: Concatenate (intersection or union as desired)\n",
    "# =====================================\n",
    "# Use join=\"outer\" for union, \"inner\" for intersection across datasets\n",
    "All_REF_LAM = sc.concat(\n",
    "    list(harmonized_datasets.values()),\n",
    "    join=\"outer\",\n",
    "    label=\"dataset\",\n",
    "    keys=list(harmonized_datasets.keys()),\n",
    ")\n",
    "\n",
    "print(\"\\nFinal concatenated AnnData:\")\n",
    "print(f\"Shape: {All_REF_LAM.shape}\")   # (cells, genes)\n",
    "\n",
    "# Show unmapped summary\n",
    "print(\"\\n Unmapped genes summary:\")\n",
    "display(all_unmapped.head())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b376fc95-93e4-4686-8dc0-905fe2eddc90",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "All_REF_LAM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9a2b1a69-df47-4f6c-88cd-143bd0842f5e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "All_REF_LAM.obs['sample'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9a51890a-5c87-42e2-ac80-25a7923471b4",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "All_REF_LAM.obs['dataset'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cb9aa7b2-cac4-471f-aeac-ca6c1e281aff",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "# Run PCA\n",
    "#sc.pp.scale(AT2_HLCA)                # optional but recommended\n",
    "sc.tl.pca(All_REF_LAM, svd_solver=\"arpack\")\n",
    "\n",
    "sce.pp.harmony_integrate(All_REF_LAM, key = 'dataset')\n",
    "'X_pca_harmony' in All_REF_LAM.obsm\n",
    "True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bda5899a-ea05-42e2-b743-844142c4d612",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "print(f\"Total counts in lcc_h_filt: {All_REF_LAM.X.sum()}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "25fc13e5-9ca9-4b8b-8c11-879927ebfadc",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.normalize_total(All_REF_LAM, target_sum=1e4)\n",
    "sc.pp.log1p(All_REF_LAM)\n",
    "\n",
    "#if sp.issparse(AT2_HLCA.X):\n",
    "#    AT2_HLCA.X = AT2_HLCA.X.toarray()  # Convert sparse to dense matrix\n",
    "#AT2_HLCA.X.shape\n",
    "\n",
    "All_REF_LAM.obsm['X_pca'] = All_REF_LAM.obsm['X_pca_harmony']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "551ee142-9936-49ce-aa7e-d763742ef315",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.neighbors(All_REF_LAM)\n",
    "sc.tl.umap(All_REF_LAM)\n",
    "sc.tl.leiden(All_REF_LAM,flavor = \"igraph\", resolution=0.5, n_iterations = 2)\n",
    "sc.tl.dendrogram(All_REF_LAM, groupby = \"leiden\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "643ab6b4-ea36-48a1-9700-16a911341529",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "sc.pl.umap(All_REF_LAM,color=['leiden'], legend_fontsize=8)\n",
    "sc.pl.umap(All_REF_LAM,color=['dataset'], legend_fontsize=8)\n",
    "sc.pl.umap(All_REF_LAM,color=['sample'], legend_fontsize=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "480f9109-91eb-419c-b777-ec74e4d9ffbf",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "sc.pl.umap(All_REF_LAM,color=['MPO','ITGAM','SELL','ICAM1','CXCL10','CXCL1','CXCL2','CEACAM1'], legend_fontsize=20, cmap = 'Reds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c749fdf6-3d16-464b-84bf-9db17f018ab6",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.pl.dotplot(\n",
    "    All_REF_LAM,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"Cell_Type\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    label = 'Dotplot of Neutrophil specific markers - Lam 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71df6060-c651-495e-8472-5bb3a130be8f",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "\n",
    "sc.pl.dotplot(\n",
    "    All_REF_LAM,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"sample\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    label = 'Dotplot of Neutrophil specific markers - Lam 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3c67ed66-1350-407c-89cb-e977d1898b5f",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "\n",
    "sc.pl.dotplot(\n",
    "    All_REF_LAM,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"sample\",'Cell_Type'], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    label = 'Dotplot of Neutrophil specific markers - Lam 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5e664f26-80ef-4425-b488-fe7bad860d4d",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "\n",
    "sc.pl.dotplot(\n",
    "    All_REF_LAM,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"dataset\",'Cell_Type'], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    label = 'Dotplot of Neutrophil specific markers - Lam 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "11b2e777-fbd8-4393-b6db-fd0b1f40167b",
   "metadata": {
    "tags": []
   },
   "source": [
    "## MPO+ Cells Only"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "61342302-a9ce-45f6-b181-2e09bcf27d4c",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "All_NeutroMPO = All_REF_LAM[All_REF_LAM.obs[\"Cell_Type\"] == \"Neutrophil MPO+\"].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7d4ffb1c-6af4-42e8-830b-49863dfa45e2",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pp.neighbors(All_NeutroMPO)\n",
    "sc.tl.umap(All_NeutroMPO)\n",
    "sc.tl.leiden(All_NeutroMPO,flavor = \"igraph\", resolution=0.5, n_iterations = 2)\n",
    "sc.tl.dendrogram(All_NeutroMPO, groupby = \"leiden\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6cc2f678-732b-44b2-8e42-930f93028a9a",
   "metadata": {
    "tags": [
     "remove-cell"
    ]
   },
   "outputs": [],
   "source": [
    "#| remove-cell\n",
    "sc.pl.umap(All_NeutroMPO,color=['leiden'], legend_fontsize=8)\n",
    "sc.pl.umap(All_NeutroMPO,color=['dataset'], legend_fontsize=8)\n",
    "sc.pl.umap(All_NeutroMPO,color=['sample'], legend_fontsize=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "478d8a6d-2dfa-4dd5-bf0a-86d42147cdbe",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "sc.pl.umap(All_NeutroMPO,color=['MPO','ITGAM','SELL','ICAM1','CXCL10','CXCL1','CXCL2','CEACAM1'], legend_fontsize=20, cmap = 'Reds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "497da194-7a8d-4eb2-9cef-cac490882578",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.tl.dendrogram(All_NeutroMPO, groupby = ['sample'])\n",
    "sc.pl.dotplot(\n",
    "    All_NeutroMPO,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"sample\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    label = 'Dotplot of Neutrophil specific markers - Lam 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cfa164d7-44e3-4213-bd86-91f49ea6c713",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "\n",
    "sc.pl.dotplot(\n",
    "    All_NeutroMPO,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"dataset\"], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    label = 'Dotplot of Neutrophil specific markers - Lam 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f6b78617-ba16-4d9c-83f4-80646c3f1094",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "\n",
    "sc.pl.dotplot(\n",
    "    All_NeutroMPO,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"dataset\",'Cell_Type'], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    label = 'Dotplot of Neutrophil specific markers - Lam 2 Samples [GSE135851]'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b8940c4b-d7ff-4422-b379-949c3daddb16",
   "metadata": {
    "tags": [
     "hide-input"
    ]
   },
   "outputs": [],
   "source": [
    "#| hide-input\n",
    "marker_Neutrophil = {\n",
    "    \"Neutrophil\": ['MPO','ELANE','FCGR3B','NAMPT','IFITM2','TNFAIP2','ITGAM','FUT4','CXCR3','FCGR3A'],\n",
    "    \"Neutrophil N1\": [\"ITGAM\",\"FCGR3B\",\"CEACAM1\",\"GPI\",\"ICAM1\",\"CXCL1\",'CXCL2','CXCL3','CXCL10','TNF','MYD88'],\n",
    "    \"Neutrophil N2\": ['ITGAM','FCGR3B','CEACAM1','GPI','SELL','MRC1','PI3','SLPI']\n",
    "}\n",
    "sc.tl.dendrogram(All_NeutroMPO, groupby = ['sample','Cell_Type'])\n",
    "sc.pl.dotplot(\n",
    "    All_NeutroMPO,\n",
    "    marker_Neutrophil,\n",
    "    groupby = [\"sample\",'Cell_Type'], \n",
    "    dendrogram=True,\n",
    "    use_raw=False,\n",
    "    label = 'Dotplot of Neutrophil specific markers - Lam 2 Samples [GSE135851]'\n",
    ")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (scRNASeqHannan)",
   "language": "python",
   "name": "scrnaseqhannan"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
