
<!DOCTYPE html>


<html lang="en" data-content_root="../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>GProfiler &amp; Gene Set Enrichment Analysis &#8212; Hannan Lab Single Cell Seq Analysis</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css?v=56efc87c" />
    <link rel="stylesheet" type="text/css" href="../_static/.ipynb_checkpoints/custom-checkpoint.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../_static/copybutton.js?v=f281be69"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'content/GSEA';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="prev" title="Final Annotation" href="Final_annotation.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="Intro.html">
  
  
  
  
  
  
    <p class="title logo__title">Hannan Lab Single Cell Seq Analysis</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="Intro.html">
                    Intro slide
                </a>
            </li>
        </ul>
        <ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="Preprocessing.html">Filter and Process Single Cell Object</a></li>
<li class="toctree-l1"><a class="reference internal" href="Validation.html">CellTypist Validation (Healthy)</a></li>



<li class="toctree-l1"><a class="reference internal" href="RNA_velocity.html">RNA Velocity Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="Final_annotation.html">Final Annotation</a></li>
<li class="toctree-l1 current active"><a class="current reference internal" href="#">GProfiler &amp; Gene Set Enrichment Analysis</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/content/GSEA.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>GProfiler & Gene Set Enrichment Analysis</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#late-at2-like-cells">Late AT2 Like Cells</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gsea-cell-type-top-250-down-250">GSEA - Cell Type [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gsea-cell-type-top-500">GSEA - Cell Type [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gsea-gene-ontology-top-250-down-250">GSEA - Gene Ontology [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gsea-gene-ontology-top-500">GSEA - Gene Ontology [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gsea-pathway-analysis-top-250-down-250">GSEA - Pathway Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gsea-pathway-analysis-top-500">GSEA - Pathway Analysis [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gsea-proteomic-analysis-top-250-down-250">GSEA - Proteomic Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gsea-proteomic-analysis-top-500">GSEA - Proteomic Analysis [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gsea-disease-analysis-top-250-down-250">GSEA - Disease Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gsea-disease-analysis-top-500">GSEA - Disease Analysis [Top 500]</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#early-at2-like-cells">Early AT2 Like Cells</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">GSEA - Cell Type [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">GSEA - Cell Type [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id3">GSEA - Gene Ontology [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id4">GSEA - Gene Ontology [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id5">GSEA - Pathway Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id6">GSEA - Pathway Analysis [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id7">GSEA - Proteomic Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id8">GSEA - Proteomic Analysis [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id9">GSEA - Disease Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id10">GSEA - Disease Analysis [Top 500]</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#proliferating-at2-like-cells">Proliferating AT2 Like Cells</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id11">GSEA - Cell Type [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id12">GSEA - Cell Type [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id13">GSEA - Gene Ontology [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id14">GSEA - Gene Ontology [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id15">GSEA - Pathway Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id16">GSEA - Pathway Analysis [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id17">GSEA - Proteomic Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id18">GSEA - Proteomic Analysis [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id19">GSEA - Disease Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id20">GSEA - Disease Analysis [Top 500]</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#ascl1-at2-like-cells">‘ASCL1+’ -AT2 Like Cells</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id21">GSEA - Cell Type [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id22">GSEA - Cell Type [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id23">GSEA - Gene Ontology [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id24">GSEA - Gene Ontology [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id25">GSEA - Pathway Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id26">GSEA - Pathway Analysis [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id27">GSEA - Proteomic Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id28">GSEA - Proteomic Analysis [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id29">GSEA - Disease Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id30">GSEA - Disease Analysis [Top 500]</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="gprofiler-gene-set-enrichment-analysis">
<h1>GProfiler &amp; Gene Set Enrichment Analysis<a class="headerlink" href="#gprofiler-gene-set-enrichment-analysis" title="Link to this heading">#</a></h1>
<p>This is the statistical analysis done after the final annotation of cell clusters was done on the sample.</p>
<p>A wilcoxon test was used to analyse the differences between annotated clusters of cells.</p>
<p>GProfiler was used to further characterise the top 250 variable genes from each cluster. This gave us results from multiple datasets (Gene Ontology, KEGG, Reactome, and WikiPathways).
A bar chart is shown with the top results after filtering the gene list:</p>
<ul class="simple">
<li><p>LogFC Expression &gt; 0.01</p></li>
<li><p>Adjusted P-Value &lt; 0.05</p></li>
</ul>
<p>Gene Set Enrichment Analysis (GSEA) further validates and provides an extra statistical test with the gene list of top 250 variables genes.
For this analysis the top 250 upregulated genes and top 250 downregulated genes were obtained, a list was made ranking them based on their variablity (positive, or negative) and then used to further get a more statistical significant result of the gene list given in comparison to only using GProfiler. Additionally, the top 500 genes were used to further study the effect of ranking only the upregulated genes.
A dot plot is showing the top results after filtering the gene set enrichment result with only using the terms that had an adjusted p-value (FDR q-val) &lt; 0.05. Additionally, a gsea plot was done on the top 5 enriched results. This should be validated with the dotplot to confirm that the gsea plot results show actual significant terms.</p>
<p>Databases used for GSEA Analysis:</p>
<ul class="simple">
<li><p>Cell Types</p>
<ul>
<li><p>‘Azimuth_2023’</p></li>
<li><p>‘CellMarker_2024’</p></li>
<li><p>‘Descartes_Cell_Types_and_Tissue_2021’</p></li>
<li><p>‘Human_Gene_Atlas’</p></li>
<li><p>‘Human_Phenotype_Ontology’</p></li>
</ul>
</li>
<li><p>Gene Ontology</p>
<ul>
<li><p>‘GO_Biological_Process_2025’</p></li>
<li><p>‘GO_Cellular_Component_2025’</p></li>
<li><p>‘GO_Molecular_Function_2025’</p></li>
</ul>
</li>
<li><p>Pathway Analysis</p>
<ul>
<li><p>‘Reactome_Pathways_2024’</p></li>
<li><p>‘KEGG_2021_Human’</p></li>
<li><p>‘WikiPathways_2024_Human’</p></li>
</ul>
</li>
<li><p>Proteomic Analysis</p>
<ul>
<li><p>‘ProteomicsDB_2020’</p></li>
<li><p>‘Tissue_Protein_Expression_from_Human_Proteome_Map’</p></li>
<li><p>‘Tissue_Protein_Expression_from_ProteomicsDB’</p></li>
</ul>
</li>
<li><p>Disease Analysis</p>
<ul>
<li><p>‘DisGeNET’</p></li>
<li><p>‘Disease_Perturbations_from_GEO_down’</p></li>
<li><p>‘Disease_Perturbations_from_GEO_up’</p></li>
<li><p>‘HDSigDB_Human_2021’</p></li>
</ul>
</li>
</ul>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">dendrogram</span><span class="p">(</span><span class="n">at2_h_lp_filt</span><span class="p">,</span> <span class="n">groupby</span> <span class="o">=</span> <span class="s1">&#39;HL_annot_final&#39;</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">rank_genes_groups</span><span class="p">(</span><span class="n">at2_h_lp_filt</span><span class="p">,</span> <span class="n">groupby</span> <span class="o">=</span> <span class="s1">&#39;HL_annot_final&#39;</span><span class="p">,</span><span class="n">method</span> <span class="o">=</span> <span class="s1">&#39;wilcoxon&#39;</span><span class="p">)</span>
<span class="c1">#sc.pl.rank_genes_groups_heatmap(at2_h_lp_filt, n_genes=20,cmap=&#39;turbo&#39;,show_gene_labels=True)</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">sc</span><span class="o">.</span><span class="n">pl</span><span class="o">.</span><span class="n">rank_genes_groups_heatmap</span><span class="p">,</span>
    <span class="s2">&quot;gsea_1&quot;</span><span class="p">,</span>
    <span class="n">at2_h_lp_filt</span><span class="p">,</span>
    <span class="n">n_genes</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span>
    <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span>
    <span class="n">show_gene_labels</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;Most variable and differentially expressed genes between cell clusters.&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;scanpy&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/rank_genes_groups_heatmap_gsea_1.png" alt="gsea_1" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>Most variable and differentially expressed genes between cell clusters.</em><br>
        <a href="_static/Figures/rank_genes_groups_heatmap_gsea_1.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<section id="late-at2-like-cells">
<h2>Late AT2 Like Cells<a class="headerlink" href="#late-at2-like-cells" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">numpy</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">np</span>

<span class="n">deg_results</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({</span>
    <span class="s1">&#39;names&#39;</span><span class="p">:</span> <span class="n">at2_h_lp_filt</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;rank_genes_groups&#39;</span><span class="p">][</span><span class="s1">&#39;names&#39;</span><span class="p">][</span><span class="s1">&#39;Late AT2 Like Cells&#39;</span><span class="p">],</span>    <span class="c1"># Gene names</span>
    <span class="s1">&#39;logfoldchanges&#39;</span><span class="p">:</span> <span class="n">at2_h_lp_filt</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;rank_genes_groups&#39;</span><span class="p">][</span><span class="s1">&#39;logfoldchanges&#39;</span><span class="p">][</span><span class="s1">&#39;Late AT2 Like Cells&#39;</span><span class="p">],</span>     <span class="c1"># Log fold changes</span>
    <span class="s1">&#39;pvals&#39;</span><span class="p">:</span> <span class="n">at2_h_lp_filt</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;rank_genes_groups&#39;</span><span class="p">][</span><span class="s1">&#39;pvals&#39;</span><span class="p">][</span><span class="s1">&#39;Late AT2 Like Cells&#39;</span><span class="p">],</span>          <span class="c1"># p-values</span>
    <span class="s1">&#39;pvals_adj&#39;</span><span class="p">:</span> <span class="n">at2_h_lp_filt</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;rank_genes_groups&#39;</span><span class="p">][</span><span class="s1">&#39;pvals_adj&#39;</span><span class="p">][</span><span class="s1">&#39;Late AT2 Like Cells&#39;</span><span class="p">]</span> <span class="c1"># Adjusted p-values (FDR corrected)</span>
<span class="p">})</span>

<span class="c1"># Step 1: Set thresholds for logFC and p-value (adjusted)</span>
<span class="n">logfc_threshold</span> <span class="o">=</span> <span class="mf">0.01</span>   <span class="c1"># Example: filter for logFC &gt; 1 (or &lt; -1 for downregulated)</span>
<span class="n">pval_adj_threshold</span> <span class="o">=</span> <span class="mf">0.05</span>  <span class="c1"># Example: adjusted p-value &lt; 0.05</span>

<span class="c1"># Step 2: Apply the filtering based on logFC and adjusted p-value</span>
<span class="n">filtered_genes</span> <span class="o">=</span> <span class="n">deg_results</span><span class="p">[</span>
    <span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">abs</span><span class="p">(</span><span class="n">deg_results</span><span class="p">[</span><span class="s1">&#39;logfoldchanges&#39;</span><span class="p">])</span> <span class="o">&gt;=</span> <span class="n">logfc_threshold</span><span class="p">)</span> <span class="o">&amp;</span>  <span class="c1"># Absolute logFC above threshold</span>
    <span class="p">(</span><span class="n">deg_results</span><span class="p">[</span><span class="s1">&#39;pvals_adj&#39;</span><span class="p">]</span> <span class="o">&lt;</span> <span class="n">pval_adj_threshold</span><span class="p">)</span>               <span class="c1"># Adjusted p-value below threshold</span>
<span class="p">]</span>

<span class="c1"># Step 3: Extract the list of filtered gene names</span>
<span class="n">filtered_gene_names</span> <span class="o">=</span> <span class="n">filtered_genes</span><span class="p">[</span><span class="s1">&#39;names&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
<span class="c1">#filtered_gene_names[:50]</span>

<span class="c1"># Run g:Profiler</span>
<span class="n">go_results</span> <span class="o">=</span> <span class="n">gp</span><span class="o">.</span><span class="n">profile</span><span class="p">(</span><span class="n">organism</span><span class="o">=</span><span class="s1">&#39;hsapiens&#39;</span><span class="p">,</span> <span class="n">query</span><span class="o">=</span><span class="n">filtered_gene_names</span><span class="p">[:</span><span class="mi">250</span><span class="p">])</span>

<span class="c1"># Define the sources you want to loop over</span>
<span class="n">sources</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;GO:BP&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:MF&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:CC&#39;</span><span class="p">,</span> <span class="s1">&#39;KEGG&#39;</span><span class="p">,</span> <span class="s1">&#39;REAC&#39;</span><span class="p">,</span> <span class="s1">&#39;HPA&#39;</span><span class="p">,</span> <span class="s1">&#39;WP&#39;</span><span class="p">]</span>

<span class="c1"># Create a stacked figure with one row per source</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">nrows</span><span class="o">=</span><span class="nb">len</span><span class="p">(</span><span class="n">sources</span><span class="p">),</span> <span class="n">ncols</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">5.5</span><span class="p">,</span> <span class="mf">2.5</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">sources</span><span class="p">)))</span>
<span class="n">axes</span> <span class="o">=</span> <span class="n">axes</span><span class="o">.</span><span class="n">flatten</span><span class="p">()</span>

<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">source</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">sources</span><span class="p">):</span>
    <span class="c1"># Filter results for this source</span>
    <span class="n">specific_results</span> <span class="o">=</span> <span class="n">go_results</span><span class="p">[</span><span class="n">go_results</span><span class="p">[</span><span class="s1">&#39;source&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">source</span><span class="p">]</span>
    
    <span class="k">if</span> <span class="n">specific_results</span><span class="o">.</span><span class="n">empty</span><span class="p">:</span>
        <span class="k">continue</span>  <span class="c1"># skip if no results for this source</span>
    
    <span class="c1"># Select top 8–10 terms</span>
    <span class="n">go_results_top</span> <span class="o">=</span> <span class="p">(</span>
        <span class="n">specific_results</span><span class="p">[[</span><span class="s1">&#39;name&#39;</span><span class="p">,</span> <span class="s1">&#39;p_value&#39;</span><span class="p">]]</span>
        <span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;p_value&#39;</span><span class="p">)</span>
        <span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">8</span><span class="p">)</span>
    <span class="p">)</span>

    <span class="n">labels</span> <span class="o">=</span> <span class="n">go_results_top</span><span class="p">[</span><span class="s1">&#39;name&#39;</span><span class="p">]</span>
    <span class="n">values</span> <span class="o">=</span> <span class="o">-</span><span class="n">np</span><span class="o">.</span><span class="n">log10</span><span class="p">(</span><span class="n">go_results_top</span><span class="p">[</span><span class="s1">&#39;p_value&#39;</span><span class="p">])</span>

    <span class="n">ax</span> <span class="o">=</span> <span class="n">axes</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">barh</span><span class="p">(</span><span class="n">labels</span><span class="p">,</span> <span class="n">values</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s2">&quot;#1f77b4&quot;</span><span class="p">)</span>
    
    <span class="c1"># Aesthetics</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">source</span><span class="si">}</span><span class="s2"> enrichment&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span> <span class="n">loc</span><span class="o">=</span><span class="s2">&quot;left&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;-log10(P-value)&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">invert_yaxis</span><span class="p">()</span>  <span class="c1"># highest term on top</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="s2">&quot;y&quot;</span><span class="p">,</span> <span class="n">labelsize</span><span class="o">=</span><span class="mi">9</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="s2">&quot;x&quot;</span><span class="p">,</span> <span class="n">labelsize</span><span class="o">=</span><span class="mi">9</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s2">&quot;top&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="kc">False</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s2">&quot;right&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="kc">False</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">grid</span><span class="p">(</span><span class="kc">False</span><span class="p">)</span>
<span class="c1"># Adjust layout</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="c1"># Save if needed</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_1&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GProfiler Analysis of top 250 differentially expressed genes from Late AT2 Like Cells.&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_lateAT2_1.png" alt="gsea_lateAT2_1" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GProfiler Analysis of top 250 differentially expressed genes from Late AT2 Like Cells.</em><br>
        <a href="_static/Figures/gsea_lateAT2_1.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>

<span class="n">group</span> <span class="o">=</span> <span class="s2">&quot;Late AT2 Like Cells&quot;</span>  <span class="c1"># change for each group</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">get</span><span class="o">.</span><span class="n">rank_genes_groups_df</span><span class="p">(</span><span class="n">at2_h_lp_filt</span><span class="p">,</span> <span class="n">group</span><span class="o">=</span><span class="n">group</span><span class="p">)</span>

<span class="c1"># GSEAPY prerank needs index=gene, score=stat/logFC</span>
<span class="n">ranked_list</span> <span class="o">=</span> <span class="n">df</span><span class="p">[[</span><span class="s1">&#39;names&#39;</span><span class="p">,</span> <span class="s1">&#39;scores&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;names&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">squeeze</span><span class="p">()</span>

<span class="n">df_ranked_list_top</span> <span class="o">=</span> <span class="n">ranked_list</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">500</span><span class="p">]</span>

<span class="n">ranked_list_pos</span> <span class="o">=</span> <span class="n">ranked_list</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">250</span><span class="p">]</span>
<span class="n">ranked_list_neg</span> <span class="o">=</span> <span class="n">ranked_list</span><span class="p">[(</span><span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">)</span><span class="o">-</span><span class="mi">250</span><span class="p">):</span><span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">)]</span>

<span class="n">ranked_lists</span> <span class="o">=</span> <span class="p">[</span><span class="n">ranked_list_pos</span><span class="p">,</span><span class="n">ranked_list_neg</span><span class="p">]</span>

<span class="n">df_ranked_list</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">ranked_lists</span><span class="p">)</span>
<span class="c1"># Save to .rnk if needed</span>
<span class="c1">#ranked_list.to_csv(&quot;lateAT2.rnk&quot;, sep=&quot;\t&quot;)</span>
<span class="n">df_ranked_list</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>names
SFTPB       94.647346
SLC34A2     75.418358
LRRK2       65.422585
NEAT1       61.137497
TACC2       60.737698
              ...    
PTMA       -34.993572
SOX5       -35.191536
MKI67      -35.620640
GRM8       -38.168594
MIR924HG   -44.057720
Name: scores, Length: 500, dtype: float32
</pre></div>
</div>
</div>
</div>
<section id="gsea-cell-type-top-250-down-250">
<h3>GSEA - Cell Type [Top 250 &amp; Down 250]<a class="headerlink" href="#gsea-cell-type-top-250-down-250" title="Link to this heading">#</a></h3>
<p>‘Azimuth_2023’,’CellMarker_2024’,’Descartes_Cell_Types_and_Tissue_2021’,’Human_Gene_Atlas’,’Human_Phenotype_Ontology’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Azimuth_2023&#39;</span><span class="p">,</span><span class="s1">&#39;CellMarker_2024&#39;</span><span class="p">,</span><span class="s1">&#39;Descartes_Cell_Types_and_Tissue_2021&#39;</span><span class="p">,</span><span class="s1">&#39;Human_Gene_Atlas&#39;</span><span class="p">,</span><span class="s1">&#39;Human_Phenotype_Ontology&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">verbose</span><span class="o">=</span> <span class="kc">False</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>CellMarker_2024__Cycling Basal Cell Trachea Mouse</td>
      <td>-0.592075</td>
      <td>-4.820699</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>71/71</td>
      <td>49.40%</td>
      <td>MKI67;PTMA;DIAPH3;TOP2A;RRM2;BUB1B;CENPF;ASPM;...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>Human_Gene_Atlas__721 B lymphoblasts</td>
      <td>-0.529772</td>
      <td>-4.41392</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>87/95</td>
      <td>46.60%</td>
      <td>MKI67;PTMA;TOP2A;HELLS;RRM2;BUB1B;CENPF;ASPM;P...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>CellMarker_2024__Neural Progenitor Cell Embryo...</td>
      <td>-0.521202</td>
      <td>-3.891322</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>54/57</td>
      <td>46.00%</td>
      <td>MKI67;TOP2A;HELLS;RRM2;CENPF;ASPM;ATAD2;KIF11;...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>CellMarker_2024__Fibroblast Lung Mouse</td>
      <td>-0.665021</td>
      <td>-3.882005</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>21/28</td>
      <td>22.00%</td>
      <td>MKI67;TOP2A;RRM2;CENPF;ATAD2;TUBB;STMN1;TPX2;C...</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>Human_Gene_Atlas__CD105+ Endothelial</td>
      <td>-0.565215</td>
      <td>-3.406501</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>27/30</td>
      <td>40.80%</td>
      <td>TOP2A;RRM2;CENPF;FANCI;RFC3;PRC1;POLA1;KIF4A;S...</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>313</th>
      <td>prerank</td>
      <td>CellMarker_2024__Meiotic Prophase Fetal Germ C...</td>
      <td>0.129186</td>
      <td>0.527521</td>
      <td>0.964989</td>
      <td>0.988986</td>
      <td>1.0</td>
      <td>7/36</td>
      <td>20.00%</td>
      <td>IFT57;ITSN2;SCEL;PTPN13;MACF1;GLCCI1;CD55</td>
    </tr>
    <tr>
      <th>314</th>
      <td>prerank</td>
      <td>Human_Gene_Atlas__Colorectaladenocarcinoma</td>
      <td>0.154421</td>
      <td>0.519651</td>
      <td>0.968905</td>
      <td>0.986999</td>
      <td>1.0</td>
      <td>7/16</td>
      <td>41.80%</td>
      <td>BCL2L1;ABLIM1;CCND1;HIP1;CD9;KRT18;KRT8</td>
    </tr>
    <tr>
      <th>315</th>
      <td>prerank</td>
      <td>CellMarker_2024__Morula Cell (Blastomere) Embr...</td>
      <td>0.211889</td>
      <td>0.487489</td>
      <td>0.97882</td>
      <td>0.993035</td>
      <td>1.0</td>
      <td>2/5</td>
      <td>35.60%</td>
      <td>CFLAR;GULP1</td>
    </tr>
    <tr>
      <th>316</th>
      <td>prerank</td>
      <td>Human_Phenotype_Ontology__Malar flattening (HP...</td>
      <td>0.155254</td>
      <td>0.450657</td>
      <td>0.98545</td>
      <td>0.996417</td>
      <td>1.0</td>
      <td>1/10</td>
      <td>4.60%</td>
      <td>PDE4D</td>
    </tr>
    <tr>
      <th>317</th>
      <td>prerank</td>
      <td>Human_Phenotype_Ontology__Skin ulcer (HP:0200042)</td>
      <td>0.179383</td>
      <td>0.433584</td>
      <td>0.991254</td>
      <td>0.994214</td>
      <td>1.0</td>
      <td>3/6</td>
      <td>44.60%</td>
      <td>PROS1;STAT3;LAMB3</td>
    </tr>
  </tbody>
</table>
<p>318 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Cell Type - Late AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_2&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Cell Type Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_lateAT2_2.png" alt="gsea_lateAT2_2" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Cell Type Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_lateAT2_2.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_3&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Cell Type Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_lateAT2_3.png" alt="gsea_lateAT2_3" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Cell Type Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_lateAT2_3.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="gsea-cell-type-top-500">
<h3>GSEA - Cell Type [Top 500]<a class="headerlink" href="#gsea-cell-type-top-500" title="Link to this heading">#</a></h3>
<p>‘Azimuth_2023’,’CellMarker_2024’,’Descartes_Cell_Types_and_Tissue_2021’,’Human_Gene_Atlas’,’Human_Phenotype_Ontology’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list_top</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Azimuth_2023&#39;</span><span class="p">,</span><span class="s1">&#39;CellMarker_2024&#39;</span><span class="p">,</span><span class="s1">&#39;Descartes_Cell_Types_and_Tissue_2021&#39;</span><span class="p">,</span><span class="s1">&#39;Human_Gene_Atlas&#39;</span><span class="p">,</span><span class="s1">&#39;Human_Phenotype_Ontology&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>CellMarker_2024__Type II Pneumocyte Lung Mouse</td>
      <td>0.859676</td>
      <td>2.589167</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>7/8</td>
      <td>9.80%</td>
      <td>SFTPB;SLC34A2;FASN;SFTPA1;SFTPC;LAMP3;SCD</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>Descartes_Cell_Types_and_Tissue_2021__Bronchio...</td>
      <td>0.727916</td>
      <td>2.449858</td>
      <td>0.0</td>
      <td>0.001665</td>
      <td>0.003</td>
      <td>7/11</td>
      <td>7.60%</td>
      <td>SFTPB;SLC34A2;ROS1;SFTPC;LAMP3;ACOXL;PGC</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>Human_Gene_Atlas__Lung</td>
      <td>0.48197</td>
      <td>2.431196</td>
      <td>0.0</td>
      <td>0.00148</td>
      <td>0.004</td>
      <td>16/38</td>
      <td>13.00%</td>
      <td>SFTPB;SLC34A2;TACC2;DLC1;AKAP13;RHOBTB2;SCNN1A...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>Azimuth_2023__Lung v1-L1-Alveolar Epithelial T...</td>
      <td>0.702977</td>
      <td>2.280919</td>
      <td>0.001357</td>
      <td>0.004996</td>
      <td>0.018</td>
      <td>6/9</td>
      <td>12.00%</td>
      <td>SFTPB;SLC34A2;NAPSA;SFTPC;LAMP3;ABCA3</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>CellMarker_2024__Alveolar Type II (ATII) Cell ...</td>
      <td>0.878788</td>
      <td>2.179111</td>
      <td>0.0</td>
      <td>0.013323</td>
      <td>0.057</td>
      <td>5/5</td>
      <td>13.00%</td>
      <td>SFTPB;SFTPA1;SFTPC;LAMP3;MUC1</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>262</th>
      <td>prerank</td>
      <td>Human_Gene_Atlas__CD56+ NKCells</td>
      <td>0.11977</td>
      <td>0.504412</td>
      <td>0.969484</td>
      <td>0.99278</td>
      <td>1.0</td>
      <td>7/21</td>
      <td>32.00%</td>
      <td>AKAP13;MACF1;CCND3;CFLAR;LPCAT1;JAK1;TAOK3</td>
    </tr>
    <tr>
      <th>263</th>
      <td>prerank</td>
      <td>CellMarker_2024__Enterocyte Progenitor Cell Sm...</td>
      <td>0.162368</td>
      <td>0.493866</td>
      <td>0.987705</td>
      <td>0.992163</td>
      <td>1.0</td>
      <td>3/8</td>
      <td>30.60%</td>
      <td>ASAH1;MGST1;PON2</td>
    </tr>
    <tr>
      <th>264</th>
      <td>prerank</td>
      <td>Human_Gene_Atlas__Liver</td>
      <td>0.168627</td>
      <td>0.485202</td>
      <td>0.980142</td>
      <td>0.990552</td>
      <td>1.0</td>
      <td>3/7</td>
      <td>35.20%</td>
      <td>FTL;ARHGEF38;CYB5A</td>
    </tr>
    <tr>
      <th>265</th>
      <td>prerank</td>
      <td>CellMarker_2024__Astrocyte Brain Mouse</td>
      <td>0.087913</td>
      <td>0.443533</td>
      <td>0.985808</td>
      <td>0.996421</td>
      <td>1.0</td>
      <td>22/42</td>
      <td>52.40%</td>
      <td>ATP13A4;MAST4;SDC4;PTPN13;MACF1;MAOA;ACSL3;MSM...</td>
    </tr>
    <tr>
      <th>266</th>
      <td>prerank</td>
      <td>Human_Phenotype_Ontology__Skin ulcer (HP:0200042)</td>
      <td>0.158341</td>
      <td>0.42572</td>
      <td>0.99564</td>
      <td>0.994826</td>
      <td>1.0</td>
      <td>4/6</td>
      <td>57.40%</td>
      <td>PROS1;STAT3;LAMB3;LYST</td>
    </tr>
  </tbody>
</table>
<p>267 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Cell Type - Late AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">6</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_4&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Cell Type Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_lateAT2_4.png" alt="gsea_lateAT2_4" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Cell Type Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_lateAT2_4.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_5&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Cell Type Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_lateAT2_5.png" alt="gsea_lateAT2_5" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Cell Type Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_lateAT2_5.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="gsea-gene-ontology-top-250-down-250">
<h3>GSEA - Gene Ontology [Top 250 &amp; Down 250]<a class="headerlink" href="#gsea-gene-ontology-top-250-down-250" title="Link to this heading">#</a></h3>
<p>‘GO_Biological_Process_2025’,’GO_Cellular_Component_2025’,’GO_Molecular_Function_2025’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;GO_Biological_Process_2025&#39;</span><span class="p">,</span><span class="s1">&#39;GO_Cellular_Component_2025&#39;</span><span class="p">,</span><span class="s1">&#39;GO_Molecular_Function_2025&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__DNA Metabolic Proc...</td>
      <td>-0.49494</td>
      <td>-3.452484</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>34/40</td>
      <td>40.20%</td>
      <td>TOP2A;BRIP1;FAM111B;RFC3;POLA1;HMGB2;RAD51B;CD...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Mitotic Sister Chr...</td>
      <td>-0.583807</td>
      <td>-3.356685</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>22/28</td>
      <td>29.80%</td>
      <td>KIF11;NCAPG2;TPX2;CENPE;PRC1;NUSAP1;KIF15;NCAP...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__DNA Repair (GO:000...</td>
      <td>-0.550372</td>
      <td>-3.292992</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>31/33</td>
      <td>42.00%</td>
      <td>BRIP1;HMGA2;FANCI;RFC3;POLA1;RAD51B;FANCD2;CDK...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>GO_Cellular_Component_2025__Chromosome (GO:000...</td>
      <td>-0.61476</td>
      <td>-3.14649</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>16/19</td>
      <td>31.80%</td>
      <td>MKI67;TOP2A;HMGA2;CENPE;H1-5;EZH2;HMGB2;FANCD2...</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>GO_Cellular_Component_2025__Spindle (GO:0005819)</td>
      <td>-0.528267</td>
      <td>-3.060495</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>23/27</td>
      <td>36.80%</td>
      <td>BUB1B;CENPF;TUBB;KIF11;TPX2;CENPE;PRC1;KIF4A;N...</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>524</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Positive Regulatio...</td>
      <td>0.120637</td>
      <td>0.488759</td>
      <td>0.979098</td>
      <td>0.991755</td>
      <td>1.0</td>
      <td>7/32</td>
      <td>23.40%</td>
      <td>LMO3;MICAL2;MUC1;KAT2B;EPAS1;CITED2;SQSTM1</td>
    </tr>
    <tr>
      <th>525</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Response to Endopl...</td>
      <td>0.210402</td>
      <td>0.480385</td>
      <td>0.989264</td>
      <td>0.991168</td>
      <td>1.0</td>
      <td>3/5</td>
      <td>48.40%</td>
      <td>RNF145;XBP1;PTPN1</td>
    </tr>
    <tr>
      <th>526</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Regulation of Tran...</td>
      <td>0.107029</td>
      <td>0.476072</td>
      <td>0.981857</td>
      <td>0.989351</td>
      <td>1.0</td>
      <td>14/53</td>
      <td>29.20%</td>
      <td>HOPX;LMO3;MICAL2;MUC1;FNIP2;KAT2B;EPAS1;CITED2...</td>
    </tr>
    <tr>
      <th>527</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Generation of Neur...</td>
      <td>-0.165657</td>
      <td>-0.439242</td>
      <td>1.0</td>
      <td>0.99907</td>
      <td>1.0</td>
      <td>5/5</td>
      <td>83.80%</td>
      <td>ADGRL3;FGF13;RACGAP1;RAPGEF2;ACSL3</td>
    </tr>
    <tr>
      <th>528</th>
      <td>prerank</td>
      <td>GO_Molecular_Function_2025__mRNA Binding (GO:0...</td>
      <td>0.132924</td>
      <td>0.394529</td>
      <td>1.0</td>
      <td>0.998089</td>
      <td>1.0</td>
      <td>11/11</td>
      <td>87.00%</td>
      <td>SECISBP2L;SLPI;RBMS1;IGF2BP2;ELAVL2;ZC3H12B;MS...</td>
    </tr>
  </tbody>
</table>
<p>529 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Gene Ontology - Late AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_6&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Gene Ontology Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_lateAT2_6.png" alt="gsea_lateAT2_6" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Gene Ontology Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_lateAT2_6.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_7&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Gene Ontology Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_lateAT2_7.png" alt="gsea_lateAT2_7" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Gene Ontology Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_lateAT2_7.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="gsea-gene-ontology-top-500">
<h3>GSEA - Gene Ontology [Top 500]<a class="headerlink" href="#gsea-gene-ontology-top-500" title="Link to this heading">#</a></h3>
<p>‘GO_Biological_Process_2025’,’GO_Cellular_Component_2025’,’GO_Molecular_Function_2025’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list_top</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;GO_Biological_Process_2025&#39;</span><span class="p">,</span><span class="s1">&#39;GO_Cellular_Component_2025&#39;</span><span class="p">,</span><span class="s1">&#39;GO_Molecular_Function_2025&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>GO_Cellular_Component_2025__Alveolar Lamellar ...</td>
      <td>0.921212</td>
      <td>2.358564</td>
      <td>0.0</td>
      <td>0.011732</td>
      <td>0.009</td>
      <td>5/5</td>
      <td>8.80%</td>
      <td>SFTPB;NAPSA;SFTPC;LAMP3;CTSH</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>GO_Cellular_Component_2025__Lamellar Body (GO:...</td>
      <td>0.818053</td>
      <td>2.340262</td>
      <td>0.0</td>
      <td>0.006933</td>
      <td>0.011</td>
      <td>6/7</td>
      <td>12.00%</td>
      <td>SFTPB;SFTPA1;NAPSA;SFTPC;CTSH;ABCA3</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>GO_Cellular_Component_2025__Multivesicular Bod...</td>
      <td>0.680657</td>
      <td>2.2419</td>
      <td>0.0</td>
      <td>0.019553</td>
      <td>0.045</td>
      <td>6/10</td>
      <td>8.80%</td>
      <td>SFTPB;LRRK2;SFTPA1;NAPSA;SFTPC;CTSH</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Heart Development ...</td>
      <td>0.671635</td>
      <td>1.838283</td>
      <td>0.009943</td>
      <td>0.76072</td>
      <td>0.841</td>
      <td>5/6</td>
      <td>22.80%</td>
      <td>DLC1;AKAP13;MICAL2;KAT2B;CITED2</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>GO_Molecular_Function_2025__Flavin Adenine Din...</td>
      <td>0.631592</td>
      <td>1.811562</td>
      <td>0.005472</td>
      <td>0.750214</td>
      <td>0.894</td>
      <td>5/7</td>
      <td>18.60%</td>
      <td>DPYD;MICAL2;ACOXL;MAOA;STEAP4</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>500</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Organophosphate Me...</td>
      <td>0.188785</td>
      <td>0.507777</td>
      <td>0.970501</td>
      <td>0.984935</td>
      <td>1.0</td>
      <td>4/6</td>
      <td>55.20%</td>
      <td>ALOX15B;PFKFB2;HOOK3;GFPT1</td>
    </tr>
    <tr>
      <th>501</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Positive Regulatio...</td>
      <td>0.106495</td>
      <td>0.505009</td>
      <td>0.961582</td>
      <td>0.983483</td>
      <td>1.0</td>
      <td>13/31</td>
      <td>40.60%</td>
      <td>LMO3;MICAL2;MUC1;KAT2B;EPAS1;CITED2;SQSTM1;ZBT...</td>
    </tr>
    <tr>
      <th>502</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Positive Regulatio...</td>
      <td>0.108391</td>
      <td>0.495429</td>
      <td>0.970149</td>
      <td>0.984459</td>
      <td>1.0</td>
      <td>7/28</td>
      <td>23.40%</td>
      <td>LMO3;MICAL2;MUC1;KAT2B;EPAS1;CITED2;SQSTM1</td>
    </tr>
    <tr>
      <th>503</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Regulation of Prot...</td>
      <td>0.14705</td>
      <td>0.483716</td>
      <td>0.984106</td>
      <td>0.985838</td>
      <td>1.0</td>
      <td>9/10</td>
      <td>77.80%</td>
      <td>NAPSA;FNIP2;SELENOP;MPP7;PRKCZ;CD36;LAMP2;TANK...</td>
    </tr>
    <tr>
      <th>504</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__ERAD Pathway (GO:0...</td>
      <td>0.185272</td>
      <td>0.457549</td>
      <td>0.996951</td>
      <td>0.989884</td>
      <td>1.0</td>
      <td>1/5</td>
      <td>10.80%</td>
      <td>RNF145</td>
    </tr>
  </tbody>
</table>
<p>505 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Gene Ontology - Late AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">6</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_8&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Gene Ontology Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_lateAT2_8.png" alt="gsea_lateAT2_8" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Gene Ontology Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_lateAT2_8.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_9&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Gene Ontology Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_lateAT2_9.png" alt="gsea_lateAT2_9" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Gene Ontology Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_lateAT2_9.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="gsea-pathway-analysis-top-250-down-250">
<h3>GSEA - Pathway Analysis [Top 250 &amp; Down 250]<a class="headerlink" href="#gsea-pathway-analysis-top-250-down-250" title="Link to this heading">#</a></h3>
<p>‘Reactome_Pathways_2024’,’KEGG_2021_Human’,’WikiPathways_2024_Human’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Reactome_Pathways_2024&#39;</span><span class="p">,</span><span class="s1">&#39;KEGG_2021_Human&#39;</span><span class="p">,</span><span class="s1">&#39;WikiPathways_2024_Human&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Cell Cycle</td>
      <td>-0.45299</td>
      <td>-4.205819</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>65/82</td>
      <td>39.80%</td>
      <td>TOP2A;RRM2;BRIP1;BUB1B;CENPF;CENPP;TUBB;LMNB1;...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Cell Cycle, Mitotic</td>
      <td>-0.410575</td>
      <td>-3.549785</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>52/68</td>
      <td>39.80%</td>
      <td>TOP2A;RRM2;BUB1B;CENPF;CENPP;TUBB;LMNB1;RFC3;N...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__M Phase</td>
      <td>-0.521411</td>
      <td>-3.489979</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>29/37</td>
      <td>35.40%</td>
      <td>BUB1B;CENPF;CENPP;TUBB;LMNB1;NCAPG2;CENPE;TMPO...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Cell Cycle Checkpoints</td>
      <td>-0.47873</td>
      <td>-3.437459</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>36/42</td>
      <td>41.20%</td>
      <td>BRIP1;BUB1B;CENPF;CENPP;RFC3;CLSPN;CENPE;CDK1;...</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__DNA Repair</td>
      <td>-0.555493</td>
      <td>-3.380749</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>29/32</td>
      <td>42.00%</td>
      <td>BRIP1;POLQ;FANCI;RFC3;CLSPN;RAD51B;FANCD2;BRCA...</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>476</th>
      <td>prerank</td>
      <td>KEGG_2021_Human__Axon guidance</td>
      <td>0.199281</td>
      <td>0.536583</td>
      <td>0.970711</td>
      <td>0.978097</td>
      <td>1.0</td>
      <td>3/8</td>
      <td>36.80%</td>
      <td>ABLIM1;PLXNA2;EPHA4</td>
    </tr>
    <tr>
      <th>477</th>
      <td>prerank</td>
      <td>KEGG_2021_Human__Huntington disease</td>
      <td>0.20938</td>
      <td>0.51547</td>
      <td>0.979885</td>
      <td>0.983798</td>
      <td>1.0</td>
      <td>3/6</td>
      <td>42.60%</td>
      <td>IFT57;HIP1;MAPK10</td>
    </tr>
    <tr>
      <th>478</th>
      <td>prerank</td>
      <td>WikiPathways_2024_Human__Pleural Mesothelioma ...</td>
      <td>0.129915</td>
      <td>0.497861</td>
      <td>0.966555</td>
      <td>0.986749</td>
      <td>1.0</td>
      <td>7/26</td>
      <td>30.20%</td>
      <td>TNIK;CCND3;CDKN1A;CCND2;MDM2;CCND1;JAK1</td>
    </tr>
    <tr>
      <th>479</th>
      <td>prerank</td>
      <td>WikiPathways_2024_Human__7Q11 23 Copy Number V...</td>
      <td>0.180894</td>
      <td>0.497622</td>
      <td>0.983516</td>
      <td>0.983565</td>
      <td>1.0</td>
      <td>8/8</td>
      <td>82.20%</td>
      <td>PRKG1;SQSTM1;CLDN1;PCNA;HMGA1;GRIP1;DEK;SMARCA5</td>
    </tr>
    <tr>
      <th>480</th>
      <td>prerank</td>
      <td>KEGG_2021_Human__Adherens junction</td>
      <td>0.214363</td>
      <td>0.496852</td>
      <td>0.989568</td>
      <td>0.980516</td>
      <td>1.0</td>
      <td>3/5</td>
      <td>49.20%</td>
      <td>PTPRM;PTPN1;PTPRJ</td>
    </tr>
  </tbody>
</table>
<p>481 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Pathway Analysis - Late AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">6</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_10&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Pathway Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_lateAT2_10.png" alt="gsea_lateAT2_10" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Pathway Analysis Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_lateAT2_10.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_11&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Pathway Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_lateAT2_11.png" alt="gsea_lateAT2_11" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Pathway Analysis Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_lateAT2_11.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="gsea-pathway-analysis-top-500">
<h3>GSEA - Pathway Analysis [Top 500]<a class="headerlink" href="#gsea-pathway-analysis-top-500" title="Link to this heading">#</a></h3>
<p>‘Reactome_Pathways_2024’,’KEGG_2021_Human’,’WikiPathways_2024_Human’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list_top</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Reactome_Pathways_2024&#39;</span><span class="p">,</span><span class="s1">&#39;KEGG_2021_Human&#39;</span><span class="p">,</span><span class="s1">&#39;WikiPathways_2024_Human&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Surfactant Metabolism</td>
      <td>0.79297</td>
      <td>2.614236</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>8/10</td>
      <td>12.00%</td>
      <td>SFTPB;SLC34A2;ADGRF5;SFTPA1;NAPSA;SFTPC;CTSH;A...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>KEGG_2021_Human__Salmonella infection</td>
      <td>-0.451013</td>
      <td>-2.173291</td>
      <td>0.0</td>
      <td>0.299323</td>
      <td>0.163</td>
      <td>14/15</td>
      <td>48.60%</td>
      <td>ACBD3;ARPC2;FLNB;RAB5A;ARF1;ARPC1B;DNM2;CYTH1;...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Diseases Associated Wi...</td>
      <td>0.748719</td>
      <td>2.161974</td>
      <td>0.0</td>
      <td>0.050412</td>
      <td>0.079</td>
      <td>5/7</td>
      <td>12.00%</td>
      <td>SFTPB;SLC34A2;SFTPA1;SFTPC;ABCA3</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__PI Metabolism</td>
      <td>-0.439706</td>
      <td>-1.931159</td>
      <td>0.012552</td>
      <td>0.727538</td>
      <td>0.506</td>
      <td>9/12</td>
      <td>24.20%</td>
      <td>RAB14;PIP4K2C;GDE1;RAB5A;INPP4A;ARF1;PLEKHA5;S...</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>WikiPathways_2024_Human__Nuclear Receptors Met...</td>
      <td>0.429458</td>
      <td>1.895393</td>
      <td>0.007009</td>
      <td>0.504123</td>
      <td>0.569</td>
      <td>15/25</td>
      <td>29.20%</td>
      <td>AKAP13;FASN;SCNN1A;STOM;SCD;ABCC4;FTL;SLC6A14;...</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>519</th>
      <td>prerank</td>
      <td>KEGG_2021_Human__Human immunodeficiency virus ...</td>
      <td>0.151173</td>
      <td>0.463809</td>
      <td>0.986613</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>8/9</td>
      <td>76.20%</td>
      <td>BCL2L1;B2M;MAPK10;MAPK1;NFKBIA;ITPR3;TAPBP;PIK3CB</td>
    </tr>
    <tr>
      <th>520</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__MAPK Family Signaling ...</td>
      <td>0.112659</td>
      <td>0.443105</td>
      <td>0.987981</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>10/18</td>
      <td>51.80%</td>
      <td>BCL2L1;CCND3;IL6ST;JAK1;CDC42EP3;ARRB1;RAPGEF2...</td>
    </tr>
    <tr>
      <th>521</th>
      <td>prerank</td>
      <td>KEGG_2021_Human__Yersinia infection</td>
      <td>0.134573</td>
      <td>0.399248</td>
      <td>0.994374</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>1/8</td>
      <td>6.80%</td>
      <td>RPS6KA2</td>
    </tr>
    <tr>
      <th>522</th>
      <td>prerank</td>
      <td>WikiPathways_2024_Human__VEGFA VEGFR2 Signalin...</td>
      <td>0.077558</td>
      <td>0.386154</td>
      <td>0.992265</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>20/37</td>
      <td>54.20%</td>
      <td>MICAL2;BCL2L1;ICAM1;PRKG1;F3;MDM2;CCND1;STAT3;...</td>
    </tr>
    <tr>
      <th>523</th>
      <td>prerank</td>
      <td>KEGG_2021_Human__Amyotrophic lateral sclerosis</td>
      <td>0.108834</td>
      <td>0.350405</td>
      <td>1.0</td>
      <td>0.999575</td>
      <td>1.0</td>
      <td>5/10</td>
      <td>47.20%</td>
      <td>BCL2L1;SQSTM1;XBP1;OPTN;CAT</td>
    </tr>
  </tbody>
</table>
<p>524 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s1">&#39;FWER p-val&#39;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Pathway Analysis - Late AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">6</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.0505</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_12&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Pathway Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_lateAT2_12.png" alt="gsea_lateAT2_12" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Pathway Analysis Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_lateAT2_12.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_13&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Pathway Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_lateAT2_13.png" alt="gsea_lateAT2_13" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Pathway Analysis Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_lateAT2_13.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="gsea-proteomic-analysis-top-250-down-250">
<h3>GSEA - Proteomic Analysis [Top 250 &amp; Down 250]<a class="headerlink" href="#gsea-proteomic-analysis-top-250-down-250" title="Link to this heading">#</a></h3>
<p>‘ProteomicsDB_2020’,’Tissue_Protein_Expression_from_Human_Proteome_Map’,’Tissue_Protein_Expression_from_ProteomicsDB’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;ProteomicsDB_2020&#39;</span><span class="p">,</span><span class="s1">&#39;Tissue_Protein_Expression_from_Human_Proteome_Map&#39;</span><span class="p">,</span><span class="s1">&#39;Tissue_Protein_Expression_from_ProteomicsDB&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Hematopoietic JURKAT BTO:00...</td>
      <td>-0.452351</td>
      <td>-3.932422</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>60/77</td>
      <td>39.80%</td>
      <td>TOP2A;RRM2;BRIP1;CENPF;CEP128;ATAD2;FANCI;RFC3...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Hematopoietic JURKAT BTO:00...</td>
      <td>-0.432125</td>
      <td>-3.698497</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>65/70</td>
      <td>50.80%</td>
      <td>WWOX;BRIP1;CENPF;ASPM;CEP128;RFC3;KIF11;CLSPN;...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Hematopoietic JURKAT BTO:00...</td>
      <td>-0.432125</td>
      <td>-3.698497</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>65/70</td>
      <td>50.80%</td>
      <td>WWOX;BRIP1;CENPF;ASPM;CEP128;RFC3;KIF11;CLSPN;...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Hematopoietic K-562 BTO:000...</td>
      <td>-0.527409</td>
      <td>-3.415757</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>24/32</td>
      <td>31.60%</td>
      <td>TOP2A;CENPF;ASPM;RFC3;CLSPN;TPX2;PRC1;EZH2;MEL...</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Hematopoietic CCRF-CEM BTO:...</td>
      <td>-0.535242</td>
      <td>-3.263965</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>26/29</td>
      <td>40.00%</td>
      <td>RRM2;TUBB;FANCI;KIF11;STMN1;TMPO;KIF4A;KIF15;S...</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>1041</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Ovary OVCAR-4 BTO:0002561 O...</td>
      <td>0.128541</td>
      <td>0.396467</td>
      <td>0.994812</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>7/12</td>
      <td>54.00%</td>
      <td>IFT57;SQSTM1;PLXNA2;RBMS1;NCOA4;SHCBP1;CDC25C</td>
    </tr>
    <tr>
      <th>1042</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_ProteomicsDB__H...</td>
      <td>0.130708</td>
      <td>0.388592</td>
      <td>0.997389</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>1/11</td>
      <td>5.60%</td>
      <td>CHSY3</td>
    </tr>
    <tr>
      <th>1043</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_ProteomicsDB__A...</td>
      <td>0.11626</td>
      <td>0.355809</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>3/12</td>
      <td>29.60%</td>
      <td>OSMR;ABCC3;COL8A1</td>
    </tr>
    <tr>
      <th>1044</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_ProteomicsDB__c...</td>
      <td>0.13836</td>
      <td>0.355558</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>3/7</td>
      <td>42.80%</td>
      <td>DGKD;PLXNA2;BTBD9</td>
    </tr>
    <tr>
      <th>1045</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Myeloma RPMI-8226 BTO:00007...</td>
      <td>0.105534</td>
      <td>0.336425</td>
      <td>1.0</td>
      <td>0.999678</td>
      <td>1.0</td>
      <td>12/14</td>
      <td>75.00%</td>
      <td>ICAM1;CCND2;PLXNA2;CD9;TKT;MAN2A1;MCM6;PCNA;CS...</td>
    </tr>
  </tbody>
</table>
<p>1046 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Proteomic Analysis - Late AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_14&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Proteomic Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_lateAT2_14.png" alt="gsea_lateAT2_14" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Proteomic Analysis Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_lateAT2_14.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_15&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Proteomic Analysis Datasets [Top 250 &amp;  Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_lateAT2_15.png" alt="gsea_lateAT2_15" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Proteomic Analysis Datasets [Top 250 &  Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_lateAT2_15.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="gsea-proteomic-analysis-top-500">
<h3>GSEA - Proteomic Analysis [Top 500]<a class="headerlink" href="#gsea-proteomic-analysis-top-500" title="Link to this heading">#</a></h3>
<p>‘ProteomicsDB_2020’,’Tissue_Protein_Expression_from_Human_Proteome_Map’,’Tissue_Protein_Expression_from_ProteomicsDB’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list_top</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;ProteomicsDB_2020&#39;</span><span class="p">,</span><span class="s1">&#39;Tissue_Protein_Expression_from_Human_Proteome_Map&#39;</span><span class="p">,</span><span class="s1">&#39;Tissue_Protein_Expression_from_ProteomicsDB&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Hematopoietic K-562 BTO:000...</td>
      <td>-0.466691</td>
      <td>-2.546998</td>
      <td>0.0</td>
      <td>0.089727</td>
      <td>0.021</td>
      <td>18/19</td>
      <td>48.80%</td>
      <td>ARL15;KIDINS220;SAR1A;BMP2K;PANK3;TM9SF2;ACACA...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_ProteomicsDB__U...</td>
      <td>-0.715291</td>
      <td>-2.149268</td>
      <td>0.0</td>
      <td>0.517362</td>
      <td>0.194</td>
      <td>5/6</td>
      <td>11.60%</td>
      <td>GGCX;RPH3AL;NDNF;TMBIM6;BMP2K</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_Human_Proteome_...</td>
      <td>0.601534</td>
      <td>2.094824</td>
      <td>0.002656</td>
      <td>0.587962</td>
      <td>0.25</td>
      <td>8/12</td>
      <td>21.00%</td>
      <td>SFTPB;SLC34A2;SFTPA1;SFTPC;LAMP3;IFT57;ITGB6;S...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Lymphoblastoid BTO:0000773 ...</td>
      <td>-0.321824</td>
      <td>-1.968178</td>
      <td>0.012903</td>
      <td>0.955816</td>
      <td>0.436</td>
      <td>19/23</td>
      <td>46.20%</td>
      <td>OS9;COPB2;GYS1;NEK6;RIOK3;NR3C1;NCOA3;LMTK2;AC...</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Renal SN-12C BTO:0004221 SN...</td>
      <td>-0.331623</td>
      <td>-1.958863</td>
      <td>0.0</td>
      <td>0.753135</td>
      <td>0.443</td>
      <td>18/21</td>
      <td>51.40%</td>
      <td>TOM1L2;CLIP1;TBC1D2;MVP;SPAG9;ZHX3;OPHN1;WWC1;...</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>1047</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Cervix HeLa BTO:0000567 Lys...</td>
      <td>0.074522</td>
      <td>0.41283</td>
      <td>0.984076</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>45/60</td>
      <td>72.80%</td>
      <td>FASN;PDE4D;SECISBP2L;SCD;PTPN13;ACSL3;ABCC3;DO...</td>
    </tr>
    <tr>
      <th>1048</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Embryonic stem HES-3 PDB:20...</td>
      <td>0.102343</td>
      <td>0.399398</td>
      <td>0.99382</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>11/17</td>
      <td>61.00%</td>
      <td>FMN1;DHCR24;IL6ST;PLXNA2;SLC39A8;TTC28;NCOA4;L...</td>
    </tr>
    <tr>
      <th>1049</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_ProteomicsDB__H...</td>
      <td>0.099671</td>
      <td>0.368442</td>
      <td>0.998767</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>11/15</td>
      <td>68.80%</td>
      <td>SCD;SCEL;MSMO1;JAK1;ABCD3;LAMB3;ITGA3;MYH14;TM...</td>
    </tr>
    <tr>
      <th>1050</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Breast MCF-7 BTO:0000093 HC...</td>
      <td>0.08247</td>
      <td>0.345512</td>
      <td>0.998788</td>
      <td>0.99942</td>
      <td>1.0</td>
      <td>1/20</td>
      <td>2.00%</td>
      <td>FASN</td>
    </tr>
    <tr>
      <th>1051</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Breast HCC-202 PDB:200003 H...</td>
      <td>0.08247</td>
      <td>0.345512</td>
      <td>0.998788</td>
      <td>0.99942</td>
      <td>1.0</td>
      <td>1/20</td>
      <td>2.00%</td>
      <td>FASN</td>
    </tr>
  </tbody>
</table>
<p>1052 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Proteomic Analysis - Late AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_16&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Proteomic Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">ValueError</span><span class="g g-Whitespace">                                </span>Traceback (most recent call last)
<span class="n">Cell</span> <span class="n">In</span><span class="p">[</span><span class="mi">84</span><span class="p">],</span> <span class="n">line</span> <span class="mi">2</span>
<span class="g g-Whitespace">      </span><span class="mi">1</span> <span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="ne">----&gt; </span><span class="mi">2</span> <span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">3</span>                  <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">4</span>                  <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Proteomic Analysis - Late AT2 Like Cells&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">5</span>                  <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">6</span>                  <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">7</span>                  <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
<span class="g g-Whitespace">      </span><span class="mi">8</span>                  <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">9</span> <span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="g g-Whitespace">     </span><span class="mi">11</span> <span class="n">sc_plot_and_embed</span><span class="p">(</span>
<span class="g g-Whitespace">     </span><span class="mi">12</span>     <span class="n">ax_dot</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">13</span>     <span class="s2">&quot;gsea_lateAT2_16&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">14</span>     <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Proteomic Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">15</span>     <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="g g-Whitespace">     </span><span class="mi">16</span> <span class="p">)</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:1225,</span> in <span class="ni">dotplot</span><span class="nt">(df, column, x, y, x_order, y_order, title, cutoff, top_term, size, ax, figsize, cmap, ofname, xticklabels_rot, yticklabels_rot, marker, show_ring, **kwargs)</span>
<span class="g g-Whitespace">   </span><span class="mi">1222</span>     <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span><span class="s2">&quot;group is deprecated; use x instead&quot;</span><span class="p">,</span> <span class="ne">DeprecationWarning</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1223</span>     <span class="k">return</span>
<span class="ne">-&gt; </span><span class="mi">1225</span> <span class="n">dot</span> <span class="o">=</span> <span class="n">DotPlot</span><span class="p">(</span>
<span class="g g-Whitespace">   </span><span class="mi">1226</span>     <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1227</span>     <span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1228</span>     <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1229</span>     <span class="n">x_order</span><span class="o">=</span><span class="n">x_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1230</span>     <span class="n">y_order</span><span class="o">=</span><span class="n">y_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1231</span>     <span class="n">hue</span><span class="o">=</span><span class="n">column</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1232</span>     <span class="n">title</span><span class="o">=</span><span class="n">title</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1233</span>     <span class="n">thresh</span><span class="o">=</span><span class="n">cutoff</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1234</span>     <span class="n">n_terms</span><span class="o">=</span><span class="nb">int</span><span class="p">(</span><span class="n">top_term</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1235</span>     <span class="n">dot_scale</span><span class="o">=</span><span class="n">size</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1236</span>     <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1237</span>     <span class="n">figsize</span><span class="o">=</span><span class="n">figsize</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1238</span>     <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1239</span>     <span class="n">ofname</span><span class="o">=</span><span class="n">ofname</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1240</span>     <span class="n">marker</span><span class="o">=</span><span class="n">marker</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1241</span> <span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1242</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dot</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">outer_ring</span><span class="o">=</span><span class="n">show_ring</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1244</span> <span class="k">if</span> <span class="n">xticklabels_rot</span><span class="p">:</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:692,</span> in <span class="ni">DotPlot.__init__</span><span class="nt">(self, df, x, y, hue, dot_scale, x_order, y_order, thresh, n_terms, title, ax, figsize, cmap, ofname, marker, **kwargs)</span>
<span class="g g-Whitespace">    </span><span class="mi">690</span> <span class="bp">self</span><span class="o">.</span><span class="n">n_terms</span> <span class="o">=</span> <span class="n">n_terms</span>
<span class="g g-Whitespace">    </span><span class="mi">691</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span> <span class="o">=</span> <span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">692</span> <span class="bp">self</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">693</span> <span class="n">plt</span><span class="o">.</span><span class="n">rcParams</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="s2">&quot;pdf.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">,</span> <span class="s2">&quot;ps.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">})</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:717,</span> in <span class="ni">DotPlot.process</span><span class="nt">(self, df)</span>
<span class="g g-Whitespace">    </span><span class="mi">715</span> <span class="k">if</span> <span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">716</span>     <span class="n">msg</span> <span class="o">=</span> <span class="s2">&quot;Warning: No enrich terms when cutoff = </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">717</span>     <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="n">msg</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">718</span> <span class="bp">self</span><span class="o">.</span><span class="n">cbar_title</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">colname</span>
<span class="g g-Whitespace">    </span><span class="mi">719</span> <span class="c1"># clip GSEA lower bounds</span>
<span class="g g-Whitespace">    </span><span class="mi">720</span> <span class="c1"># if self.colname in [&quot;NOM p-val&quot;, &quot;FDR q-val&quot;]:</span>
<span class="g g-Whitespace">    </span><span class="mi">721</span> <span class="c1">#     df[self.colname].clip(1e-5, 1.0, inplace=True)</span>
<span class="g g-Whitespace">    </span><span class="mi">722</span> <span class="c1"># sorting the dataframe for better visualization</span>

<span class="ne">ValueError</span>: Warning: No enrich terms when cutoff = 0.05
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_17&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Proteomic Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_lateAT2_17.png" alt="gsea_lateAT2_17" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Proteomic Analysis Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_lateAT2_17.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="gsea-disease-analysis-top-250-down-250">
<h3>GSEA - Disease Analysis [Top 250 &amp; Down 250]<a class="headerlink" href="#gsea-disease-analysis-top-250-down-250" title="Link to this heading">#</a></h3>
<p>‘DisGeNET’,’Disease_Perturbations_from_GEO_down’,’Disease_Perturbations_from_GEO_up’,’HDSigDB_Human_2021’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;DisGeNET&#39;</span><span class="p">,</span><span class="s1">&#39;Disease_Perturbations_from_GEO_down&#39;</span><span class="p">,</span><span class="s1">&#39;Disease_Perturbations_from_GEO_up&#39;</span><span class="p">,</span><span class="s1">&#39;HDSigDB_Human_2021&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_down__prostate ...</td>
      <td>-0.621027</td>
      <td>-5.423826</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>91/91</td>
      <td>49.40%</td>
      <td>MKI67;DIAPH3;TOP2A;HELLS;RRM2;BRIP1;BUB1B;CENP...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_down__multiple ...</td>
      <td>-0.611025</td>
      <td>-5.411356</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>96/97</td>
      <td>49.40%</td>
      <td>MKI67;TOP2A;HELLS;RRM2;BRIP1;BUB1B;CENPF;ASPM;...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Up-Regulated In Cont...</td>
      <td>-0.519685</td>
      <td>-4.926535</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>126/144</td>
      <td>46.60%</td>
      <td>MKI67;SOX5;DIAPH3;TOP2A;RRM2;BRIP1;BUB1B;CENPF...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Liver RNA M16 Module (Ligh...</td>
      <td>-0.626168</td>
      <td>-4.88624</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>72/72</td>
      <td>46.60%</td>
      <td>MKI67;DIAPH3;TOP2A;BUB1B;CENPF;ASPM;POLQ;KIF11...</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Enriched In Activate...</td>
      <td>-0.539011</td>
      <td>-4.774691</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>98/111</td>
      <td>45.00%</td>
      <td>MKI67;PTMA;TOP2A;HELLS;RRM2;NKD1;BUB1B;CENPF;A...</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>4417</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Up-Regulated In dSPN...</td>
      <td>0.083337</td>
      <td>0.389769</td>
      <td>0.998979</td>
      <td>0.998969</td>
      <td>1.0</td>
      <td>6/95</td>
      <td>6.20%</td>
      <td>DLC1;AKAP13;DPYD;PDE4D;DSTN;MAST4</td>
    </tr>
    <tr>
      <th>4418</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Down-Regulated In Ne...</td>
      <td>0.081261</td>
      <td>0.374995</td>
      <td>0.997904</td>
      <td>0.999461</td>
      <td>1.0</td>
      <td>62/63</td>
      <td>90.80%</td>
      <td>DSTN;BCL2L1;TNIK;MACF1;CDKN1A;CFLAR;ABCC3;KAT2...</td>
    </tr>
    <tr>
      <th>4419</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Expression In HD109i.1 NSC...</td>
      <td>0.068483</td>
      <td>0.341578</td>
      <td>0.997992</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>23/169</td>
      <td>16.40%</td>
      <td>DLC1;AKAP13;FASN;DRAM1;LMO3;PDE4D;FMN1;RPS6KA2...</td>
    </tr>
    <tr>
      <th>4420</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Down-Regulated In Ex...</td>
      <td>0.068717</td>
      <td>0.340107</td>
      <td>1.0</td>
      <td>0.999892</td>
      <td>1.0</td>
      <td>43/121</td>
      <td>42.20%</td>
      <td>SNX25;PARM1;SCNN1A;STOM;ACOXL;CTSH;MLPH;ABCC4;...</td>
    </tr>
    <tr>
      <th>4421</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Down-Regulated In Fr...</td>
      <td>0.061843</td>
      <td>0.295023</td>
      <td>1.0</td>
      <td>0.999945</td>
      <td>1.0</td>
      <td>27/88</td>
      <td>38.60%</td>
      <td>SNX25;STOM;DSTN;ACSL1;ITSN2;ABCC4;DGKD;PTPN13;...</td>
    </tr>
  </tbody>
</table>
<p>4422 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Pathway Analysis - Late AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_17&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Disease Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_lateAT2_17.png" alt="gsea_lateAT2_17" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Disease Analysis Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_lateAT2_17.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_18&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Disease Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_lateAT2_18.png" alt="gsea_lateAT2_18" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Disease Analysis Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_lateAT2_18.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="gsea-disease-analysis-top-500">
<h3>GSEA - Disease Analysis [Top 500]<a class="headerlink" href="#gsea-disease-analysis-top-500" title="Link to this heading">#</a></h3>
<p>‘DisGeNET’,’Disease_Perturbations_from_GEO_down’,’Disease_Perturbations_from_GEO_up’,’HDSigDB_Human_2021’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list_top</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;DisGeNET&#39;</span><span class="p">,</span><span class="s1">&#39;Disease_Perturbations_from_GEO_down&#39;</span><span class="p">,</span><span class="s1">&#39;Disease_Perturbations_from_GEO_up&#39;</span><span class="p">,</span><span class="s1">&#39;HDSigDB_Human_2021&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_down__Squamous ...</td>
      <td>0.490737</td>
      <td>2.674324</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>28/51</td>
      <td>22.80%</td>
      <td>SFTPB;SLC34A2;HOPX;DPYD;DRAM1;LMO3;SFTPC;RPS6K...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Cortex RNA M5 Module (Gree...</td>
      <td>-0.649681</td>
      <td>-2.543823</td>
      <td>0.0</td>
      <td>0.068469</td>
      <td>0.082</td>
      <td>9/10</td>
      <td>25.80%</td>
      <td>HK1;TMBIM6;EEF2;ARPC1B;GIPC1;DNM2;CYTH1;MGRN1;...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_down__pulmonary...</td>
      <td>0.616175</td>
      <td>2.50395</td>
      <td>0.0</td>
      <td>0.006372</td>
      <td>0.017</td>
      <td>11/19</td>
      <td>14.20%</td>
      <td>SFTPB;ZNF385B;LIPH;SFTPA1;ATP13A4;ROS1;SFTPC;A...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_up__pulmonary s...</td>
      <td>0.616175</td>
      <td>2.50395</td>
      <td>0.0</td>
      <td>0.006372</td>
      <td>0.017</td>
      <td>11/19</td>
      <td>14.20%</td>
      <td>SFTPB;ZNF385B;LIPH;SFTPA1;ATP13A4;ROS1;SFTPC;A...</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__AD Up-Regulated Genes In C...</td>
      <td>0.441527</td>
      <td>2.412173</td>
      <td>0.0</td>
      <td>0.017789</td>
      <td>0.06</td>
      <td>35/55</td>
      <td>33.20%</td>
      <td>AKAP13;DRAM1;RHOBTB2;PDE4D;OSMR;FMN1;RPS6KA2;B...</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>4292</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Down-Regulated In Ex...</td>
      <td>0.056736</td>
      <td>0.364009</td>
      <td>0.99389</td>
      <td>0.999598</td>
      <td>1.0</td>
      <td>42/116</td>
      <td>39.00%</td>
      <td>ZNF385B;SNX25;SCNN1A;STOM;CTSH;MLPH;ASAH1;SDC4...</td>
    </tr>
    <tr>
      <th>4293</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_up__retinoblast...</td>
      <td>0.08171</td>
      <td>0.352464</td>
      <td>0.998843</td>
      <td>0.999787</td>
      <td>1.0</td>
      <td>9/24</td>
      <td>38.40%</td>
      <td>DSTN;PTPRE;RRBP1;MSMO1;WSB1;PRKAR1A;CD63;XBP1;...</td>
    </tr>
    <tr>
      <th>4294</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_down__Spinal Co...</td>
      <td>0.090499</td>
      <td>0.346428</td>
      <td>0.998753</td>
      <td>0.999706</td>
      <td>1.0</td>
      <td>4/16</td>
      <td>25.40%</td>
      <td>DSTN;ACSL3;CD55;WSB1</td>
    </tr>
    <tr>
      <th>4295</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Up-Regulated In HD M...</td>
      <td>0.054363</td>
      <td>0.338904</td>
      <td>1.0</td>
      <td>0.999639</td>
      <td>1.0</td>
      <td>11/105</td>
      <td>11.20%</td>
      <td>HOPX;DPYD;PDE4D;ACSL1;IFT57;SECISBP2L;ITSN2;AB...</td>
    </tr>
    <tr>
      <th>4296</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Down-Regulated In HD...</td>
      <td>0.047609</td>
      <td>0.293407</td>
      <td>0.998973</td>
      <td>0.999905</td>
      <td>1.0</td>
      <td>20/103</td>
      <td>22.80%</td>
      <td>AKAP13;HOPX;DPYD;PDE4D;PAG1;ACSL1;TNIK;SCD;RNF...</td>
    </tr>
  </tbody>
</table>
<p>4297 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Pathway Analysis - Late AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_19&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Disease Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_lateAT2_19.png" alt="gsea_lateAT2_19" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Disease Analysis Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_lateAT2_19.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_lateAT2_20&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Disease Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_lateAT2_20.png" alt="gsea_lateAT2_20" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Disease Analysis Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_lateAT2_20.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
</section>
<section id="early-at2-like-cells">
<h2>Early AT2 Like Cells<a class="headerlink" href="#early-at2-like-cells" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">numpy</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">np</span>

<span class="n">deg_results</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({</span>
    <span class="s1">&#39;names&#39;</span><span class="p">:</span> <span class="n">at2_h_lp_filt</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;rank_genes_groups&#39;</span><span class="p">][</span><span class="s1">&#39;names&#39;</span><span class="p">][</span><span class="s1">&#39;Early AT2 Like Cells&#39;</span><span class="p">],</span>    <span class="c1"># Gene names</span>
    <span class="s1">&#39;logfoldchanges&#39;</span><span class="p">:</span> <span class="n">at2_h_lp_filt</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;rank_genes_groups&#39;</span><span class="p">][</span><span class="s1">&#39;logfoldchanges&#39;</span><span class="p">][</span><span class="s1">&#39;Early AT2 Like Cells&#39;</span><span class="p">],</span>     <span class="c1"># Log fold changes</span>
    <span class="s1">&#39;pvals&#39;</span><span class="p">:</span> <span class="n">at2_h_lp_filt</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;rank_genes_groups&#39;</span><span class="p">][</span><span class="s1">&#39;pvals&#39;</span><span class="p">][</span><span class="s1">&#39;Early AT2 Like Cells&#39;</span><span class="p">],</span>          <span class="c1"># p-values</span>
    <span class="s1">&#39;pvals_adj&#39;</span><span class="p">:</span> <span class="n">at2_h_lp_filt</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;rank_genes_groups&#39;</span><span class="p">][</span><span class="s1">&#39;pvals_adj&#39;</span><span class="p">][</span><span class="s1">&#39;Early AT2 Like Cells&#39;</span><span class="p">]</span> <span class="c1"># Adjusted p-values (FDR corrected)</span>
<span class="p">})</span>

<span class="c1"># Step 1: Set thresholds for logFC and p-value (adjusted)</span>
<span class="n">logfc_threshold</span> <span class="o">=</span> <span class="mf">0.01</span>   <span class="c1"># Example: filter for logFC &gt; 1 (or &lt; -1 for downregulated)</span>
<span class="n">pval_adj_threshold</span> <span class="o">=</span> <span class="mf">0.05</span>  <span class="c1"># Example: adjusted p-value &lt; 0.05</span>

<span class="c1"># Step 2: Apply the filtering based on logFC and adjusted p-value</span>
<span class="n">filtered_genes</span> <span class="o">=</span> <span class="n">deg_results</span><span class="p">[</span>
    <span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">abs</span><span class="p">(</span><span class="n">deg_results</span><span class="p">[</span><span class="s1">&#39;logfoldchanges&#39;</span><span class="p">])</span> <span class="o">&gt;=</span> <span class="n">logfc_threshold</span><span class="p">)</span> <span class="o">&amp;</span>  <span class="c1"># Absolute logFC above threshold</span>
    <span class="p">(</span><span class="n">deg_results</span><span class="p">[</span><span class="s1">&#39;pvals_adj&#39;</span><span class="p">]</span> <span class="o">&lt;</span> <span class="n">pval_adj_threshold</span><span class="p">)</span>               <span class="c1"># Adjusted p-value below threshold</span>
<span class="p">]</span>

<span class="c1"># Step 3: Extract the list of filtered gene names</span>
<span class="n">filtered_gene_names</span> <span class="o">=</span> <span class="n">filtered_genes</span><span class="p">[</span><span class="s1">&#39;names&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
<span class="c1">#filtered_gene_names[:50]</span>

<span class="c1"># Run g:Profiler</span>
<span class="n">go_results</span> <span class="o">=</span> <span class="n">gp</span><span class="o">.</span><span class="n">profile</span><span class="p">(</span><span class="n">organism</span><span class="o">=</span><span class="s1">&#39;hsapiens&#39;</span><span class="p">,</span> <span class="n">query</span><span class="o">=</span><span class="n">filtered_gene_names</span><span class="p">[:</span><span class="mi">250</span><span class="p">])</span>

<span class="c1"># Define the sources you want to loop over</span>
<span class="n">sources</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;GO:BP&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:MF&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:CC&#39;</span><span class="p">,</span> <span class="s1">&#39;KEGG&#39;</span><span class="p">,</span> <span class="s1">&#39;REAC&#39;</span><span class="p">,</span> <span class="s1">&#39;HPA&#39;</span><span class="p">,</span> <span class="s1">&#39;WP&#39;</span><span class="p">]</span>

<span class="c1"># Create a stacked figure with one row per source</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">nrows</span><span class="o">=</span><span class="nb">len</span><span class="p">(</span><span class="n">sources</span><span class="p">),</span> <span class="n">ncols</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="mf">2.5</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">sources</span><span class="p">)))</span>
<span class="n">axes</span> <span class="o">=</span> <span class="n">axes</span><span class="o">.</span><span class="n">flatten</span><span class="p">()</span>

<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">source</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">sources</span><span class="p">):</span>
    <span class="c1"># Filter results for this source</span>
    <span class="n">specific_results</span> <span class="o">=</span> <span class="n">go_results</span><span class="p">[</span><span class="n">go_results</span><span class="p">[</span><span class="s1">&#39;source&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">source</span><span class="p">]</span>
    
    <span class="k">if</span> <span class="n">specific_results</span><span class="o">.</span><span class="n">empty</span><span class="p">:</span>
        <span class="k">continue</span>  <span class="c1"># skip if no results for this source</span>
    
    <span class="c1"># Select top 8–10 terms</span>
    <span class="n">go_results_top</span> <span class="o">=</span> <span class="p">(</span>
        <span class="n">specific_results</span><span class="p">[[</span><span class="s1">&#39;name&#39;</span><span class="p">,</span> <span class="s1">&#39;p_value&#39;</span><span class="p">]]</span>
        <span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;p_value&#39;</span><span class="p">)</span>
        <span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">8</span><span class="p">)</span>
    <span class="p">)</span>

    <span class="n">labels</span> <span class="o">=</span> <span class="n">go_results_top</span><span class="p">[</span><span class="s1">&#39;name&#39;</span><span class="p">]</span>
    <span class="n">values</span> <span class="o">=</span> <span class="o">-</span><span class="n">np</span><span class="o">.</span><span class="n">log10</span><span class="p">(</span><span class="n">go_results_top</span><span class="p">[</span><span class="s1">&#39;p_value&#39;</span><span class="p">])</span>

    <span class="n">ax</span> <span class="o">=</span> <span class="n">axes</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">barh</span><span class="p">(</span><span class="n">labels</span><span class="p">,</span> <span class="n">values</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s2">&quot;#1f77b4&quot;</span><span class="p">)</span>
    
    <span class="c1"># Aesthetics</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">source</span><span class="si">}</span><span class="s2"> enrichment&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span> <span class="n">loc</span><span class="o">=</span><span class="s2">&quot;left&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;-log10(P-value)&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">invert_yaxis</span><span class="p">()</span>  <span class="c1"># highest term on top</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="s2">&quot;y&quot;</span><span class="p">,</span> <span class="n">labelsize</span><span class="o">=</span><span class="mi">9</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="s2">&quot;x&quot;</span><span class="p">,</span> <span class="n">labelsize</span><span class="o">=</span><span class="mi">9</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s2">&quot;top&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="kc">False</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s2">&quot;right&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="kc">False</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">grid</span><span class="p">(</span><span class="kc">False</span><span class="p">)</span>
<span class="c1"># Adjust layout</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="c1"># Save if needed</span>
<span class="c1"># Save if needed</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_1&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GProfiler Analysis of top 250 differentially expressed genes from Early AT2 Like Cells.&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_earlyAT2_1.png" alt="gsea_earlyAT2_1" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GProfiler Analysis of top 250 differentially expressed genes from Early AT2 Like Cells.</em><br>
        <a href="_static/Figures/gsea_earlyAT2_1.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>

<span class="n">group</span> <span class="o">=</span> <span class="s2">&quot;Early AT2 Like Cells&quot;</span>  <span class="c1"># change for each group</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">get</span><span class="o">.</span><span class="n">rank_genes_groups_df</span><span class="p">(</span><span class="n">at2_h_lp_filt</span><span class="p">,</span> <span class="n">group</span><span class="o">=</span><span class="n">group</span><span class="p">)</span>

<span class="c1"># GSEAPY prerank needs index=gene, score=stat/logFC</span>
<span class="n">ranked_list</span> <span class="o">=</span> <span class="n">df</span><span class="p">[[</span><span class="s1">&#39;names&#39;</span><span class="p">,</span> <span class="s1">&#39;scores&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;names&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">squeeze</span><span class="p">()</span>

<span class="n">df_ranked_list_top</span> <span class="o">=</span> <span class="n">ranked_list</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">500</span><span class="p">]</span>

<span class="n">ranked_list_pos</span> <span class="o">=</span> <span class="n">ranked_list</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">250</span><span class="p">]</span>
<span class="n">ranked_list_neg</span> <span class="o">=</span> <span class="n">ranked_list</span><span class="p">[(</span><span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">)</span><span class="o">-</span><span class="mi">250</span><span class="p">):</span><span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">)]</span>

<span class="n">ranked_lists</span> <span class="o">=</span> <span class="p">[</span><span class="n">ranked_list_pos</span><span class="p">,</span><span class="n">ranked_list_neg</span><span class="p">]</span>

<span class="n">df_ranked_list</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">ranked_lists</span><span class="p">)</span>
<span class="c1"># Save to .rnk if needed</span>
<span class="c1">#ranked_list.to_csv(&quot;lateAT2.rnk&quot;, sep=&quot;\t&quot;)</span>
<span class="n">df_ranked_list</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>names
ADAMTSL1    41.869076
GPC5        27.370707
NCKAP5      22.340950
SYN3        15.959242
UTRN        15.616731
              ...    
SLC34A2    -56.681625
MKI67      -57.141468
LRRK2      -57.339821
TOP2A      -60.060707
MIR924HG   -65.477432
Name: scores, Length: 500, dtype: float32
</pre></div>
</div>
</div>
</div>
<section id="id1">
<h3>GSEA - Cell Type [Top 250 &amp; Down 250]<a class="headerlink" href="#id1" title="Link to this heading">#</a></h3>
<p>‘Azimuth_2023’,’CellMarker_2024’,’Descartes_Cell_Types_and_Tissue_2021’,’Human_Gene_Atlas’,’Human_Phenotype_Ontology’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Azimuth_2023&#39;</span><span class="p">,</span><span class="s1">&#39;CellMarker_2024&#39;</span><span class="p">,</span><span class="s1">&#39;Descartes_Cell_Types_and_Tissue_2021&#39;</span><span class="p">,</span><span class="s1">&#39;Human_Gene_Atlas&#39;</span><span class="p">,</span><span class="s1">&#39;Human_Phenotype_Ontology&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>2025-09-04 05:41:48,910 [WARNING] Duplicated values found in preranked stats: 19.20% of genes
The order of those genes will be arbitrary, which may produce unexpected results.
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>Descartes_Cell_Types_and_Tissue_2021__Hepatobl...</td>
      <td>0.994324</td>
      <td>1.696436</td>
      <td>0.017857</td>
      <td>0.062203</td>
      <td>0.094</td>
      <td>1/5</td>
      <td>0.40%</td>
      <td>GPC5</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>Descartes_Cell_Types_and_Tissue_2021__Schwann ...</td>
      <td>0.940653</td>
      <td>1.56588</td>
      <td>0.125</td>
      <td>0.111566</td>
      <td>0.3</td>
      <td>3/5</td>
      <td>6.00%</td>
      <td>CADM1;CDH1;FRMD5</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>Descartes_Cell_Types_and_Tissue_2021__Erythrob...</td>
      <td>-0.963717</td>
      <td>-1.512921</td>
      <td>0.004141</td>
      <td>1.0</td>
      <td>0.67</td>
      <td>1/5</td>
      <td>3.80%</td>
      <td>RRM2</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>Descartes_Cell_Types_and_Tissue_2021__Mesothel...</td>
      <td>-0.893005</td>
      <td>-1.479959</td>
      <td>0.016162</td>
      <td>1.0</td>
      <td>0.822</td>
      <td>2/7</td>
      <td>11.00%</td>
      <td>SLC34A2;ACOXL</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>Descartes_Cell_Types_and_Tissue_2021__Antigen ...</td>
      <td>-0.898566</td>
      <td>-1.448696</td>
      <td>0.019608</td>
      <td>1.0</td>
      <td>0.913</td>
      <td>4/6</td>
      <td>11.00%</td>
      <td>LRRK2;ABCC4;CTSH;ACOXL</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>193</th>
      <td>prerank</td>
      <td>CellMarker_2024__Schwalie Et al.Nature.P3 Adip...</td>
      <td>-0.435724</td>
      <td>-0.677322</td>
      <td>0.98081</td>
      <td>0.998115</td>
      <td>1.0</td>
      <td>1/5</td>
      <td>35.60%</td>
      <td>DOCK9</td>
    </tr>
    <tr>
      <th>194</th>
      <td>prerank</td>
      <td>CellMarker_2024__FGFR1HighNME5- Epithelial Cel...</td>
      <td>0.405889</td>
      <td>0.650185</td>
      <td>0.857143</td>
      <td>0.869888</td>
      <td>0.949</td>
      <td>2/5</td>
      <td>1.60%</td>
      <td>ADAMTSL1;ADGRL3</td>
    </tr>
    <tr>
      <th>195</th>
      <td>prerank</td>
      <td>Human_Gene_Atlas__CiliaryGanglion</td>
      <td>-0.402517</td>
      <td>-0.64531</td>
      <td>0.985462</td>
      <td>0.998306</td>
      <td>1.0</td>
      <td>5/6</td>
      <td>42.40%</td>
      <td>FMN1;CD59;DOCK9;MDM2;KIAA1958</td>
    </tr>
    <tr>
      <th>196</th>
      <td>prerank</td>
      <td>Human_Gene_Atlas__CerebellumPeduncles</td>
      <td>-0.400339</td>
      <td>-0.625686</td>
      <td>0.989451</td>
      <td>0.995726</td>
      <td>1.0</td>
      <td>2/5</td>
      <td>48.20%</td>
      <td>ETV1;AFF3</td>
    </tr>
    <tr>
      <th>197</th>
      <td>prerank</td>
      <td>CellMarker_2024__Schwalie Et al.Nature.G4 Adip...</td>
      <td>-0.326507</td>
      <td>-0.550427</td>
      <td>0.994919</td>
      <td>0.996768</td>
      <td>1.0</td>
      <td>5/8</td>
      <td>48.40%</td>
      <td>NPC2;JAK1;RAN;AFF3;DNAJA1</td>
    </tr>
  </tbody>
</table>
<p>198 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Cell Type - Early AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_2&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Cell Type Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">ValueError</span><span class="g g-Whitespace">                                </span>Traceback (most recent call last)
<span class="n">Cell</span> <span class="n">In</span><span class="p">[</span><span class="mi">75</span><span class="p">],</span> <span class="n">line</span> <span class="mi">2</span>
<span class="g g-Whitespace">      </span><span class="mi">1</span> <span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="ne">----&gt; </span><span class="mi">2</span> <span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">3</span>                  <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">4</span>                  <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Cell Type - Early AT2 Like Cells&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">5</span>                  <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">6</span>                  <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">7</span>                  <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
<span class="g g-Whitespace">      </span><span class="mi">8</span>                  <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">9</span> <span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="g g-Whitespace">     </span><span class="mi">10</span> <span class="n">plt</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s2">&quot;top5_gsea_plots.tiff&quot;</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">600</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s2">&quot;tight&quot;</span><span class="p">)</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:1225,</span> in <span class="ni">dotplot</span><span class="nt">(df, column, x, y, x_order, y_order, title, cutoff, top_term, size, ax, figsize, cmap, ofname, xticklabels_rot, yticklabels_rot, marker, show_ring, **kwargs)</span>
<span class="g g-Whitespace">   </span><span class="mi">1222</span>     <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span><span class="s2">&quot;group is deprecated; use x instead&quot;</span><span class="p">,</span> <span class="ne">DeprecationWarning</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1223</span>     <span class="k">return</span>
<span class="ne">-&gt; </span><span class="mi">1225</span> <span class="n">dot</span> <span class="o">=</span> <span class="n">DotPlot</span><span class="p">(</span>
<span class="g g-Whitespace">   </span><span class="mi">1226</span>     <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1227</span>     <span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1228</span>     <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1229</span>     <span class="n">x_order</span><span class="o">=</span><span class="n">x_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1230</span>     <span class="n">y_order</span><span class="o">=</span><span class="n">y_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1231</span>     <span class="n">hue</span><span class="o">=</span><span class="n">column</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1232</span>     <span class="n">title</span><span class="o">=</span><span class="n">title</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1233</span>     <span class="n">thresh</span><span class="o">=</span><span class="n">cutoff</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1234</span>     <span class="n">n_terms</span><span class="o">=</span><span class="nb">int</span><span class="p">(</span><span class="n">top_term</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1235</span>     <span class="n">dot_scale</span><span class="o">=</span><span class="n">size</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1236</span>     <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1237</span>     <span class="n">figsize</span><span class="o">=</span><span class="n">figsize</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1238</span>     <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1239</span>     <span class="n">ofname</span><span class="o">=</span><span class="n">ofname</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1240</span>     <span class="n">marker</span><span class="o">=</span><span class="n">marker</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1241</span> <span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1242</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dot</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">outer_ring</span><span class="o">=</span><span class="n">show_ring</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1244</span> <span class="k">if</span> <span class="n">xticklabels_rot</span><span class="p">:</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:692,</span> in <span class="ni">DotPlot.__init__</span><span class="nt">(self, df, x, y, hue, dot_scale, x_order, y_order, thresh, n_terms, title, ax, figsize, cmap, ofname, marker, **kwargs)</span>
<span class="g g-Whitespace">    </span><span class="mi">690</span> <span class="bp">self</span><span class="o">.</span><span class="n">n_terms</span> <span class="o">=</span> <span class="n">n_terms</span>
<span class="g g-Whitespace">    </span><span class="mi">691</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span> <span class="o">=</span> <span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">692</span> <span class="bp">self</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">693</span> <span class="n">plt</span><span class="o">.</span><span class="n">rcParams</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="s2">&quot;pdf.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">,</span> <span class="s2">&quot;ps.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">})</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:717,</span> in <span class="ni">DotPlot.process</span><span class="nt">(self, df)</span>
<span class="g g-Whitespace">    </span><span class="mi">715</span> <span class="k">if</span> <span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">716</span>     <span class="n">msg</span> <span class="o">=</span> <span class="s2">&quot;Warning: No enrich terms when cutoff = </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">717</span>     <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="n">msg</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">718</span> <span class="bp">self</span><span class="o">.</span><span class="n">cbar_title</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">colname</span>
<span class="g g-Whitespace">    </span><span class="mi">719</span> <span class="c1"># clip GSEA lower bounds</span>
<span class="g g-Whitespace">    </span><span class="mi">720</span> <span class="c1"># if self.colname in [&quot;NOM p-val&quot;, &quot;FDR q-val&quot;]:</span>
<span class="g g-Whitespace">    </span><span class="mi">721</span> <span class="c1">#     df[self.colname].clip(1e-5, 1.0, inplace=True)</span>
<span class="g g-Whitespace">    </span><span class="mi">722</span> <span class="c1"># sorting the dataframe for better visualization</span>

<span class="ne">ValueError</span>: Warning: No enrich terms when cutoff = 0.05
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_3&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Cell Type Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_earlyAT2_3.png" alt="gsea_earlyAT2_3" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Cell Type Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_earlyAT2_3.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id2">
<h3>GSEA - Cell Type [Top 500]<a class="headerlink" href="#id2" title="Link to this heading">#</a></h3>
<p>‘Azimuth_2023’,’CellMarker_2024’,’Descartes_Cell_Types_and_Tissue_2021’,’Human_Gene_Atlas’,’Human_Phenotype_Ontology’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list_top</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Azimuth_2023&#39;</span><span class="p">,</span><span class="s1">&#39;CellMarker_2024&#39;</span><span class="p">,</span><span class="s1">&#39;Descartes_Cell_Types_and_Tissue_2021&#39;</span><span class="p">,</span><span class="s1">&#39;Human_Gene_Atlas&#39;</span><span class="p">,</span><span class="s1">&#39;Human_Phenotype_Ontology&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>2025-09-04 05:42:54,565 [WARNING] Duplicated values found in preranked stats: 55.40% of genes
The order of those genes will be arbitrary, which may produce unexpected results.
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>Descartes_Cell_Types_and_Tissue_2021__ENS neur...</td>
      <td>0.984618</td>
      <td>1.574646</td>
      <td>0.067776</td>
      <td>1.0</td>
      <td>0.747</td>
      <td>1/6</td>
      <td>0.80%</td>
      <td>SYN3</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>CellMarker_2024__Neuroendocrine Cell Trachea M...</td>
      <td>0.962302</td>
      <td>1.541727</td>
      <td>0.122249</td>
      <td>1.0</td>
      <td>0.963</td>
      <td>2/5</td>
      <td>3.40%</td>
      <td>LRBA;TTC3</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>Human_Gene_Atlas__BronchialEpithelialCells</td>
      <td>0.957827</td>
      <td>1.540506</td>
      <td>0.16545</td>
      <td>1.0</td>
      <td>0.968</td>
      <td>2/5</td>
      <td>4.00%</td>
      <td>MYOF;ANXA2</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>CellMarker_2024__Sertoli Cell Fetal Gonad Human</td>
      <td>0.954398</td>
      <td>1.536647</td>
      <td>0.151554</td>
      <td>1.0</td>
      <td>0.978</td>
      <td>3/5</td>
      <td>5.00%</td>
      <td>DSP;CADM1;SMARCA5</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>Human_Gene_Atlas__pineal day</td>
      <td>0.948545</td>
      <td>1.528566</td>
      <td>0.178703</td>
      <td>1.0</td>
      <td>0.987</td>
      <td>3/6</td>
      <td>5.40%</td>
      <td>GPC5;CADM1;UNC13B</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>81</th>
      <td>prerank</td>
      <td>Descartes_Cell_Types_and_Tissue_2021__Microgli...</td>
      <td>0.337767</td>
      <td>0.476317</td>
      <td>0.834737</td>
      <td>0.938494</td>
      <td>1.0</td>
      <td>6/14</td>
      <td>38.40%</td>
      <td>KCNJ5;RASAL3;LAPTM5;MS4A6A;C1QB;ITGAM</td>
    </tr>
    <tr>
      <th>82</th>
      <td>prerank</td>
      <td>Descartes_Cell_Types_and_Tissue_2021__Myeloid ...</td>
      <td>0.29899</td>
      <td>0.467405</td>
      <td>0.930087</td>
      <td>0.923563</td>
      <td>1.0</td>
      <td>5/5</td>
      <td>70.40%</td>
      <td>MS4A6A;C1QB;HLA-DOA;C1QC;MS4A14</td>
    </tr>
    <tr>
      <th>83</th>
      <td>prerank</td>
      <td>Descartes_Cell_Types_and_Tissue_2021__Myeloid ...</td>
      <td>0.29899</td>
      <td>0.467405</td>
      <td>0.930087</td>
      <td>0.923563</td>
      <td>1.0</td>
      <td>5/5</td>
      <td>70.40%</td>
      <td>MS4A6A;C1QB;HLA-DOA;C1QC;MS4A14</td>
    </tr>
    <tr>
      <th>84</th>
      <td>prerank</td>
      <td>CellMarker_2024__Microglial Cell Embryonic Pre...</td>
      <td>0.284175</td>
      <td>0.443728</td>
      <td>0.934625</td>
      <td>0.932956</td>
      <td>1.0</td>
      <td>3/6</td>
      <td>38.40%</td>
      <td>LAPTM5;C1QB;ITGAM</td>
    </tr>
    <tr>
      <th>85</th>
      <td>prerank</td>
      <td>Descartes_Cell_Types_and_Tissue_2021__Lymphoid...</td>
      <td>0.292554</td>
      <td>0.43775</td>
      <td>0.865957</td>
      <td>0.92628</td>
      <td>1.0</td>
      <td>4/11</td>
      <td>36.40%</td>
      <td>CD52;GPA33;IL21R;RUNX3</td>
    </tr>
  </tbody>
</table>
<p>86 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Cell Type - Early AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_4&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Cell Type Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">ValueError</span><span class="g g-Whitespace">                                </span>Traceback (most recent call last)
<span class="n">Cell</span> <span class="n">In</span><span class="p">[</span><span class="mi">78</span><span class="p">],</span> <span class="n">line</span> <span class="mi">2</span>
<span class="g g-Whitespace">      </span><span class="mi">1</span> <span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="ne">----&gt; </span><span class="mi">2</span> <span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">3</span>                  <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">4</span>                  <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Cell Type - Early AT2 Like Cells&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">5</span>                  <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">6</span>                  <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">7</span>                  <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
<span class="g g-Whitespace">      </span><span class="mi">8</span>                  <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">9</span> <span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="g g-Whitespace">     </span><span class="mi">10</span> <span class="n">plt</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s2">&quot;top5_gsea_plots.tiff&quot;</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">600</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s2">&quot;tight&quot;</span><span class="p">)</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:1225,</span> in <span class="ni">dotplot</span><span class="nt">(df, column, x, y, x_order, y_order, title, cutoff, top_term, size, ax, figsize, cmap, ofname, xticklabels_rot, yticklabels_rot, marker, show_ring, **kwargs)</span>
<span class="g g-Whitespace">   </span><span class="mi">1222</span>     <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span><span class="s2">&quot;group is deprecated; use x instead&quot;</span><span class="p">,</span> <span class="ne">DeprecationWarning</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1223</span>     <span class="k">return</span>
<span class="ne">-&gt; </span><span class="mi">1225</span> <span class="n">dot</span> <span class="o">=</span> <span class="n">DotPlot</span><span class="p">(</span>
<span class="g g-Whitespace">   </span><span class="mi">1226</span>     <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1227</span>     <span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1228</span>     <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1229</span>     <span class="n">x_order</span><span class="o">=</span><span class="n">x_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1230</span>     <span class="n">y_order</span><span class="o">=</span><span class="n">y_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1231</span>     <span class="n">hue</span><span class="o">=</span><span class="n">column</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1232</span>     <span class="n">title</span><span class="o">=</span><span class="n">title</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1233</span>     <span class="n">thresh</span><span class="o">=</span><span class="n">cutoff</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1234</span>     <span class="n">n_terms</span><span class="o">=</span><span class="nb">int</span><span class="p">(</span><span class="n">top_term</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1235</span>     <span class="n">dot_scale</span><span class="o">=</span><span class="n">size</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1236</span>     <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1237</span>     <span class="n">figsize</span><span class="o">=</span><span class="n">figsize</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1238</span>     <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1239</span>     <span class="n">ofname</span><span class="o">=</span><span class="n">ofname</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1240</span>     <span class="n">marker</span><span class="o">=</span><span class="n">marker</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1241</span> <span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1242</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dot</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">outer_ring</span><span class="o">=</span><span class="n">show_ring</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1244</span> <span class="k">if</span> <span class="n">xticklabels_rot</span><span class="p">:</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:692,</span> in <span class="ni">DotPlot.__init__</span><span class="nt">(self, df, x, y, hue, dot_scale, x_order, y_order, thresh, n_terms, title, ax, figsize, cmap, ofname, marker, **kwargs)</span>
<span class="g g-Whitespace">    </span><span class="mi">690</span> <span class="bp">self</span><span class="o">.</span><span class="n">n_terms</span> <span class="o">=</span> <span class="n">n_terms</span>
<span class="g g-Whitespace">    </span><span class="mi">691</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span> <span class="o">=</span> <span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">692</span> <span class="bp">self</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">693</span> <span class="n">plt</span><span class="o">.</span><span class="n">rcParams</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="s2">&quot;pdf.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">,</span> <span class="s2">&quot;ps.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">})</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:717,</span> in <span class="ni">DotPlot.process</span><span class="nt">(self, df)</span>
<span class="g g-Whitespace">    </span><span class="mi">715</span> <span class="k">if</span> <span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">716</span>     <span class="n">msg</span> <span class="o">=</span> <span class="s2">&quot;Warning: No enrich terms when cutoff = </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">717</span>     <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="n">msg</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">718</span> <span class="bp">self</span><span class="o">.</span><span class="n">cbar_title</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">colname</span>
<span class="g g-Whitespace">    </span><span class="mi">719</span> <span class="c1"># clip GSEA lower bounds</span>
<span class="g g-Whitespace">    </span><span class="mi">720</span> <span class="c1"># if self.colname in [&quot;NOM p-val&quot;, &quot;FDR q-val&quot;]:</span>
<span class="g g-Whitespace">    </span><span class="mi">721</span> <span class="c1">#     df[self.colname].clip(1e-5, 1.0, inplace=True)</span>
<span class="g g-Whitespace">    </span><span class="mi">722</span> <span class="c1"># sorting the dataframe for better visualization</span>

<span class="ne">ValueError</span>: Warning: No enrich terms when cutoff = 0.05
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_5&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Cell Type Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_earlyAT2_5.png" alt="gsea_earlyAT2_5" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Cell Type Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_earlyAT2_5.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id3">
<h3>GSEA - Gene Ontology [Top 250 &amp; Down 250]<a class="headerlink" href="#id3" title="Link to this heading">#</a></h3>
<p>‘GO_Biological_Process_2025’,’GO_Cellular_Component_2025’,’GO_Molecular_Function_2025’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;GO_Biological_Process_2025&#39;</span><span class="p">,</span><span class="s1">&#39;GO_Cellular_Component_2025&#39;</span><span class="p">,</span><span class="s1">&#39;GO_Molecular_Function_2025&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>2025-09-04 05:44:15,746 [WARNING] Duplicated values found in preranked stats: 19.20% of genes
The order of those genes will be arbitrary, which may produce unexpected results.
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>GO_Cellular_Component_2025__Collagen-Containin...</td>
      <td>-0.848739</td>
      <td>-1.461537</td>
      <td>0.009082</td>
      <td>1.0</td>
      <td>0.916</td>
      <td>2/9</td>
      <td>10.80%</td>
      <td>ADAMTS9;CTSH</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>GO_Molecular_Function_2025__Cytokine Activity ...</td>
      <td>-0.887471</td>
      <td>-1.44867</td>
      <td>0.021739</td>
      <td>1.0</td>
      <td>0.938</td>
      <td>1/6</td>
      <td>11.20%</td>
      <td>HMGB1</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Positive Regulatio...</td>
      <td>-0.821385</td>
      <td>-1.338511</td>
      <td>0.053886</td>
      <td>1.0</td>
      <td>0.999</td>
      <td>4/6</td>
      <td>18.60%</td>
      <td>LAMP3;DNMT1;CTSH;ANK3</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Dephosphorylation ...</td>
      <td>-0.813166</td>
      <td>-1.30508</td>
      <td>0.088083</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>1/6</td>
      <td>10.00%</td>
      <td>PTPN13</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Protein Dephosphor...</td>
      <td>-0.813166</td>
      <td>-1.30508</td>
      <td>0.088083</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>1/6</td>
      <td>10.00%</td>
      <td>PTPN13</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>271</th>
      <td>prerank</td>
      <td>GO_Cellular_Component_2025__Cortical Cytoskele...</td>
      <td>-0.455677</td>
      <td>-0.718752</td>
      <td>0.963041</td>
      <td>0.984414</td>
      <td>1.0</td>
      <td>2/5</td>
      <td>46.80%</td>
      <td>MLPH;HIP1</td>
    </tr>
    <tr>
      <th>272</th>
      <td>prerank</td>
      <td>GO_Cellular_Component_2025__Cortical Actin Cyt...</td>
      <td>-0.455677</td>
      <td>-0.718752</td>
      <td>0.963041</td>
      <td>0.984414</td>
      <td>1.0</td>
      <td>2/5</td>
      <td>46.80%</td>
      <td>MLPH;HIP1</td>
    </tr>
    <tr>
      <th>273</th>
      <td>prerank</td>
      <td>GO_Molecular_Function_2025__Calcium Ion Bindin...</td>
      <td>-0.398359</td>
      <td>-0.696541</td>
      <td>0.970913</td>
      <td>0.986506</td>
      <td>1.0</td>
      <td>4/10</td>
      <td>46.00%</td>
      <td>ITPR3;ITPR2;TKT;ATP2C1</td>
    </tr>
    <tr>
      <th>274</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Generation of Neur...</td>
      <td>-0.415637</td>
      <td>-0.687485</td>
      <td>0.974438</td>
      <td>0.984883</td>
      <td>1.0</td>
      <td>1/7</td>
      <td>35.20%</td>
      <td>AUTS2</td>
    </tr>
    <tr>
      <th>275</th>
      <td>prerank</td>
      <td>GO_Molecular_Function_2025__Ubiquitin-Protein ...</td>
      <td>-0.396564</td>
      <td>-0.65378</td>
      <td>0.984741</td>
      <td>0.987077</td>
      <td>1.0</td>
      <td>2/7</td>
      <td>45.60%</td>
      <td>MDM2;FBXL5</td>
    </tr>
  </tbody>
</table>
<p>276 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Gene Ontology - Early AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">6</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_6&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Gene Ontology Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">ValueError</span><span class="g g-Whitespace">                                </span>Traceback (most recent call last)
<span class="n">Cell</span> <span class="n">In</span><span class="p">[</span><span class="mi">100</span><span class="p">],</span> <span class="n">line</span> <span class="mi">2</span>
<span class="g g-Whitespace">      </span><span class="mi">1</span> <span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="ne">----&gt; </span><span class="mi">2</span> <span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">3</span>                  <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">4</span>                  <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Gene Ontology - Early AT2 Like Cells&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">5</span>                  <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">6</span>                  <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">7</span>                  <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">6</span><span class="p">),</span> 
<span class="g g-Whitespace">      </span><span class="mi">8</span>                  <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">9</span> <span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="g g-Whitespace">     </span><span class="mi">11</span> <span class="n">sc_plot_and_embed</span><span class="p">(</span>
<span class="g g-Whitespace">     </span><span class="mi">12</span>     <span class="n">ax_dot</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">13</span>     <span class="s2">&quot;gsea_earlyAT2_6&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">14</span>     <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Gene Ontology Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">15</span>     <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="g g-Whitespace">     </span><span class="mi">16</span> <span class="p">)</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:1225,</span> in <span class="ni">dotplot</span><span class="nt">(df, column, x, y, x_order, y_order, title, cutoff, top_term, size, ax, figsize, cmap, ofname, xticklabels_rot, yticklabels_rot, marker, show_ring, **kwargs)</span>
<span class="g g-Whitespace">   </span><span class="mi">1222</span>     <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span><span class="s2">&quot;group is deprecated; use x instead&quot;</span><span class="p">,</span> <span class="ne">DeprecationWarning</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1223</span>     <span class="k">return</span>
<span class="ne">-&gt; </span><span class="mi">1225</span> <span class="n">dot</span> <span class="o">=</span> <span class="n">DotPlot</span><span class="p">(</span>
<span class="g g-Whitespace">   </span><span class="mi">1226</span>     <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1227</span>     <span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1228</span>     <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1229</span>     <span class="n">x_order</span><span class="o">=</span><span class="n">x_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1230</span>     <span class="n">y_order</span><span class="o">=</span><span class="n">y_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1231</span>     <span class="n">hue</span><span class="o">=</span><span class="n">column</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1232</span>     <span class="n">title</span><span class="o">=</span><span class="n">title</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1233</span>     <span class="n">thresh</span><span class="o">=</span><span class="n">cutoff</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1234</span>     <span class="n">n_terms</span><span class="o">=</span><span class="nb">int</span><span class="p">(</span><span class="n">top_term</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1235</span>     <span class="n">dot_scale</span><span class="o">=</span><span class="n">size</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1236</span>     <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1237</span>     <span class="n">figsize</span><span class="o">=</span><span class="n">figsize</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1238</span>     <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1239</span>     <span class="n">ofname</span><span class="o">=</span><span class="n">ofname</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1240</span>     <span class="n">marker</span><span class="o">=</span><span class="n">marker</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1241</span> <span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1242</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dot</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">outer_ring</span><span class="o">=</span><span class="n">show_ring</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1244</span> <span class="k">if</span> <span class="n">xticklabels_rot</span><span class="p">:</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:692,</span> in <span class="ni">DotPlot.__init__</span><span class="nt">(self, df, x, y, hue, dot_scale, x_order, y_order, thresh, n_terms, title, ax, figsize, cmap, ofname, marker, **kwargs)</span>
<span class="g g-Whitespace">    </span><span class="mi">690</span> <span class="bp">self</span><span class="o">.</span><span class="n">n_terms</span> <span class="o">=</span> <span class="n">n_terms</span>
<span class="g g-Whitespace">    </span><span class="mi">691</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span> <span class="o">=</span> <span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">692</span> <span class="bp">self</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">693</span> <span class="n">plt</span><span class="o">.</span><span class="n">rcParams</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="s2">&quot;pdf.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">,</span> <span class="s2">&quot;ps.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">})</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:717,</span> in <span class="ni">DotPlot.process</span><span class="nt">(self, df)</span>
<span class="g g-Whitespace">    </span><span class="mi">715</span> <span class="k">if</span> <span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">716</span>     <span class="n">msg</span> <span class="o">=</span> <span class="s2">&quot;Warning: No enrich terms when cutoff = </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">717</span>     <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="n">msg</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">718</span> <span class="bp">self</span><span class="o">.</span><span class="n">cbar_title</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">colname</span>
<span class="g g-Whitespace">    </span><span class="mi">719</span> <span class="c1"># clip GSEA lower bounds</span>
<span class="g g-Whitespace">    </span><span class="mi">720</span> <span class="c1"># if self.colname in [&quot;NOM p-val&quot;, &quot;FDR q-val&quot;]:</span>
<span class="g g-Whitespace">    </span><span class="mi">721</span> <span class="c1">#     df[self.colname].clip(1e-5, 1.0, inplace=True)</span>
<span class="g g-Whitespace">    </span><span class="mi">722</span> <span class="c1"># sorting the dataframe for better visualization</span>

<span class="ne">ValueError</span>: Warning: No enrich terms when cutoff = 0.05
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_7&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Gene Ontology Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_earlyAT2_7.png" alt="gsea_earlyAT2_7" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Gene Ontology Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_earlyAT2_7.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id4">
<h3>GSEA - Gene Ontology [Top 500]<a class="headerlink" href="#id4" title="Link to this heading">#</a></h3>
<p>‘GO_Biological_Process_2025’,’GO_Cellular_Component_2025’,’GO_Molecular_Function_2025’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list_top</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;GO_Biological_Process_2025&#39;</span><span class="p">,</span><span class="s1">&#39;GO_Cellular_Component_2025&#39;</span><span class="p">,</span><span class="s1">&#39;GO_Molecular_Function_2025&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>2025-09-04 05:45:09,040 [WARNING] Duplicated values found in preranked stats: 55.40% of genes
The order of those genes will be arbitrary, which may produce unexpected results.
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Generation of Neur...</td>
      <td>0.97421</td>
      <td>1.578393</td>
      <td>0.102066</td>
      <td>1.0</td>
      <td>0.682</td>
      <td>1/6</td>
      <td>1.60%</td>
      <td>ADGRL3</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Cell-Cell Adhesion...</td>
      <td>0.98079</td>
      <td>1.561147</td>
      <td>0.085603</td>
      <td>1.0</td>
      <td>0.842</td>
      <td>2/5</td>
      <td>2.00%</td>
      <td>ADGRL3;CADM1</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Protein Localizati...</td>
      <td>0.95398</td>
      <td>1.55915</td>
      <td>0.161412</td>
      <td>1.0</td>
      <td>0.858</td>
      <td>2/5</td>
      <td>4.20%</td>
      <td>LRBA;PTPRK</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Positive Regulatio...</td>
      <td>0.970936</td>
      <td>1.559125</td>
      <td>0.112923</td>
      <td>0.828399</td>
      <td>0.859</td>
      <td>1/5</td>
      <td>2.00%</td>
      <td>CADM1</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>GO_Molecular_Function_2025__Ubiquitin-Protein ...</td>
      <td>0.966203</td>
      <td>1.558098</td>
      <td>0.106329</td>
      <td>0.676786</td>
      <td>0.863</td>
      <td>2/5</td>
      <td>3.40%</td>
      <td>LMO7;TTC3</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>71</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Anterior/Posterior...</td>
      <td>-0.305051</td>
      <td>-0.763555</td>
      <td>0.812207</td>
      <td>0.799413</td>
      <td>1.0</td>
      <td>5/5</td>
      <td>70.00%</td>
      <td>HOXB3;HELT;NEUROG1;CDX2;HOXB6</td>
    </tr>
    <tr>
      <th>72</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Negative Regulatio...</td>
      <td>0.458706</td>
      <td>0.740384</td>
      <td>0.65</td>
      <td>0.678311</td>
      <td>1.0</td>
      <td>4/5</td>
      <td>47.80%</td>
      <td>WNT11;DLX2;APOA4;NEUROG1</td>
    </tr>
    <tr>
      <th>73</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Positive Regulatio...</td>
      <td>0.443169</td>
      <td>0.686128</td>
      <td>0.667442</td>
      <td>0.712468</td>
      <td>1.0</td>
      <td>4/7</td>
      <td>33.00%</td>
      <td>FGF5;LAPTM5;FASLG;USP17L2</td>
    </tr>
    <tr>
      <th>74</th>
      <td>prerank</td>
      <td>GO_Molecular_Function_2025__Cis-Regulatory Reg...</td>
      <td>0.486995</td>
      <td>0.683434</td>
      <td>0.691239</td>
      <td>0.704971</td>
      <td>1.0</td>
      <td>11/15</td>
      <td>39.80%</td>
      <td>MECOM;EMX1;ZIC2;FOXD1;EGR4;HOXB6;ZNF280A;RUNX3...</td>
    </tr>
    <tr>
      <th>75</th>
      <td>prerank</td>
      <td>GO_Molecular_Function_2025__RNA Polymerase II ...</td>
      <td>0.424899</td>
      <td>0.586393</td>
      <td>0.754202</td>
      <td>0.783737</td>
      <td>1.0</td>
      <td>11/17</td>
      <td>39.80%</td>
      <td>MECOM;EMX1;ZIC2;FOXD1;EGR4;HOXB6;ZNF280A;RUNX3...</td>
    </tr>
  </tbody>
</table>
<p>76 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Gene Ontology - Early AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">6</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_8&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Gene Ontology Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">ValueError</span><span class="g g-Whitespace">                                </span>Traceback (most recent call last)
<span class="n">Cell</span> <span class="n">In</span><span class="p">[</span><span class="mi">103</span><span class="p">],</span> <span class="n">line</span> <span class="mi">2</span>
<span class="g g-Whitespace">      </span><span class="mi">1</span> <span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="ne">----&gt; </span><span class="mi">2</span> <span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">3</span>                  <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">4</span>                  <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Gene Ontology - Early AT2 Like Cells&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">5</span>                  <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">6</span>                  <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">7</span>                  <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">6</span><span class="p">),</span> 
<span class="g g-Whitespace">      </span><span class="mi">8</span>                  <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">9</span> <span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="g g-Whitespace">     </span><span class="mi">11</span> <span class="n">sc_plot_and_embed</span><span class="p">(</span>
<span class="g g-Whitespace">     </span><span class="mi">12</span>     <span class="n">ax_dot</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">13</span>     <span class="s2">&quot;gsea_earlyAT2_8&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">14</span>     <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Gene Ontology Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">15</span>     <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="g g-Whitespace">     </span><span class="mi">16</span> <span class="p">)</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:1225,</span> in <span class="ni">dotplot</span><span class="nt">(df, column, x, y, x_order, y_order, title, cutoff, top_term, size, ax, figsize, cmap, ofname, xticklabels_rot, yticklabels_rot, marker, show_ring, **kwargs)</span>
<span class="g g-Whitespace">   </span><span class="mi">1222</span>     <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span><span class="s2">&quot;group is deprecated; use x instead&quot;</span><span class="p">,</span> <span class="ne">DeprecationWarning</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1223</span>     <span class="k">return</span>
<span class="ne">-&gt; </span><span class="mi">1225</span> <span class="n">dot</span> <span class="o">=</span> <span class="n">DotPlot</span><span class="p">(</span>
<span class="g g-Whitespace">   </span><span class="mi">1226</span>     <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1227</span>     <span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1228</span>     <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1229</span>     <span class="n">x_order</span><span class="o">=</span><span class="n">x_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1230</span>     <span class="n">y_order</span><span class="o">=</span><span class="n">y_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1231</span>     <span class="n">hue</span><span class="o">=</span><span class="n">column</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1232</span>     <span class="n">title</span><span class="o">=</span><span class="n">title</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1233</span>     <span class="n">thresh</span><span class="o">=</span><span class="n">cutoff</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1234</span>     <span class="n">n_terms</span><span class="o">=</span><span class="nb">int</span><span class="p">(</span><span class="n">top_term</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1235</span>     <span class="n">dot_scale</span><span class="o">=</span><span class="n">size</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1236</span>     <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1237</span>     <span class="n">figsize</span><span class="o">=</span><span class="n">figsize</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1238</span>     <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1239</span>     <span class="n">ofname</span><span class="o">=</span><span class="n">ofname</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1240</span>     <span class="n">marker</span><span class="o">=</span><span class="n">marker</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1241</span> <span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1242</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dot</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">outer_ring</span><span class="o">=</span><span class="n">show_ring</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1244</span> <span class="k">if</span> <span class="n">xticklabels_rot</span><span class="p">:</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:692,</span> in <span class="ni">DotPlot.__init__</span><span class="nt">(self, df, x, y, hue, dot_scale, x_order, y_order, thresh, n_terms, title, ax, figsize, cmap, ofname, marker, **kwargs)</span>
<span class="g g-Whitespace">    </span><span class="mi">690</span> <span class="bp">self</span><span class="o">.</span><span class="n">n_terms</span> <span class="o">=</span> <span class="n">n_terms</span>
<span class="g g-Whitespace">    </span><span class="mi">691</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span> <span class="o">=</span> <span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">692</span> <span class="bp">self</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">693</span> <span class="n">plt</span><span class="o">.</span><span class="n">rcParams</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="s2">&quot;pdf.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">,</span> <span class="s2">&quot;ps.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">})</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:717,</span> in <span class="ni">DotPlot.process</span><span class="nt">(self, df)</span>
<span class="g g-Whitespace">    </span><span class="mi">715</span> <span class="k">if</span> <span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">716</span>     <span class="n">msg</span> <span class="o">=</span> <span class="s2">&quot;Warning: No enrich terms when cutoff = </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">717</span>     <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="n">msg</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">718</span> <span class="bp">self</span><span class="o">.</span><span class="n">cbar_title</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">colname</span>
<span class="g g-Whitespace">    </span><span class="mi">719</span> <span class="c1"># clip GSEA lower bounds</span>
<span class="g g-Whitespace">    </span><span class="mi">720</span> <span class="c1"># if self.colname in [&quot;NOM p-val&quot;, &quot;FDR q-val&quot;]:</span>
<span class="g g-Whitespace">    </span><span class="mi">721</span> <span class="c1">#     df[self.colname].clip(1e-5, 1.0, inplace=True)</span>
<span class="g g-Whitespace">    </span><span class="mi">722</span> <span class="c1"># sorting the dataframe for better visualization</span>

<span class="ne">ValueError</span>: Warning: No enrich terms when cutoff = 0.05
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_9&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Gene Ontology Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_earlyAT2_9.png" alt="gsea_earlyAT2_9" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Gene Ontology Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_earlyAT2_9.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id5">
<h3>GSEA - Pathway Analysis [Top 250 &amp; Down 250]<a class="headerlink" href="#id5" title="Link to this heading">#</a></h3>
<p>‘Reactome_Pathways_2024’,’KEGG_2021_Human’,’WikiPathways_2024_Human’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Reactome_Pathways_2024&#39;</span><span class="p">,</span><span class="s1">&#39;KEGG_2021_Human&#39;</span><span class="p">,</span><span class="s1">&#39;WikiPathways_2024_Human&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>2025-09-04 05:49:07,308 [WARNING] Duplicated values found in preranked stats: 19.20% of genes
The order of those genes will be arbitrary, which may produce unexpected results.
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>WikiPathways_2024_Human__Allograft Rejection W...</td>
      <td>-0.946801</td>
      <td>-1.525155</td>
      <td>0.009365</td>
      <td>1.0</td>
      <td>0.623</td>
      <td>2/6</td>
      <td>4.00%</td>
      <td>LRRK2;TUBA1B</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>WikiPathways_2024_Human__MAPK Signaling WP382</td>
      <td>-0.855674</td>
      <td>-1.44022</td>
      <td>0.025536</td>
      <td>1.0</td>
      <td>0.939</td>
      <td>2/8</td>
      <td>12.80%</td>
      <td>LRRK2;HSPA8</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>KEGG_2021_Human__Phagosome</td>
      <td>-0.906969</td>
      <td>-1.426734</td>
      <td>0.034043</td>
      <td>1.0</td>
      <td>0.962</td>
      <td>4/5</td>
      <td>10.20%</td>
      <td>TUBA1B;TUBB;SFTPA1;TUBB4B</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>WikiPathways_2024_Human__Sudden Infant Death S...</td>
      <td>0.868511</td>
      <td>1.410636</td>
      <td>0.297872</td>
      <td>0.637326</td>
      <td>0.623</td>
      <td>1/5</td>
      <td>5.60%</td>
      <td>SPTBN1</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>KEGG_2021_Human__MAPK signaling pathway</td>
      <td>-0.82625</td>
      <td>-1.384848</td>
      <td>0.046012</td>
      <td>1.0</td>
      <td>0.994</td>
      <td>1/8</td>
      <td>12.80%</td>
      <td>HSPA8</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>299</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Metabolism of Vitamins...</td>
      <td>-0.424146</td>
      <td>-0.715908</td>
      <td>0.96945</td>
      <td>0.987513</td>
      <td>1.0</td>
      <td>5/8</td>
      <td>46.60%</td>
      <td>GPC6;FASN;SDC4;MTHFD1;PARP14</td>
    </tr>
    <tr>
      <th>300</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Signaling by ALK in Ca...</td>
      <td>-0.444994</td>
      <td>-0.699197</td>
      <td>0.968487</td>
      <td>0.985057</td>
      <td>1.0</td>
      <td>3/5</td>
      <td>46.80%</td>
      <td>DNMT1;MDM2;HIP1</td>
    </tr>
    <tr>
      <th>301</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Signaling by ALK Fusio...</td>
      <td>-0.444994</td>
      <td>-0.699197</td>
      <td>0.968487</td>
      <td>0.985057</td>
      <td>1.0</td>
      <td>3/5</td>
      <td>46.80%</td>
      <td>DNMT1;MDM2;HIP1</td>
    </tr>
    <tr>
      <th>302</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Antigen Processing Ubi...</td>
      <td>-0.442451</td>
      <td>-0.695587</td>
      <td>0.968421</td>
      <td>0.98263</td>
      <td>1.0</td>
      <td>4/5</td>
      <td>50.20%</td>
      <td>ZBTB16;FBXL5;TPP2;PSMD1</td>
    </tr>
    <tr>
      <th>303</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Diseases of Glycosylation</td>
      <td>-0.427549</td>
      <td>-0.671609</td>
      <td>0.98</td>
      <td>0.984078</td>
      <td>1.0</td>
      <td>2/5</td>
      <td>6.00%</td>
      <td>GPC6;ADAMTS9</td>
    </tr>
  </tbody>
</table>
<p>304 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Pathway Analysis - Early AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_10&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Pathway Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">ValueError</span><span class="g g-Whitespace">                                </span>Traceback (most recent call last)
<span class="n">Cell</span> <span class="n">In</span><span class="p">[</span><span class="mi">114</span><span class="p">],</span> <span class="n">line</span> <span class="mi">2</span>
<span class="g g-Whitespace">      </span><span class="mi">1</span> <span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="ne">----&gt; </span><span class="mi">2</span> <span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">3</span>                  <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">4</span>                  <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Pathway Analysis - Early AT2 Like Cells&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">5</span>                  <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">6</span>                  <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">7</span>                  <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
<span class="g g-Whitespace">      </span><span class="mi">8</span>                  <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">9</span> <span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="g g-Whitespace">     </span><span class="mi">11</span> <span class="n">sc_plot_and_embed</span><span class="p">(</span>
<span class="g g-Whitespace">     </span><span class="mi">12</span>     <span class="n">ax_dot</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">13</span>     <span class="s2">&quot;gsea_earlyAT2_10&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">14</span>     <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Pathway Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">15</span>     <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="g g-Whitespace">     </span><span class="mi">16</span> <span class="p">)</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:1225,</span> in <span class="ni">dotplot</span><span class="nt">(df, column, x, y, x_order, y_order, title, cutoff, top_term, size, ax, figsize, cmap, ofname, xticklabels_rot, yticklabels_rot, marker, show_ring, **kwargs)</span>
<span class="g g-Whitespace">   </span><span class="mi">1222</span>     <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span><span class="s2">&quot;group is deprecated; use x instead&quot;</span><span class="p">,</span> <span class="ne">DeprecationWarning</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1223</span>     <span class="k">return</span>
<span class="ne">-&gt; </span><span class="mi">1225</span> <span class="n">dot</span> <span class="o">=</span> <span class="n">DotPlot</span><span class="p">(</span>
<span class="g g-Whitespace">   </span><span class="mi">1226</span>     <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1227</span>     <span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1228</span>     <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1229</span>     <span class="n">x_order</span><span class="o">=</span><span class="n">x_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1230</span>     <span class="n">y_order</span><span class="o">=</span><span class="n">y_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1231</span>     <span class="n">hue</span><span class="o">=</span><span class="n">column</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1232</span>     <span class="n">title</span><span class="o">=</span><span class="n">title</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1233</span>     <span class="n">thresh</span><span class="o">=</span><span class="n">cutoff</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1234</span>     <span class="n">n_terms</span><span class="o">=</span><span class="nb">int</span><span class="p">(</span><span class="n">top_term</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1235</span>     <span class="n">dot_scale</span><span class="o">=</span><span class="n">size</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1236</span>     <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1237</span>     <span class="n">figsize</span><span class="o">=</span><span class="n">figsize</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1238</span>     <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1239</span>     <span class="n">ofname</span><span class="o">=</span><span class="n">ofname</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1240</span>     <span class="n">marker</span><span class="o">=</span><span class="n">marker</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1241</span> <span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1242</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dot</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">outer_ring</span><span class="o">=</span><span class="n">show_ring</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1244</span> <span class="k">if</span> <span class="n">xticklabels_rot</span><span class="p">:</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:692,</span> in <span class="ni">DotPlot.__init__</span><span class="nt">(self, df, x, y, hue, dot_scale, x_order, y_order, thresh, n_terms, title, ax, figsize, cmap, ofname, marker, **kwargs)</span>
<span class="g g-Whitespace">    </span><span class="mi">690</span> <span class="bp">self</span><span class="o">.</span><span class="n">n_terms</span> <span class="o">=</span> <span class="n">n_terms</span>
<span class="g g-Whitespace">    </span><span class="mi">691</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span> <span class="o">=</span> <span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">692</span> <span class="bp">self</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">693</span> <span class="n">plt</span><span class="o">.</span><span class="n">rcParams</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="s2">&quot;pdf.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">,</span> <span class="s2">&quot;ps.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">})</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:717,</span> in <span class="ni">DotPlot.process</span><span class="nt">(self, df)</span>
<span class="g g-Whitespace">    </span><span class="mi">715</span> <span class="k">if</span> <span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">716</span>     <span class="n">msg</span> <span class="o">=</span> <span class="s2">&quot;Warning: No enrich terms when cutoff = </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">717</span>     <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="n">msg</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">718</span> <span class="bp">self</span><span class="o">.</span><span class="n">cbar_title</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">colname</span>
<span class="g g-Whitespace">    </span><span class="mi">719</span> <span class="c1"># clip GSEA lower bounds</span>
<span class="g g-Whitespace">    </span><span class="mi">720</span> <span class="c1"># if self.colname in [&quot;NOM p-val&quot;, &quot;FDR q-val&quot;]:</span>
<span class="g g-Whitespace">    </span><span class="mi">721</span> <span class="c1">#     df[self.colname].clip(1e-5, 1.0, inplace=True)</span>
<span class="g g-Whitespace">    </span><span class="mi">722</span> <span class="c1"># sorting the dataframe for better visualization</span>

<span class="ne">ValueError</span>: Warning: No enrich terms when cutoff = 0.05
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_11&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Pathway Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_earlyAT2_11.png" alt="gsea_earlyAT2_11" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Pathway Analysis Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_earlyAT2_11.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id6">
<h3>GSEA - Pathway Analysis [Top 500]<a class="headerlink" href="#id6" title="Link to this heading">#</a></h3>
<p>Reactome_Pathways_2024’,’KEGG_2021_Human’,’WikiPathways_2024_Human’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list_top</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Reactome_Pathways_2024&#39;</span><span class="p">,</span><span class="s1">&#39;KEGG_2021_Human&#39;</span><span class="p">,</span><span class="s1">&#39;WikiPathways_2024_Human&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>2025-09-04 05:50:28,117 [WARNING] Duplicated values found in preranked stats: 55.40% of genes
The order of those genes will be arbitrary, which may produce unexpected results.
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Peptide Ligand-Binding...</td>
      <td>-0.654545</td>
      <td>-1.712598</td>
      <td>0.0199</td>
      <td>0.081146</td>
      <td>0.135081</td>
      <td>5/5</td>
      <td>35.40%</td>
      <td>NPY5R;CCL5;APLNR;UTS2R;CCKBR</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Neutrophil Degranulation</td>
      <td>0.963154</td>
      <td>1.547553</td>
      <td>0.154412</td>
      <td>1.0</td>
      <td>0.826</td>
      <td>3/5</td>
      <td>4.00%</td>
      <td>DSP;EEF1A1;ANXA2</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Cell-cell Junction Org...</td>
      <td>0.946007</td>
      <td>1.538639</td>
      <td>0.18822</td>
      <td>1.0</td>
      <td>0.881</td>
      <td>3/5</td>
      <td>5.80%</td>
      <td>CADM1;CDH1;CTNNA1</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>WikiPathways_2024_Human__Nuclear Receptors Met...</td>
      <td>0.982828</td>
      <td>1.536057</td>
      <td>0.077987</td>
      <td>1.0</td>
      <td>0.897</td>
      <td>1/5</td>
      <td>1.40%</td>
      <td>MYOF</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Cell Junction Organiza...</td>
      <td>0.944489</td>
      <td>1.532521</td>
      <td>0.200963</td>
      <td>1.0</td>
      <td>0.906</td>
      <td>3/6</td>
      <td>5.80%</td>
      <td>CADM1;CDH1;CTNNA1</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>62</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Generic Transcription ...</td>
      <td>0.449393</td>
      <td>0.703931</td>
      <td>0.660218</td>
      <td>0.701475</td>
      <td>1.0</td>
      <td>6/6</td>
      <td>55.60%</td>
      <td>ZNF558;FASLG;RUNX3;NFE2;SERPINB13;NR1I2</td>
    </tr>
    <tr>
      <th>63</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Metabolism of Lipids</td>
      <td>-0.244939</td>
      <td>-0.699373</td>
      <td>0.882353</td>
      <td>0.878699</td>
      <td>0.995968</td>
      <td>6/6</td>
      <td>76.00%</td>
      <td>APOA2;THRSP;CPNE6;CHAT;ARSH;CYP4A22</td>
    </tr>
    <tr>
      <th>64</th>
      <td>prerank</td>
      <td>KEGG_2021_Human__Complement and coagulation ca...</td>
      <td>0.362738</td>
      <td>0.575886</td>
      <td>0.823601</td>
      <td>0.820968</td>
      <td>1.0</td>
      <td>3/5</td>
      <td>38.40%</td>
      <td>C4A;C1QB;ITGAM</td>
    </tr>
    <tr>
      <th>65</th>
      <td>prerank</td>
      <td>KEGG_2021_Human__Staphylococcus aureus infection</td>
      <td>0.347282</td>
      <td>0.54847</td>
      <td>0.852619</td>
      <td>0.83674</td>
      <td>1.0</td>
      <td>3/5</td>
      <td>38.40%</td>
      <td>C4A;C1QB;ITGAM</td>
    </tr>
    <tr>
      <th>66</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Transport of Small Mol...</td>
      <td>0.330495</td>
      <td>0.52611</td>
      <td>0.83945</td>
      <td>0.84564</td>
      <td>1.0</td>
      <td>4/8</td>
      <td>41.00%</td>
      <td>LPA;ANO2;APOA4;SLCO1C1</td>
    </tr>
  </tbody>
</table>
<p>67 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Pathway Analysis - Early AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>


<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_12&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Pathway Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">ValueError</span><span class="g g-Whitespace">                                </span>Traceback (most recent call last)
<span class="n">Cell</span> <span class="n">In</span><span class="p">[</span><span class="mi">125</span><span class="p">],</span> <span class="n">line</span> <span class="mi">2</span>
<span class="g g-Whitespace">      </span><span class="mi">1</span> <span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="ne">----&gt; </span><span class="mi">2</span> <span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">3</span>                  <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">4</span>                  <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Pathway Analysis - Early AT2 Like Cells&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">5</span>                  <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">6</span>                  <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">7</span>                  <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
<span class="g g-Whitespace">      </span><span class="mi">8</span>                  <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">9</span> <span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="g g-Whitespace">     </span><span class="mi">12</span> <span class="n">sc_plot_and_embed</span><span class="p">(</span>
<span class="g g-Whitespace">     </span><span class="mi">13</span>     <span class="n">ax_dot</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">14</span>     <span class="s2">&quot;gsea_earlyAT2_12&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">15</span>     <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Pathway Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">16</span>     <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="g g-Whitespace">     </span><span class="mi">17</span> <span class="p">)</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:1225,</span> in <span class="ni">dotplot</span><span class="nt">(df, column, x, y, x_order, y_order, title, cutoff, top_term, size, ax, figsize, cmap, ofname, xticklabels_rot, yticklabels_rot, marker, show_ring, **kwargs)</span>
<span class="g g-Whitespace">   </span><span class="mi">1222</span>     <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span><span class="s2">&quot;group is deprecated; use x instead&quot;</span><span class="p">,</span> <span class="ne">DeprecationWarning</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1223</span>     <span class="k">return</span>
<span class="ne">-&gt; </span><span class="mi">1225</span> <span class="n">dot</span> <span class="o">=</span> <span class="n">DotPlot</span><span class="p">(</span>
<span class="g g-Whitespace">   </span><span class="mi">1226</span>     <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1227</span>     <span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1228</span>     <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1229</span>     <span class="n">x_order</span><span class="o">=</span><span class="n">x_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1230</span>     <span class="n">y_order</span><span class="o">=</span><span class="n">y_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1231</span>     <span class="n">hue</span><span class="o">=</span><span class="n">column</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1232</span>     <span class="n">title</span><span class="o">=</span><span class="n">title</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1233</span>     <span class="n">thresh</span><span class="o">=</span><span class="n">cutoff</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1234</span>     <span class="n">n_terms</span><span class="o">=</span><span class="nb">int</span><span class="p">(</span><span class="n">top_term</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1235</span>     <span class="n">dot_scale</span><span class="o">=</span><span class="n">size</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1236</span>     <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1237</span>     <span class="n">figsize</span><span class="o">=</span><span class="n">figsize</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1238</span>     <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1239</span>     <span class="n">ofname</span><span class="o">=</span><span class="n">ofname</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1240</span>     <span class="n">marker</span><span class="o">=</span><span class="n">marker</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1241</span> <span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1242</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dot</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">outer_ring</span><span class="o">=</span><span class="n">show_ring</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1244</span> <span class="k">if</span> <span class="n">xticklabels_rot</span><span class="p">:</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:692,</span> in <span class="ni">DotPlot.__init__</span><span class="nt">(self, df, x, y, hue, dot_scale, x_order, y_order, thresh, n_terms, title, ax, figsize, cmap, ofname, marker, **kwargs)</span>
<span class="g g-Whitespace">    </span><span class="mi">690</span> <span class="bp">self</span><span class="o">.</span><span class="n">n_terms</span> <span class="o">=</span> <span class="n">n_terms</span>
<span class="g g-Whitespace">    </span><span class="mi">691</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span> <span class="o">=</span> <span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">692</span> <span class="bp">self</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">693</span> <span class="n">plt</span><span class="o">.</span><span class="n">rcParams</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="s2">&quot;pdf.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">,</span> <span class="s2">&quot;ps.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">})</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:717,</span> in <span class="ni">DotPlot.process</span><span class="nt">(self, df)</span>
<span class="g g-Whitespace">    </span><span class="mi">715</span> <span class="k">if</span> <span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">716</span>     <span class="n">msg</span> <span class="o">=</span> <span class="s2">&quot;Warning: No enrich terms when cutoff = </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">717</span>     <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="n">msg</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">718</span> <span class="bp">self</span><span class="o">.</span><span class="n">cbar_title</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">colname</span>
<span class="g g-Whitespace">    </span><span class="mi">719</span> <span class="c1"># clip GSEA lower bounds</span>
<span class="g g-Whitespace">    </span><span class="mi">720</span> <span class="c1"># if self.colname in [&quot;NOM p-val&quot;, &quot;FDR q-val&quot;]:</span>
<span class="g g-Whitespace">    </span><span class="mi">721</span> <span class="c1">#     df[self.colname].clip(1e-5, 1.0, inplace=True)</span>
<span class="g g-Whitespace">    </span><span class="mi">722</span> <span class="c1"># sorting the dataframe for better visualization</span>

<span class="ne">ValueError</span>: Warning: No enrich terms when cutoff = 0.05
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_13&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Pathway Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_earlyAT2_13.png" alt="gsea_earlyAT2_13" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Pathway Analysis Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_earlyAT2_13.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id7">
<h3>GSEA - Proteomic Analysis [Top 250 &amp; Down 250]<a class="headerlink" href="#id7" title="Link to this heading">#</a></h3>
<p>‘ProteomicsDB_2020’,’Tissue_Protein_Expression_from_Human_Proteome_Map’,’Tissue_Protein_Expression_from_ProteomicsDB’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;ProteomicsDB_2020&#39;</span><span class="p">,</span><span class="s1">&#39;Tissue_Protein_Expression_from_Human_Proteome_Map&#39;</span><span class="p">,</span><span class="s1">&#39;Tissue_Protein_Expression_from_ProteomicsDB&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>2025-09-04 05:50:36,042 [WARNING] Duplicated values found in preranked stats: 19.20% of genes
The order of those genes will be arbitrary, which may produce unexpected results.
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_Human_Proteome_...</td>
      <td>0.990138</td>
      <td>2.708403</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>1/8</td>
      <td>0.40%</td>
      <td>GPC5</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_ProteomicsDB__h...</td>
      <td>-0.982946</td>
      <td>-1.63595</td>
      <td>0.002039</td>
      <td>0.324555</td>
      <td>0.171</td>
      <td>1/7</td>
      <td>1.60%</td>
      <td>CENPP</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_Human_Proteome_...</td>
      <td>-0.887507</td>
      <td>-1.512151</td>
      <td>0.005066</td>
      <td>1.0</td>
      <td>0.923</td>
      <td>1/8</td>
      <td>11.00%</td>
      <td>ACOXL</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_Human_Proteome_...</td>
      <td>-0.920565</td>
      <td>-1.486473</td>
      <td>0.015385</td>
      <td>1.0</td>
      <td>0.966</td>
      <td>4/6</td>
      <td>8.80%</td>
      <td>SLC34A2;LAMP3;SFTPB;SFTPA1</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_Human_Proteome_...</td>
      <td>-0.936006</td>
      <td>-1.452836</td>
      <td>0.010526</td>
      <td>1.0</td>
      <td>0.99</td>
      <td>2/5</td>
      <td>6.80%</td>
      <td>PRIM2;FAM111B</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>811</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Cecum C84 PDB:200069 C84 CRC65</td>
      <td>-0.362442</td>
      <td>-0.599469</td>
      <td>0.993821</td>
      <td>0.999428</td>
      <td>1.0</td>
      <td>2/7</td>
      <td>40.60%</td>
      <td>RALGAPA2;FDPS</td>
    </tr>
    <tr>
      <th>812</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_ProteomicsDB__K...</td>
      <td>-0.366065</td>
      <td>-0.58965</td>
      <td>0.989765</td>
      <td>0.998976</td>
      <td>1.0</td>
      <td>2/6</td>
      <td>21.60%</td>
      <td>ADAMTS9;RHOBTB2</td>
    </tr>
    <tr>
      <th>813</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Melanoma SK-MEL-2 BTO:00034...</td>
      <td>-0.336754</td>
      <td>-0.585381</td>
      <td>0.990946</td>
      <td>0.998018</td>
      <td>1.0</td>
      <td>5/10</td>
      <td>48.80%</td>
      <td>COBLL1;XRCC5;ATP1B3;TKT;PDIA4</td>
    </tr>
    <tr>
      <th>814</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_ProteomicsDB__a...</td>
      <td>-0.317429</td>
      <td>-0.528722</td>
      <td>0.996945</td>
      <td>0.999538</td>
      <td>1.0</td>
      <td>4/7</td>
      <td>48.20%</td>
      <td>DIAPH3;PDE8A;ELOVL6;AFF3</td>
    </tr>
    <tr>
      <th>815</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_ProteomicsDB__H...</td>
      <td>-0.331298</td>
      <td>-0.515125</td>
      <td>0.99582</td>
      <td>0.998741</td>
      <td>1.0</td>
      <td>2/5</td>
      <td>43.60%</td>
      <td>PBX3;SLC2A13</td>
    </tr>
  </tbody>
</table>
<p>816 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Proteomic Analysis - Early AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">6</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_14&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Proteomic Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">ValueError</span><span class="g g-Whitespace">                                </span>Traceback (most recent call last)
<span class="n">Cell</span> <span class="n">In</span><span class="p">[</span><span class="mi">128</span><span class="p">],</span> <span class="n">line</span> <span class="mi">2</span>
<span class="g g-Whitespace">      </span><span class="mi">1</span> <span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="ne">----&gt; </span><span class="mi">2</span> <span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">3</span>                  <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">4</span>                  <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Proteomic Analysis - Early AT2 Like Cells&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">5</span>                  <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">6</span>                  <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">7</span>                  <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">6</span><span class="p">),</span> 
<span class="g g-Whitespace">      </span><span class="mi">8</span>                  <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">9</span> <span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="g g-Whitespace">     </span><span class="mi">11</span> <span class="n">sc_plot_and_embed</span><span class="p">(</span>
<span class="g g-Whitespace">     </span><span class="mi">12</span>     <span class="n">ax_dot</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">13</span>     <span class="s2">&quot;gsea_earlyAT2_14&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">14</span>     <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Proteomic Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">15</span>     <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="g g-Whitespace">     </span><span class="mi">16</span> <span class="p">)</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:1225,</span> in <span class="ni">dotplot</span><span class="nt">(df, column, x, y, x_order, y_order, title, cutoff, top_term, size, ax, figsize, cmap, ofname, xticklabels_rot, yticklabels_rot, marker, show_ring, **kwargs)</span>
<span class="g g-Whitespace">   </span><span class="mi">1222</span>     <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span><span class="s2">&quot;group is deprecated; use x instead&quot;</span><span class="p">,</span> <span class="ne">DeprecationWarning</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1223</span>     <span class="k">return</span>
<span class="ne">-&gt; </span><span class="mi">1225</span> <span class="n">dot</span> <span class="o">=</span> <span class="n">DotPlot</span><span class="p">(</span>
<span class="g g-Whitespace">   </span><span class="mi">1226</span>     <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1227</span>     <span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1228</span>     <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1229</span>     <span class="n">x_order</span><span class="o">=</span><span class="n">x_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1230</span>     <span class="n">y_order</span><span class="o">=</span><span class="n">y_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1231</span>     <span class="n">hue</span><span class="o">=</span><span class="n">column</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1232</span>     <span class="n">title</span><span class="o">=</span><span class="n">title</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1233</span>     <span class="n">thresh</span><span class="o">=</span><span class="n">cutoff</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1234</span>     <span class="n">n_terms</span><span class="o">=</span><span class="nb">int</span><span class="p">(</span><span class="n">top_term</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1235</span>     <span class="n">dot_scale</span><span class="o">=</span><span class="n">size</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1236</span>     <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1237</span>     <span class="n">figsize</span><span class="o">=</span><span class="n">figsize</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1238</span>     <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1239</span>     <span class="n">ofname</span><span class="o">=</span><span class="n">ofname</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1240</span>     <span class="n">marker</span><span class="o">=</span><span class="n">marker</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1241</span> <span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1242</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dot</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">outer_ring</span><span class="o">=</span><span class="n">show_ring</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1244</span> <span class="k">if</span> <span class="n">xticklabels_rot</span><span class="p">:</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:692,</span> in <span class="ni">DotPlot.__init__</span><span class="nt">(self, df, x, y, hue, dot_scale, x_order, y_order, thresh, n_terms, title, ax, figsize, cmap, ofname, marker, **kwargs)</span>
<span class="g g-Whitespace">    </span><span class="mi">690</span> <span class="bp">self</span><span class="o">.</span><span class="n">n_terms</span> <span class="o">=</span> <span class="n">n_terms</span>
<span class="g g-Whitespace">    </span><span class="mi">691</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span> <span class="o">=</span> <span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">692</span> <span class="bp">self</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">693</span> <span class="n">plt</span><span class="o">.</span><span class="n">rcParams</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="s2">&quot;pdf.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">,</span> <span class="s2">&quot;ps.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">})</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:733,</span> in <span class="ni">DotPlot.process</span><span class="nt">(self, df)</span>
<span class="g g-Whitespace">    </span><span class="mi">730</span> <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">colname</span> <span class="ow">in</span> <span class="p">[</span><span class="s2">&quot;Adjusted P-value&quot;</span><span class="p">,</span> <span class="s2">&quot;P-value&quot;</span><span class="p">,</span> <span class="s2">&quot;NOM p-val&quot;</span><span class="p">,</span> <span class="s2">&quot;FDR q-val&quot;</span><span class="p">]:</span>
<span class="g g-Whitespace">    </span><span class="mi">731</span>     <span class="c1"># if all values are zeros, raise error</span>
<span class="g g-Whitespace">    </span><span class="mi">732</span>     <span class="k">if</span> <span class="ow">not</span> <span class="nb">any</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="bp">self</span><span class="o">.</span><span class="n">colname</span><span class="p">]</span><span class="o">.</span><span class="n">abs</span><span class="p">()</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">):</span>
<span class="ne">--&gt; </span><span class="mi">733</span>         <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
<span class="g g-Whitespace">    </span><span class="mi">734</span>             <span class="sa">f</span><span class="s2">&quot;Can not detetermine colormap. All values in </span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">colname</span><span class="si">}</span><span class="s2"> are 0s&quot;</span>
<span class="g g-Whitespace">    </span><span class="mi">735</span>         <span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">736</span>     <span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">colname</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">737</span>     <span class="n">df</span><span class="p">[</span><span class="bp">self</span><span class="o">.</span><span class="n">colname</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="bp">self</span><span class="o">.</span><span class="n">colname</span><span class="p">]</span> <span class="o">+</span> <span class="n">np</span><span class="o">.</span><span class="n">finfo</span><span class="p">(</span><span class="nb">float</span><span class="p">)</span><span class="o">.</span><span class="n">eps</span>

<span class="ne">ValueError</span>: Can not detetermine colormap. All values in FDR q-val are 0s
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_15&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Proteomic Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_earlyAT2_15.png" alt="gsea_earlyAT2_15" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Proteomic Analysis Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_earlyAT2_15.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id8">
<h3>GSEA - Proteomic Analysis [Top 500]<a class="headerlink" href="#id8" title="Link to this heading">#</a></h3>
<p>‘ProteomicsDB_2020’,’Tissue_Protein_Expression_from_Human_Proteome_Map’,’Tissue_Protein_Expression_from_ProteomicsDB’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list_top</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;ProteomicsDB_2020&#39;</span><span class="p">,</span><span class="s1">&#39;Tissue_Protein_Expression_from_Human_Proteome_Map&#39;</span><span class="p">,</span><span class="s1">&#39;Tissue_Protein_Expression_from_ProteomicsDB&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>2025-09-04 05:52:20,444 [WARNING] Duplicated values found in preranked stats: 55.40% of genes
The order of those genes will be arbitrary, which may produce unexpected results.
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_Human_Proteome_...</td>
      <td>-0.635628</td>
      <td>-1.785906</td>
      <td>0.006329</td>
      <td>0.137925</td>
      <td>0.213</td>
      <td>6/6</td>
      <td>37.40%</td>
      <td>KCNN1;FAM131B;PRRT4;ASPRV1;APLNR;HAPLN4</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_Human_Proteome_...</td>
      <td>0.986742</td>
      <td>1.575621</td>
      <td>0.041312</td>
      <td>1.0</td>
      <td>0.913</td>
      <td>1/6</td>
      <td>0.80%</td>
      <td>SYN3</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Brain GaMG BTO:0004860 GAMG R3</td>
      <td>0.965491</td>
      <td>1.574968</td>
      <td>0.141782</td>
      <td>1.0</td>
      <td>0.915</td>
      <td>4/5</td>
      <td>4.20%</td>
      <td>MYOF;MAGI3;CADM1;PTPRK</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Lymphoblastoid BTO:0000773 ...</td>
      <td>0.97321</td>
      <td>1.572288</td>
      <td>0.116788</td>
      <td>1.0</td>
      <td>0.93</td>
      <td>2/6</td>
      <td>2.80%</td>
      <td>CADM1;LRBA</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Lymphoblastoid BTO:0000773 ...</td>
      <td>0.97321</td>
      <td>1.572288</td>
      <td>0.116788</td>
      <td>1.0</td>
      <td>0.93</td>
      <td>2/6</td>
      <td>2.80%</td>
      <td>CADM1;LRBA</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>226</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Lymphoblastoid BTO:0000773 ...</td>
      <td>0.230294</td>
      <td>0.365172</td>
      <td>0.985612</td>
      <td>0.985625</td>
      <td>1.0</td>
      <td>4/6</td>
      <td>61.60%</td>
      <td>RASAL3;RUNX3;HLA-DOA;CD19</td>
    </tr>
    <tr>
      <th>227</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Lymphoblastoid BTO:0000773 ...</td>
      <td>0.230294</td>
      <td>0.365172</td>
      <td>0.985612</td>
      <td>0.985625</td>
      <td>1.0</td>
      <td>4/6</td>
      <td>61.60%</td>
      <td>RASAL3;RUNX3;HLA-DOA;CD19</td>
    </tr>
    <tr>
      <th>228</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Lymphoblastoid BTO:0000773 ...</td>
      <td>0.230294</td>
      <td>0.365172</td>
      <td>0.985612</td>
      <td>0.985625</td>
      <td>1.0</td>
      <td>4/6</td>
      <td>61.60%</td>
      <td>RASAL3;RUNX3;HLA-DOA;CD19</td>
    </tr>
    <tr>
      <th>229</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Lymphoblastoid BTO:0000773 ...</td>
      <td>0.230294</td>
      <td>0.365172</td>
      <td>0.985612</td>
      <td>0.985625</td>
      <td>1.0</td>
      <td>4/6</td>
      <td>61.60%</td>
      <td>RASAL3;RUNX3;HLA-DOA;CD19</td>
    </tr>
    <tr>
      <th>230</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_ProteomicsDB__c...</td>
      <td>0.221396</td>
      <td>0.355288</td>
      <td>1.0</td>
      <td>0.98479</td>
      <td>1.0</td>
      <td>1/5</td>
      <td>20.80%</td>
      <td>RASAL3</td>
    </tr>
  </tbody>
</table>
<p>231 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Proteomic Analysis - Early AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">6</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_16&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Proteomic Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">ValueError</span><span class="g g-Whitespace">                                </span>Traceback (most recent call last)
<span class="n">Cell</span> <span class="n">In</span><span class="p">[</span><span class="mi">136</span><span class="p">],</span> <span class="n">line</span> <span class="mi">2</span>
<span class="g g-Whitespace">      </span><span class="mi">1</span> <span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="ne">----&gt; </span><span class="mi">2</span> <span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">3</span>                  <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">4</span>                  <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Proteomic Analysis - Early AT2 Like Cells&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">5</span>                  <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">6</span>                  <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">7</span>                  <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">6</span><span class="p">),</span> 
<span class="g g-Whitespace">      </span><span class="mi">8</span>                  <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">9</span> <span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="g g-Whitespace">     </span><span class="mi">11</span> <span class="n">sc_plot_and_embed</span><span class="p">(</span>
<span class="g g-Whitespace">     </span><span class="mi">12</span>     <span class="n">ax_dot</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">13</span>     <span class="s2">&quot;gsea_earlyAT2_16&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">14</span>     <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Proteomic Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">15</span>     <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="g g-Whitespace">     </span><span class="mi">16</span> <span class="p">)</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:1225,</span> in <span class="ni">dotplot</span><span class="nt">(df, column, x, y, x_order, y_order, title, cutoff, top_term, size, ax, figsize, cmap, ofname, xticklabels_rot, yticklabels_rot, marker, show_ring, **kwargs)</span>
<span class="g g-Whitespace">   </span><span class="mi">1222</span>     <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span><span class="s2">&quot;group is deprecated; use x instead&quot;</span><span class="p">,</span> <span class="ne">DeprecationWarning</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1223</span>     <span class="k">return</span>
<span class="ne">-&gt; </span><span class="mi">1225</span> <span class="n">dot</span> <span class="o">=</span> <span class="n">DotPlot</span><span class="p">(</span>
<span class="g g-Whitespace">   </span><span class="mi">1226</span>     <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1227</span>     <span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1228</span>     <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1229</span>     <span class="n">x_order</span><span class="o">=</span><span class="n">x_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1230</span>     <span class="n">y_order</span><span class="o">=</span><span class="n">y_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1231</span>     <span class="n">hue</span><span class="o">=</span><span class="n">column</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1232</span>     <span class="n">title</span><span class="o">=</span><span class="n">title</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1233</span>     <span class="n">thresh</span><span class="o">=</span><span class="n">cutoff</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1234</span>     <span class="n">n_terms</span><span class="o">=</span><span class="nb">int</span><span class="p">(</span><span class="n">top_term</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1235</span>     <span class="n">dot_scale</span><span class="o">=</span><span class="n">size</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1236</span>     <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1237</span>     <span class="n">figsize</span><span class="o">=</span><span class="n">figsize</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1238</span>     <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1239</span>     <span class="n">ofname</span><span class="o">=</span><span class="n">ofname</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1240</span>     <span class="n">marker</span><span class="o">=</span><span class="n">marker</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1241</span> <span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1242</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dot</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">outer_ring</span><span class="o">=</span><span class="n">show_ring</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1244</span> <span class="k">if</span> <span class="n">xticklabels_rot</span><span class="p">:</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:692,</span> in <span class="ni">DotPlot.__init__</span><span class="nt">(self, df, x, y, hue, dot_scale, x_order, y_order, thresh, n_terms, title, ax, figsize, cmap, ofname, marker, **kwargs)</span>
<span class="g g-Whitespace">    </span><span class="mi">690</span> <span class="bp">self</span><span class="o">.</span><span class="n">n_terms</span> <span class="o">=</span> <span class="n">n_terms</span>
<span class="g g-Whitespace">    </span><span class="mi">691</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span> <span class="o">=</span> <span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">692</span> <span class="bp">self</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">693</span> <span class="n">plt</span><span class="o">.</span><span class="n">rcParams</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="s2">&quot;pdf.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">,</span> <span class="s2">&quot;ps.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">})</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:717,</span> in <span class="ni">DotPlot.process</span><span class="nt">(self, df)</span>
<span class="g g-Whitespace">    </span><span class="mi">715</span> <span class="k">if</span> <span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">716</span>     <span class="n">msg</span> <span class="o">=</span> <span class="s2">&quot;Warning: No enrich terms when cutoff = </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">717</span>     <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="n">msg</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">718</span> <span class="bp">self</span><span class="o">.</span><span class="n">cbar_title</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">colname</span>
<span class="g g-Whitespace">    </span><span class="mi">719</span> <span class="c1"># clip GSEA lower bounds</span>
<span class="g g-Whitespace">    </span><span class="mi">720</span> <span class="c1"># if self.colname in [&quot;NOM p-val&quot;, &quot;FDR q-val&quot;]:</span>
<span class="g g-Whitespace">    </span><span class="mi">721</span> <span class="c1">#     df[self.colname].clip(1e-5, 1.0, inplace=True)</span>
<span class="g g-Whitespace">    </span><span class="mi">722</span> <span class="c1"># sorting the dataframe for better visualization</span>

<span class="ne">ValueError</span>: Warning: No enrich terms when cutoff = 0.05
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_17&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Proteomic Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_earlyAT2_17.png" alt="gsea_earlyAT2_17" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Proteomic Analysis Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_earlyAT2_17.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id9">
<h3>GSEA - Disease Analysis [Top 250 &amp; Down 250]<a class="headerlink" href="#id9" title="Link to this heading">#</a></h3>
<p>‘DisGeNET’,’Disease_Perturbations_from_GEO_down’,’Disease_Perturbations_from_GEO_up’,’HDSigDB_Human_2021’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;DisGeNET&#39;</span><span class="p">,</span><span class="s1">&#39;Disease_Perturbations_from_GEO_down&#39;</span><span class="p">,</span><span class="s1">&#39;Disease_Perturbations_from_GEO_up&#39;</span><span class="p">,</span><span class="s1">&#39;HDSigDB_Human_2021&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>2025-09-04 05:52:28,709 [WARNING] Duplicated values found in preranked stats: 19.20% of genes
The order of those genes will be arbitrary, which may produce unexpected results.
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>DisGeNET__Hypothyroidism</td>
      <td>0.957611</td>
      <td>1.959374</td>
      <td>0.04</td>
      <td>0.105195</td>
      <td>0.183</td>
      <td>3/7</td>
      <td>4.00%</td>
      <td>MAGI3;LRBA;ANXA2</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_up__systemic ju...</td>
      <td>-0.970577</td>
      <td>-1.717363</td>
      <td>0.0</td>
      <td>0.068485</td>
      <td>0.061</td>
      <td>4/11</td>
      <td>3.80%</td>
      <td>TOP2A;MKI67;LAMP3;RRM2</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__PRC2 Deficiency Up-Regulat...</td>
      <td>0.981127</td>
      <td>1.711241</td>
      <td>0.05</td>
      <td>0.336198</td>
      <td>0.655</td>
      <td>1/6</td>
      <td>1.20%</td>
      <td>DSP</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_up__cystic fibr...</td>
      <td>-0.991843</td>
      <td>-1.611749</td>
      <td>0.0</td>
      <td>0.991023</td>
      <td>0.764</td>
      <td>1/6</td>
      <td>1.00%</td>
      <td>MKI67</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>DisGeNET__Hyperthyroidism</td>
      <td>0.956452</td>
      <td>1.603998</td>
      <td>0.101449</td>
      <td>0.494859</td>
      <td>0.892</td>
      <td>3/5</td>
      <td>4.60%</td>
      <td>MAGI3;ANXA2;VIM</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>3464</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_up__idiopathic ...</td>
      <td>-0.307571</td>
      <td>-0.521286</td>
      <td>0.996936</td>
      <td>0.999979</td>
      <td>1.0</td>
      <td>2/8</td>
      <td>42.60%</td>
      <td>HS6ST2;EGFEM1P</td>
    </tr>
    <tr>
      <th>3465</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_down__Alcohol p...</td>
      <td>-0.310182</td>
      <td>-0.503895</td>
      <td>0.996904</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>2/6</td>
      <td>45.40%</td>
      <td>ANP32B;SERBP1</td>
    </tr>
    <tr>
      <th>3466</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Striatum Protein Module pM...</td>
      <td>-0.319786</td>
      <td>-0.495485</td>
      <td>1.0</td>
      <td>0.999951</td>
      <td>1.0</td>
      <td>2/5</td>
      <td>45.40%</td>
      <td>SNX25;SERBP1</td>
    </tr>
    <tr>
      <th>3467</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_up__idiopathic ...</td>
      <td>-0.310434</td>
      <td>-0.491964</td>
      <td>0.998955</td>
      <td>0.999728</td>
      <td>1.0</td>
      <td>2/5</td>
      <td>42.60%</td>
      <td>HS6ST2;EGFEM1P</td>
    </tr>
    <tr>
      <th>3468</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Definite Neuron-Enriched G...</td>
      <td>-0.275771</td>
      <td>-0.456533</td>
      <td>0.998989</td>
      <td>0.999839</td>
      <td>1.0</td>
      <td>2/7</td>
      <td>43.80%</td>
      <td>HS6ST2;LMO3</td>
    </tr>
  </tbody>
</table>
<p>3469 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Pathway Analysis - Early AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_18&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Disease Analysis Datasets [Top 250 &amp; Down Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">ValueError</span><span class="g g-Whitespace">                                </span>Traceback (most recent call last)
<span class="n">Cell</span> <span class="n">In</span><span class="p">[</span><span class="mi">139</span><span class="p">],</span> <span class="n">line</span> <span class="mi">2</span>
<span class="g g-Whitespace">      </span><span class="mi">1</span> <span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="ne">----&gt; </span><span class="mi">2</span> <span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">3</span>                  <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">4</span>                  <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Pathway Analysis - Early AT2 Like Cells&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">5</span>                  <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">6</span>                  <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">7</span>                  <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
<span class="g g-Whitespace">      </span><span class="mi">8</span>                  <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">9</span> <span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="g g-Whitespace">     </span><span class="mi">11</span> <span class="n">sc_plot_and_embed</span><span class="p">(</span>
<span class="g g-Whitespace">     </span><span class="mi">12</span>     <span class="n">ax_dot</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">13</span>     <span class="s2">&quot;gsea_earlyAT2_18&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">14</span>     <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Disease Analysis Datasets [Top 250 &amp; Down Differentially Expressed Genes].&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">15</span>     <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="g g-Whitespace">     </span><span class="mi">16</span> <span class="p">)</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:1225,</span> in <span class="ni">dotplot</span><span class="nt">(df, column, x, y, x_order, y_order, title, cutoff, top_term, size, ax, figsize, cmap, ofname, xticklabels_rot, yticklabels_rot, marker, show_ring, **kwargs)</span>
<span class="g g-Whitespace">   </span><span class="mi">1222</span>     <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span><span class="s2">&quot;group is deprecated; use x instead&quot;</span><span class="p">,</span> <span class="ne">DeprecationWarning</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1223</span>     <span class="k">return</span>
<span class="ne">-&gt; </span><span class="mi">1225</span> <span class="n">dot</span> <span class="o">=</span> <span class="n">DotPlot</span><span class="p">(</span>
<span class="g g-Whitespace">   </span><span class="mi">1226</span>     <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1227</span>     <span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1228</span>     <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1229</span>     <span class="n">x_order</span><span class="o">=</span><span class="n">x_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1230</span>     <span class="n">y_order</span><span class="o">=</span><span class="n">y_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1231</span>     <span class="n">hue</span><span class="o">=</span><span class="n">column</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1232</span>     <span class="n">title</span><span class="o">=</span><span class="n">title</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1233</span>     <span class="n">thresh</span><span class="o">=</span><span class="n">cutoff</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1234</span>     <span class="n">n_terms</span><span class="o">=</span><span class="nb">int</span><span class="p">(</span><span class="n">top_term</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1235</span>     <span class="n">dot_scale</span><span class="o">=</span><span class="n">size</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1236</span>     <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1237</span>     <span class="n">figsize</span><span class="o">=</span><span class="n">figsize</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1238</span>     <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1239</span>     <span class="n">ofname</span><span class="o">=</span><span class="n">ofname</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1240</span>     <span class="n">marker</span><span class="o">=</span><span class="n">marker</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1241</span> <span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1242</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dot</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">outer_ring</span><span class="o">=</span><span class="n">show_ring</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1244</span> <span class="k">if</span> <span class="n">xticklabels_rot</span><span class="p">:</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:692,</span> in <span class="ni">DotPlot.__init__</span><span class="nt">(self, df, x, y, hue, dot_scale, x_order, y_order, thresh, n_terms, title, ax, figsize, cmap, ofname, marker, **kwargs)</span>
<span class="g g-Whitespace">    </span><span class="mi">690</span> <span class="bp">self</span><span class="o">.</span><span class="n">n_terms</span> <span class="o">=</span> <span class="n">n_terms</span>
<span class="g g-Whitespace">    </span><span class="mi">691</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span> <span class="o">=</span> <span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">692</span> <span class="bp">self</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">693</span> <span class="n">plt</span><span class="o">.</span><span class="n">rcParams</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="s2">&quot;pdf.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">,</span> <span class="s2">&quot;ps.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">})</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:717,</span> in <span class="ni">DotPlot.process</span><span class="nt">(self, df)</span>
<span class="g g-Whitespace">    </span><span class="mi">715</span> <span class="k">if</span> <span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">716</span>     <span class="n">msg</span> <span class="o">=</span> <span class="s2">&quot;Warning: No enrich terms when cutoff = </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">717</span>     <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="n">msg</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">718</span> <span class="bp">self</span><span class="o">.</span><span class="n">cbar_title</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">colname</span>
<span class="g g-Whitespace">    </span><span class="mi">719</span> <span class="c1"># clip GSEA lower bounds</span>
<span class="g g-Whitespace">    </span><span class="mi">720</span> <span class="c1"># if self.colname in [&quot;NOM p-val&quot;, &quot;FDR q-val&quot;]:</span>
<span class="g g-Whitespace">    </span><span class="mi">721</span> <span class="c1">#     df[self.colname].clip(1e-5, 1.0, inplace=True)</span>
<span class="g g-Whitespace">    </span><span class="mi">722</span> <span class="c1"># sorting the dataframe for better visualization</span>

<span class="ne">ValueError</span>: Warning: No enrich terms when cutoff = 0.05
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_19&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Disease Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_earlyAT2_19.png" alt="gsea_earlyAT2_19" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Disease Analysis Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_earlyAT2_19.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id10">
<h3>GSEA - Disease Analysis [Top 500]<a class="headerlink" href="#id10" title="Link to this heading">#</a></h3>
<p>‘DisGeNET’,’Disease_Perturbations_from_GEO_down’,’Disease_Perturbations_from_GEO_up’,’HDSigDB_Human_2021’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list_top</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;DisGeNET&#39;</span><span class="p">,</span><span class="s1">&#39;Disease_Perturbations_from_GEO_down&#39;</span><span class="p">,</span><span class="s1">&#39;Disease_Perturbations_from_GEO_up&#39;</span><span class="p">,</span><span class="s1">&#39;HDSigDB_Human_2021&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>2025-09-04 05:53:04,780 [WARNING] Duplicated values found in preranked stats: 55.40% of genes
The order of those genes will be arbitrary, which may produce unexpected results.
</pre></div>
</div>
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_up__Down Syndro...</td>
      <td>0.991143</td>
      <td>1.640545</td>
      <td>0.021871</td>
      <td>0.193326</td>
      <td>0.142</td>
      <td>1/6</td>
      <td>0.60%</td>
      <td>NCKAP5</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Changed In HD Vs Con...</td>
      <td>0.996901</td>
      <td>1.632813</td>
      <td>0.00861</td>
      <td>0.170364</td>
      <td>0.288</td>
      <td>1/5</td>
      <td>0.20%</td>
      <td>ADAMTSL1</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Down-Regulated In HD...</td>
      <td>0.996901</td>
      <td>1.632813</td>
      <td>0.00861</td>
      <td>0.170364</td>
      <td>0.288</td>
      <td>1/5</td>
      <td>0.20%</td>
      <td>ADAMTSL1</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_up__acute myoca...</td>
      <td>0.984628</td>
      <td>1.626526</td>
      <td>0.060026</td>
      <td>0.239991</td>
      <td>0.441</td>
      <td>1/5</td>
      <td>1.20%</td>
      <td>DSP</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_down__Cardiomyo...</td>
      <td>0.984112</td>
      <td>1.625534</td>
      <td>0.049296</td>
      <td>0.213326</td>
      <td>0.475</td>
      <td>1/6</td>
      <td>1.20%</td>
      <td>DSP</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>1940</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Differentially Expressed G...</td>
      <td>0.236133</td>
      <td>0.265875</td>
      <td>0.997</td>
      <td>0.994275</td>
      <td>1.0</td>
      <td>21/76</td>
      <td>30.60%</td>
      <td>GRM8;R3HDML;CRHR1;CEACAM8;RPL31P49;CASR;UPK1A-...</td>
    </tr>
    <tr>
      <th>1941</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_up__Psoriasis v...</td>
      <td>0.176886</td>
      <td>0.264764</td>
      <td>0.993428</td>
      <td>0.993964</td>
      <td>1.0</td>
      <td>9/10</td>
      <td>78.80%</td>
      <td>CD52;C1QB;NFE2;PDZK1IP1;SERPINB13;C1QC;ASPRV1;...</td>
    </tr>
    <tr>
      <th>1942</th>
      <td>prerank</td>
      <td>DisGeNET__Metabolic Syndrome X</td>
      <td>0.172737</td>
      <td>0.26453</td>
      <td>1.0</td>
      <td>0.993489</td>
      <td>1.0</td>
      <td>3/8</td>
      <td>40.80%</td>
      <td>DUSP9;APOA4;MZB1</td>
    </tr>
    <tr>
      <th>1943</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Down-regulated Genes In HD...</td>
      <td>0.207047</td>
      <td>0.233282</td>
      <td>0.998</td>
      <td>0.997079</td>
      <td>1.0</td>
      <td>20/73</td>
      <td>33.60%</td>
      <td>GRM8;R3HDML;CRHR1;CEACAM8;RPL31P49;CASR;UPK1A-...</td>
    </tr>
    <tr>
      <th>1944</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Down-regulated Genes In HD...</td>
      <td>0.194148</td>
      <td>0.218708</td>
      <td>0.999</td>
      <td>0.997789</td>
      <td>1.0</td>
      <td>54/76</td>
      <td>67.60%</td>
      <td>TEKT1;EVX1-AS;CEACAM8;ALAS2;CASR;LINC01744;RPL...</td>
    </tr>
  </tbody>
</table>
<p>1945 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Pathway Analysis - Early AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">14</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_20&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Disease Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">ValueError</span><span class="g g-Whitespace">                                </span>Traceback (most recent call last)
<span class="n">Cell</span> <span class="n">In</span><span class="p">[</span><span class="mi">142</span><span class="p">],</span> <span class="n">line</span> <span class="mi">2</span>
<span class="g g-Whitespace">      </span><span class="mi">1</span> <span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="ne">----&gt; </span><span class="mi">2</span> <span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">3</span>                  <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">4</span>                  <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Pathway Analysis - Early AT2 Like Cells&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">5</span>                  <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">6</span>                  <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">7</span>                  <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">14</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
<span class="g g-Whitespace">      </span><span class="mi">8</span>                  <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">9</span> <span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="g g-Whitespace">     </span><span class="mi">11</span> <span class="n">sc_plot_and_embed</span><span class="p">(</span>
<span class="g g-Whitespace">     </span><span class="mi">12</span>     <span class="n">ax_dot</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">13</span>     <span class="s2">&quot;gsea_earlyAT2_20&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">14</span>     <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Disease Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">15</span>     <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="g g-Whitespace">     </span><span class="mi">16</span> <span class="p">)</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:1225,</span> in <span class="ni">dotplot</span><span class="nt">(df, column, x, y, x_order, y_order, title, cutoff, top_term, size, ax, figsize, cmap, ofname, xticklabels_rot, yticklabels_rot, marker, show_ring, **kwargs)</span>
<span class="g g-Whitespace">   </span><span class="mi">1222</span>     <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span><span class="s2">&quot;group is deprecated; use x instead&quot;</span><span class="p">,</span> <span class="ne">DeprecationWarning</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1223</span>     <span class="k">return</span>
<span class="ne">-&gt; </span><span class="mi">1225</span> <span class="n">dot</span> <span class="o">=</span> <span class="n">DotPlot</span><span class="p">(</span>
<span class="g g-Whitespace">   </span><span class="mi">1226</span>     <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1227</span>     <span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1228</span>     <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1229</span>     <span class="n">x_order</span><span class="o">=</span><span class="n">x_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1230</span>     <span class="n">y_order</span><span class="o">=</span><span class="n">y_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1231</span>     <span class="n">hue</span><span class="o">=</span><span class="n">column</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1232</span>     <span class="n">title</span><span class="o">=</span><span class="n">title</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1233</span>     <span class="n">thresh</span><span class="o">=</span><span class="n">cutoff</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1234</span>     <span class="n">n_terms</span><span class="o">=</span><span class="nb">int</span><span class="p">(</span><span class="n">top_term</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1235</span>     <span class="n">dot_scale</span><span class="o">=</span><span class="n">size</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1236</span>     <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1237</span>     <span class="n">figsize</span><span class="o">=</span><span class="n">figsize</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1238</span>     <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1239</span>     <span class="n">ofname</span><span class="o">=</span><span class="n">ofname</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1240</span>     <span class="n">marker</span><span class="o">=</span><span class="n">marker</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1241</span> <span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1242</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dot</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">outer_ring</span><span class="o">=</span><span class="n">show_ring</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1244</span> <span class="k">if</span> <span class="n">xticklabels_rot</span><span class="p">:</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:692,</span> in <span class="ni">DotPlot.__init__</span><span class="nt">(self, df, x, y, hue, dot_scale, x_order, y_order, thresh, n_terms, title, ax, figsize, cmap, ofname, marker, **kwargs)</span>
<span class="g g-Whitespace">    </span><span class="mi">690</span> <span class="bp">self</span><span class="o">.</span><span class="n">n_terms</span> <span class="o">=</span> <span class="n">n_terms</span>
<span class="g g-Whitespace">    </span><span class="mi">691</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span> <span class="o">=</span> <span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">692</span> <span class="bp">self</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">693</span> <span class="n">plt</span><span class="o">.</span><span class="n">rcParams</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="s2">&quot;pdf.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">,</span> <span class="s2">&quot;ps.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">})</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:717,</span> in <span class="ni">DotPlot.process</span><span class="nt">(self, df)</span>
<span class="g g-Whitespace">    </span><span class="mi">715</span> <span class="k">if</span> <span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">716</span>     <span class="n">msg</span> <span class="o">=</span> <span class="s2">&quot;Warning: No enrich terms when cutoff = </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">717</span>     <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="n">msg</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">718</span> <span class="bp">self</span><span class="o">.</span><span class="n">cbar_title</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">colname</span>
<span class="g g-Whitespace">    </span><span class="mi">719</span> <span class="c1"># clip GSEA lower bounds</span>
<span class="g g-Whitespace">    </span><span class="mi">720</span> <span class="c1"># if self.colname in [&quot;NOM p-val&quot;, &quot;FDR q-val&quot;]:</span>
<span class="g g-Whitespace">    </span><span class="mi">721</span> <span class="c1">#     df[self.colname].clip(1e-5, 1.0, inplace=True)</span>
<span class="g g-Whitespace">    </span><span class="mi">722</span> <span class="c1"># sorting the dataframe for better visualization</span>

<span class="ne">ValueError</span>: Warning: No enrich terms when cutoff = 0.05
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>


<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_earlyAT2_21&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Disease Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_earlyAT2_21.png" alt="gsea_earlyAT2_21" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Disease Analysis Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_earlyAT2_21.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
</section>
<section id="proliferating-at2-like-cells">
<h2>Proliferating AT2 Like Cells<a class="headerlink" href="#proliferating-at2-like-cells" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">numpy</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">np</span>

<span class="n">deg_results</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({</span>
    <span class="s1">&#39;names&#39;</span><span class="p">:</span> <span class="n">at2_h_lp_filt</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;rank_genes_groups&#39;</span><span class="p">][</span><span class="s1">&#39;names&#39;</span><span class="p">][</span><span class="s1">&#39;Proliferating AT2 Like Cells&#39;</span><span class="p">],</span>    <span class="c1"># Gene names</span>
    <span class="s1">&#39;logfoldchanges&#39;</span><span class="p">:</span> <span class="n">at2_h_lp_filt</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;rank_genes_groups&#39;</span><span class="p">][</span><span class="s1">&#39;logfoldchanges&#39;</span><span class="p">][</span><span class="s1">&#39;Proliferating AT2 Like Cells&#39;</span><span class="p">],</span>     <span class="c1"># Log fold changes</span>
    <span class="s1">&#39;pvals&#39;</span><span class="p">:</span> <span class="n">at2_h_lp_filt</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;rank_genes_groups&#39;</span><span class="p">][</span><span class="s1">&#39;pvals&#39;</span><span class="p">][</span><span class="s1">&#39;Proliferating AT2 Like Cells&#39;</span><span class="p">],</span>          <span class="c1"># p-values</span>
    <span class="s1">&#39;pvals_adj&#39;</span><span class="p">:</span> <span class="n">at2_h_lp_filt</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;rank_genes_groups&#39;</span><span class="p">][</span><span class="s1">&#39;pvals_adj&#39;</span><span class="p">][</span><span class="s1">&#39;Proliferating AT2 Like Cells&#39;</span><span class="p">]</span> <span class="c1"># Adjusted p-values (FDR corrected)</span>
<span class="p">})</span>

<span class="c1"># Step 1: Set thresholds for logFC and p-value (adjusted)</span>
<span class="n">logfc_threshold</span> <span class="o">=</span> <span class="mf">0.01</span>   <span class="c1"># Example: filter for logFC &gt; 1 (or &lt; -1 for downregulated)</span>
<span class="n">pval_adj_threshold</span> <span class="o">=</span> <span class="mf">0.05</span>  <span class="c1"># Example: adjusted p-value &lt; 0.05</span>

<span class="c1"># Step 2: Apply the filtering based on logFC and adjusted p-value</span>
<span class="n">filtered_genes</span> <span class="o">=</span> <span class="n">deg_results</span><span class="p">[</span>
    <span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">abs</span><span class="p">(</span><span class="n">deg_results</span><span class="p">[</span><span class="s1">&#39;logfoldchanges&#39;</span><span class="p">])</span> <span class="o">&gt;=</span> <span class="n">logfc_threshold</span><span class="p">)</span> <span class="o">&amp;</span>  <span class="c1"># Absolute logFC above threshold</span>
    <span class="p">(</span><span class="n">deg_results</span><span class="p">[</span><span class="s1">&#39;pvals_adj&#39;</span><span class="p">]</span> <span class="o">&lt;</span> <span class="n">pval_adj_threshold</span><span class="p">)</span>               <span class="c1"># Adjusted p-value below threshold</span>
<span class="p">]</span>

<span class="c1"># Step 3: Extract the list of filtered gene names</span>
<span class="n">filtered_gene_names</span> <span class="o">=</span> <span class="n">filtered_genes</span><span class="p">[</span><span class="s1">&#39;names&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
<span class="c1">#filtered_gene_names[:50]</span>

<span class="c1"># Run g:Profiler</span>
<span class="n">go_results</span> <span class="o">=</span> <span class="n">gp</span><span class="o">.</span><span class="n">profile</span><span class="p">(</span><span class="n">organism</span><span class="o">=</span><span class="s1">&#39;hsapiens&#39;</span><span class="p">,</span> <span class="n">query</span><span class="o">=</span><span class="n">filtered_gene_names</span><span class="p">[:</span><span class="mi">250</span><span class="p">])</span>

<span class="c1"># Define the sources you want to loop over</span>
<span class="n">sources</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;GO:BP&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:MF&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:CC&#39;</span><span class="p">,</span> <span class="s1">&#39;KEGG&#39;</span><span class="p">,</span> <span class="s1">&#39;REAC&#39;</span><span class="p">,</span> <span class="s1">&#39;HPA&#39;</span><span class="p">,</span> <span class="s1">&#39;WP&#39;</span><span class="p">]</span>

<span class="c1"># Create a stacked figure with one row per source</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">nrows</span><span class="o">=</span><span class="nb">len</span><span class="p">(</span><span class="n">sources</span><span class="p">),</span> <span class="n">ncols</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="p">,</span> <span class="mf">2.5</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">sources</span><span class="p">)))</span>
<span class="n">axes</span> <span class="o">=</span> <span class="n">axes</span><span class="o">.</span><span class="n">flatten</span><span class="p">()</span>

<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">source</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">sources</span><span class="p">):</span>
    <span class="c1"># Filter results for this source</span>
    <span class="n">specific_results</span> <span class="o">=</span> <span class="n">go_results</span><span class="p">[</span><span class="n">go_results</span><span class="p">[</span><span class="s1">&#39;source&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">source</span><span class="p">]</span>
    
    <span class="k">if</span> <span class="n">specific_results</span><span class="o">.</span><span class="n">empty</span><span class="p">:</span>
        <span class="k">continue</span>  <span class="c1"># skip if no results for this source</span>
    
    <span class="c1"># Select top 8–10 terms</span>
    <span class="n">go_results_top</span> <span class="o">=</span> <span class="p">(</span>
        <span class="n">specific_results</span><span class="p">[[</span><span class="s1">&#39;name&#39;</span><span class="p">,</span> <span class="s1">&#39;p_value&#39;</span><span class="p">]]</span>
        <span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;p_value&#39;</span><span class="p">)</span>
        <span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">8</span><span class="p">)</span>
    <span class="p">)</span>

    <span class="n">labels</span> <span class="o">=</span> <span class="n">go_results_top</span><span class="p">[</span><span class="s1">&#39;name&#39;</span><span class="p">]</span>
    <span class="n">values</span> <span class="o">=</span> <span class="o">-</span><span class="n">np</span><span class="o">.</span><span class="n">log10</span><span class="p">(</span><span class="n">go_results_top</span><span class="p">[</span><span class="s1">&#39;p_value&#39;</span><span class="p">])</span>

    <span class="n">ax</span> <span class="o">=</span> <span class="n">axes</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">barh</span><span class="p">(</span><span class="n">labels</span><span class="p">,</span> <span class="n">values</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s2">&quot;#1f77b4&quot;</span><span class="p">)</span>
    
    <span class="c1"># Aesthetics</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">source</span><span class="si">}</span><span class="s2"> enrichment&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span> <span class="n">loc</span><span class="o">=</span><span class="s2">&quot;left&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;-log10(P-value)&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">invert_yaxis</span><span class="p">()</span>  <span class="c1"># highest term on top</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="s2">&quot;y&quot;</span><span class="p">,</span> <span class="n">labelsize</span><span class="o">=</span><span class="mi">9</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="s2">&quot;x&quot;</span><span class="p">,</span> <span class="n">labelsize</span><span class="o">=</span><span class="mi">9</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s2">&quot;top&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="kc">False</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s2">&quot;right&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="kc">False</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">grid</span><span class="p">(</span><span class="kc">False</span><span class="p">)</span>
<span class="c1"># Adjust layout</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="c1"># Save if needed</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prolifAT2_1&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GProfiler Analysis of top 250 differentially expressed genes from Proliferating AT2 Like Cells.&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_prolifAT2_1.png" alt="gsea_prolifAT2_1" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GProfiler Analysis of top 250 differentially expressed genes from Proliferating AT2 Like Cells.</em><br>
        <a href="_static/Figures/gsea_prolifAT2_1.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>

<span class="n">group</span> <span class="o">=</span> <span class="s2">&quot;Proliferating AT2 Like Cells&quot;</span>  <span class="c1"># change for each group</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">get</span><span class="o">.</span><span class="n">rank_genes_groups_df</span><span class="p">(</span><span class="n">at2_h_lp_filt</span><span class="p">,</span> <span class="n">group</span><span class="o">=</span><span class="n">group</span><span class="p">)</span>

<span class="c1"># GSEAPY prerank needs index=gene, score=stat/logFC</span>
<span class="n">ranked_list</span> <span class="o">=</span> <span class="n">df</span><span class="p">[[</span><span class="s1">&#39;names&#39;</span><span class="p">,</span> <span class="s1">&#39;scores&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;names&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">squeeze</span><span class="p">()</span>

<span class="n">df_ranked_list_top</span> <span class="o">=</span> <span class="n">ranked_list</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">500</span><span class="p">]</span>

<span class="n">ranked_list_pos</span> <span class="o">=</span> <span class="n">ranked_list</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">250</span><span class="p">]</span>
<span class="n">ranked_list_neg</span> <span class="o">=</span> <span class="n">ranked_list</span><span class="p">[(</span><span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">)</span><span class="o">-</span><span class="mi">250</span><span class="p">):</span><span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">)]</span>

<span class="n">ranked_lists</span> <span class="o">=</span> <span class="p">[</span><span class="n">ranked_list_pos</span><span class="p">,</span><span class="n">ranked_list_neg</span><span class="p">]</span>

<span class="n">df_ranked_list</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">ranked_lists</span><span class="p">)</span>
<span class="c1"># Save to .rnk if needed</span>
<span class="c1">#ranked_list.to_csv(&quot;lateAT2.rnk&quot;, sep=&quot;\t&quot;)</span>
<span class="n">df_ranked_list</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>names
MIR924HG    107.935020
TOP2A        92.992737
MKI67        92.342407
DIAPH3       90.044273
HELLS        84.477745
               ...    
NEDD4L      -23.344038
SFTPB       -23.441755
TALAM1      -25.296331
GPC5        -27.519966
NEAT1       -44.600460
Name: scores, Length: 500, dtype: float32
</pre></div>
</div>
</div>
</div>
<section id="id11">
<h3>GSEA - Cell Type [Top 250 &amp; Down 250]<a class="headerlink" href="#id11" title="Link to this heading">#</a></h3>
<p>‘Azimuth_2023’,’CellMarker_2024’,’Descartes_Cell_Types_and_Tissue_2021’,’Human_Gene_Atlas’,’Human_Phenotype_Ontology’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Azimuth_2023&#39;</span><span class="p">,</span><span class="s1">&#39;CellMarker_2024&#39;</span><span class="p">,</span><span class="s1">&#39;Descartes_Cell_Types_and_Tissue_2021&#39;</span><span class="p">,</span><span class="s1">&#39;Human_Gene_Atlas&#39;</span><span class="p">,</span><span class="s1">&#39;Human_Phenotype_Ontology&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>Descartes_Cell_Types_and_Tissue_2021__Corneal ...</td>
      <td>-0.714051</td>
      <td>-2.995882</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>12/24</td>
      <td>25.00%</td>
      <td>GPRC5A;CEACAM6;DSP;RAB11FIP1;GDF15;CD55;KRT7;C...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>CellMarker_2024__Krt4/13+ Cell Trachea Mouse</td>
      <td>-0.849721</td>
      <td>-2.781191</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>6/16</td>
      <td>7.80%</td>
      <td>LMO7;MYOF;ALCAM;GPRC5A;RAB11FIP1;ANXA2</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>Human_Gene_Atlas__Lung</td>
      <td>-0.613217</td>
      <td>-2.314559</td>
      <td>0.0</td>
      <td>0.001404</td>
      <td>0.001001</td>
      <td>8/19</td>
      <td>9.40%</td>
      <td>SFTPB;SPTBN1;TACC2;AKAP13;NCOR2;GPRC5A;CEACAM6...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>Human_Gene_Atlas__Placenta</td>
      <td>-0.787369</td>
      <td>-2.129849</td>
      <td>0.0</td>
      <td>0.011586</td>
      <td>0.011011</td>
      <td>8/9</td>
      <td>22.20%</td>
      <td>TACC2;EZR;RBMS1;FNDC3B;CD55;SYNPO;PAPPA;FSTL3</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>CellMarker_2024__Microglial Cell Brain Human</td>
      <td>-0.82268</td>
      <td>-2.122885</td>
      <td>0.026316</td>
      <td>0.010111</td>
      <td>0.012012</td>
      <td>4/8</td>
      <td>12.80%</td>
      <td>TPT1;EMP2;EEF1A1;KIAA1217</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>279</th>
      <td>prerank</td>
      <td>CellMarker_2024__Sertoli Cell Fetal Gonad Human</td>
      <td>0.286854</td>
      <td>0.589659</td>
      <td>0.998999</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>5/24</td>
      <td>45.40%</td>
      <td>SYNE2;TUBB4B;RAD21;FAM111A;RBBP7</td>
    </tr>
    <tr>
      <th>280</th>
      <td>prerank</td>
      <td>Human_Gene_Atlas__BronchialEpithelialCells</td>
      <td>0.29233</td>
      <td>0.567925</td>
      <td>0.996957</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>2/13</td>
      <td>22.00%</td>
      <td>HMGA2;RTTN</td>
    </tr>
    <tr>
      <th>281</th>
      <td>prerank</td>
      <td>Human_Phenotype_Ontology__Abnormality of the g...</td>
      <td>0.324751</td>
      <td>0.558105</td>
      <td>0.991507</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>2/6</td>
      <td>38.60%</td>
      <td>BAZ1B;LBR</td>
    </tr>
    <tr>
      <th>282</th>
      <td>prerank</td>
      <td>Human_Gene_Atlas__Thyroid</td>
      <td>0.260575</td>
      <td>0.497971</td>
      <td>0.996936</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>1/11</td>
      <td>12.00%</td>
      <td>WWOX</td>
    </tr>
    <tr>
      <th>283</th>
      <td>prerank</td>
      <td>CellMarker_2024__Monocyte Fetal Kidney Human</td>
      <td>0.207315</td>
      <td>0.420813</td>
      <td>1.0</td>
      <td>0.999842</td>
      <td>1.0</td>
      <td>3/19</td>
      <td>46.20%</td>
      <td>TACC3;LBR;SRSF3</td>
    </tr>
  </tbody>
</table>
<p>284 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Cell Type - Proliferating AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prolifAT2_2&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Cell Type Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../_images/e67c63bb5e3645a9e2717b96595b1b6a7b035b8b78f2695bae804f49709697cb.png"><img alt="../_images/e67c63bb5e3645a9e2717b96595b1b6a7b035b8b78f2695bae804f49709697cb.png" src="../_images/e67c63bb5e3645a9e2717b96595b1b6a7b035b8b78f2695bae804f49709697cb.png" style="width: 4277px; height: 1466px;" />
</a>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prlolifAT2_3&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Cell Type Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_prlolifAT2_3.png" alt="gsea_prlolifAT2_3" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Cell Type Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_prlolifAT2_3.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id12">
<h3>GSEA - Cell Type [Top 500]<a class="headerlink" href="#id12" title="Link to this heading">#</a></h3>
<p>‘Azimuth_2023’,’CellMarker_2024’,’Descartes_Cell_Types_and_Tissue_2021’,’Human_Gene_Atlas’,’Human_Phenotype_Ontology’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list_top</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Azimuth_2023&#39;</span><span class="p">,</span><span class="s1">&#39;CellMarker_2024&#39;</span><span class="p">,</span><span class="s1">&#39;Descartes_Cell_Types_and_Tissue_2021&#39;</span><span class="p">,</span><span class="s1">&#39;Human_Gene_Atlas&#39;</span><span class="p">,</span><span class="s1">&#39;Human_Phenotype_Ontology&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>CellMarker_2024__Cycling Basal Cell Trachea Mouse</td>
      <td>0.485479</td>
      <td>3.219461</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>68/92</td>
      <td>40.60%</td>
      <td>TOP2A;MKI67;DIAPH3;ATAD2;RRM2;SMC4;BUB1B;TUBB;...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>CellMarker_2024__Fibroblast Lung Mouse</td>
      <td>0.611781</td>
      <td>3.142706</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>24/34</td>
      <td>22.60%</td>
      <td>TOP2A;MKI67;ATAD2;RRM2;TUBB;CENPF;TUBA1B;KIF20...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>CellMarker_2024__Neural Progenitor Cell Embryo...</td>
      <td>0.45998</td>
      <td>2.858293</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>43/65</td>
      <td>34.60%</td>
      <td>TOP2A;MKI67;HELLS;ATAD2;RRM2;SMC4;CENPF;TUBA1B...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>Azimuth_2023__Tonsil-L2-MBC-like Proli1</td>
      <td>0.851626</td>
      <td>2.590937</td>
      <td>0.0</td>
      <td>0.000275</td>
      <td>0.001</td>
      <td>8/8</td>
      <td>16.20%</td>
      <td>TUBB;TUBA1B;PTMA;HMGB1;HMGN2;HMGB2;NUSAP1;PCLAF</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>CellMarker_2024__MKI67+ Progenitor Cell Large ...</td>
      <td>0.600548</td>
      <td>2.506246</td>
      <td>0.0</td>
      <td>0.000659</td>
      <td>0.003</td>
      <td>11/18</td>
      <td>14.40%</td>
      <td>TOP2A;MKI67;RRM2;CENPF;ASPM;ANLN;KIF20B;PRC1;T...</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>233</th>
      <td>prerank</td>
      <td>CellMarker_2024__Alpha Cell Pancreatic Islet H...</td>
      <td>0.216162</td>
      <td>0.552037</td>
      <td>0.962575</td>
      <td>0.964548</td>
      <td>1.0</td>
      <td>5/5</td>
      <td>78.60%</td>
      <td>HMGB2;SLF2;HNRNPA3;UBA2;SRBD1</td>
    </tr>
    <tr>
      <th>234</th>
      <td>prerank</td>
      <td>Human_Gene_Atlas__AtrioventricularNode</td>
      <td>-0.169805</td>
      <td>-0.549935</td>
      <td>0.967213</td>
      <td>0.992328</td>
      <td>1.0</td>
      <td>3/7</td>
      <td>19.20%</td>
      <td>EPHA6;SP3;KIAA1328</td>
    </tr>
    <tr>
      <th>235</th>
      <td>prerank</td>
      <td>Human_Phenotype_Ontology__Camptodactyly of fin...</td>
      <td>0.212027</td>
      <td>0.54301</td>
      <td>0.970451</td>
      <td>0.964462</td>
      <td>1.0</td>
      <td>1/5</td>
      <td>8.40%</td>
      <td>EZH2</td>
    </tr>
    <tr>
      <th>236</th>
      <td>prerank</td>
      <td>CellMarker_2024__Gonadal Endothelial Cell Feta...</td>
      <td>0.107613</td>
      <td>0.523107</td>
      <td>0.957993</td>
      <td>0.969102</td>
      <td>1.0</td>
      <td>12/29</td>
      <td>39.40%</td>
      <td>RFC3;MMS22L;DNMT1;HMGA1;HMGA2;BUB1;RGS3;SGO2;G...</td>
    </tr>
    <tr>
      <th>237</th>
      <td>prerank</td>
      <td>CellMarker_2024__Multilymphoid Progenitor Cell...</td>
      <td>-0.121568</td>
      <td>-0.406523</td>
      <td>0.996795</td>
      <td>0.999293</td>
      <td>1.0</td>
      <td>3/7</td>
      <td>24.40%</td>
      <td>ATM;FCHSD2;HNRNPUL1</td>
    </tr>
  </tbody>
</table>
<p>238 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Cell Type - Proliferating AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">6</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prolifAT2_4&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Cell Type Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_prolifAT2_4.png" alt="gsea_prolifAT2_4" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Cell Type Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_prolifAT2_4.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prlolifAT2_5&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Cell Type Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_prlolifAT2_5.png" alt="gsea_prlolifAT2_5" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Cell Type Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_prlolifAT2_5.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id13">
<h3>GSEA - Gene Ontology [Top 250 &amp; Down 250]<a class="headerlink" href="#id13" title="Link to this heading">#</a></h3>
<p>‘GO_Biological_Process_2025’,’GO_Cellular_Component_2025’,’GO_Molecular_Function_2025’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;GO_Biological_Process_2025&#39;</span><span class="p">,</span><span class="s1">&#39;GO_Cellular_Component_2025&#39;</span><span class="p">,</span><span class="s1">&#39;GO_Molecular_Function_2025&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>GO_Cellular_Component_2025__Late Endosome (GO:...</td>
      <td>-0.834855</td>
      <td>-2.252412</td>
      <td>0.0</td>
      <td>0.011388</td>
      <td>0.005</td>
      <td>7/10</td>
      <td>16.80%</td>
      <td>SFTPB;TPT1;ANXA2;PARM1;NPC1;ATP13A4;IGF2R</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>GO_Cellular_Component_2025__Bounding Membrane ...</td>
      <td>-0.764337</td>
      <td>-2.145289</td>
      <td>0.0</td>
      <td>0.016134</td>
      <td>0.016</td>
      <td>5/11</td>
      <td>18.00%</td>
      <td>GOLGB1;CD55;PTPRN2;IGF2R;CHSY3</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>GO_Cellular_Component_2025__Cortical Cytoskele...</td>
      <td>-0.878543</td>
      <td>-1.854336</td>
      <td>0.053571</td>
      <td>0.158172</td>
      <td>0.189</td>
      <td>6/6</td>
      <td>13.40%</td>
      <td>SPTBN1;AKAP13;EZR;EEF1A1;DLC1;SHROOM3</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Protein Localizati...</td>
      <td>-0.86189</td>
      <td>-1.782173</td>
      <td>0.031746</td>
      <td>0.155262</td>
      <td>0.289</td>
      <td>4/6</td>
      <td>9.60%</td>
      <td>SPTBN1;ATP1B1;EMP2;EZR</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Protein Localizati...</td>
      <td>-0.86189</td>
      <td>-1.782173</td>
      <td>0.031746</td>
      <td>0.155262</td>
      <td>0.289</td>
      <td>4/6</td>
      <td>9.60%</td>
      <td>SPTBN1;ATP1B1;EMP2;EZR</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>399</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Ubiquitin-Dependen...</td>
      <td>0.302425</td>
      <td>0.541186</td>
      <td>0.991579</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>3/7</td>
      <td>47.20%</td>
      <td>DTL;FAF1;KIF14</td>
    </tr>
    <tr>
      <th>400</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Regulation of Anat...</td>
      <td>0.312009</td>
      <td>0.518415</td>
      <td>0.997824</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>2/5</td>
      <td>40.80%</td>
      <td>MYO19;PDE3B</td>
    </tr>
    <tr>
      <th>401</th>
      <td>prerank</td>
      <td>GO_Molecular_Function_2025__Protein Kinase Act...</td>
      <td>0.278494</td>
      <td>0.493993</td>
      <td>0.998938</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>3/7</td>
      <td>49.60%</td>
      <td>TOPBP1;PDE8A;MAP3K20</td>
    </tr>
    <tr>
      <th>402</th>
      <td>prerank</td>
      <td>GO_Molecular_Function_2025__GTPase Binding (GO...</td>
      <td>0.259827</td>
      <td>0.481384</td>
      <td>0.998971</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>4/9</td>
      <td>48.00%</td>
      <td>KNTC1;IARS1;RANBP1;HSP90AA1</td>
    </tr>
    <tr>
      <th>403</th>
      <td>prerank</td>
      <td>GO_Molecular_Function_2025__Small GTPase Bindi...</td>
      <td>0.233668</td>
      <td>0.404873</td>
      <td>0.998942</td>
      <td>0.999927</td>
      <td>1.0</td>
      <td>2/6</td>
      <td>45.20%</td>
      <td>KNTC1;RANBP1</td>
    </tr>
  </tbody>
</table>
<p>404 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Gene Ontology - Proliferating AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prolifAT2_6&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Gene Ontology Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_prolifAT2_6.png" alt="gsea_prolifAT2_6" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Gene Ontology Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_prolifAT2_6.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prlolifAT2_7&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Gene Ontology Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_prlolifAT2_7.png" alt="gsea_prlolifAT2_7" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Gene Ontology Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_prlolifAT2_7.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id14">
<h3>GSEA - Gene Ontology [Top 500]<a class="headerlink" href="#id14" title="Link to this heading">#</a></h3>
<p>‘GO_Biological_Process_2025’,’GO_Cellular_Component_2025’,’GO_Molecular_Function_2025’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list_top</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;GO_Biological_Process_2025&#39;</span><span class="p">,</span><span class="s1">&#39;GO_Cellular_Component_2025&#39;</span><span class="p">,</span><span class="s1">&#39;GO_Molecular_Function_2025&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>GO_Cellular_Component_2025__Bounding Membrane ...</td>
      <td>-0.751012</td>
      <td>-2.30301</td>
      <td>0.0</td>
      <td>0.117858</td>
      <td>0.096</td>
      <td>6/6</td>
      <td>26.00%</td>
      <td>RNF144A;ST6GALNAC3;DIAPH1;SORL1;CTSC;HS6ST2</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__DNA Metabolic Proc...</td>
      <td>0.367327</td>
      <td>2.203292</td>
      <td>0.0</td>
      <td>0.077985</td>
      <td>0.066</td>
      <td>41/65</td>
      <td>38.40%</td>
      <td>TOP2A;BRIP1;FAM111B;POLA1;PRKDC;RFC3;PRIM2;RRM...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Chromosome Condens...</td>
      <td>0.615734</td>
      <td>2.194191</td>
      <td>0.0</td>
      <td>0.040473</td>
      <td>0.069</td>
      <td>10/12</td>
      <td>28.80%</td>
      <td>TOP2A;SMC4;H1-4;NUSAP1;H1-5;H1-2;H1-3;NCAPG;NC...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Positive Regulatio...</td>
      <td>0.656831</td>
      <td>2.112622</td>
      <td>0.001412</td>
      <td>0.070417</td>
      <td>0.159</td>
      <td>7/9</td>
      <td>19.80%</td>
      <td>EZH2;CLSPN;TPX2;ECT2;CENPE;XRCC5;HMGA2</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>GO_Molecular_Function_2025__DNA Binding, Bendi...</td>
      <td>0.810101</td>
      <td>2.039775</td>
      <td>0.001513</td>
      <td>0.110314</td>
      <td>0.293</td>
      <td>5/5</td>
      <td>19.80%</td>
      <td>TOP2A;HMGB1;HMGB2;HMGA1;HMGA2</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>466</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Protein Modificati...</td>
      <td>-0.093634</td>
      <td>-0.459156</td>
      <td>1.0</td>
      <td>0.996312</td>
      <td>1.0</td>
      <td>12/16</td>
      <td>58.60%</td>
      <td>ZNF367;ATM;SYK;CCNA2;PPM1G;THBS1;PLK4;PRKD3;MA...</td>
    </tr>
    <tr>
      <th>467</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Regulation of Intr...</td>
      <td>0.14164</td>
      <td>0.452378</td>
      <td>0.993094</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>5/9</td>
      <td>48.20%</td>
      <td>ECT2;ARHGAP11A;PDE3B;CHEK1;RACGAP1</td>
    </tr>
    <tr>
      <th>468</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Negative Regulatio...</td>
      <td>0.151109</td>
      <td>0.448359</td>
      <td>0.994135</td>
      <td>0.998373</td>
      <td>1.0</td>
      <td>4/7</td>
      <td>49.60%</td>
      <td>EZH2;RTKN2;BCL2;MAP3K20</td>
    </tr>
    <tr>
      <th>469</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Positive Regulatio...</td>
      <td>0.151576</td>
      <td>0.436942</td>
      <td>0.99701</td>
      <td>0.997144</td>
      <td>1.0</td>
      <td>1/7</td>
      <td>7.20%</td>
      <td>KIF20B</td>
    </tr>
    <tr>
      <th>470</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Nucleus Organizati...</td>
      <td>0.148189</td>
      <td>0.424669</td>
      <td>0.992918</td>
      <td>0.995709</td>
      <td>1.0</td>
      <td>5/7</td>
      <td>62.40%</td>
      <td>LMNB1;NUP205;NUP155;CEP55;NUP54</td>
    </tr>
  </tbody>
</table>
<p>471 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Gene Ontology - Proliferating AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">13</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prolifAT2_8&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Gene Ontology Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_prolifAT2_8.png" alt="gsea_prolifAT2_8" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Gene Ontology Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_prolifAT2_8.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prlolifAT2_9&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Gene Ontology Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_prlolifAT2_9.png" alt="gsea_prlolifAT2_9" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Gene Ontology Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_prlolifAT2_9.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id15">
<h3>GSEA - Pathway Analysis [Top 250 &amp; Down 250]<a class="headerlink" href="#id15" title="Link to this heading">#</a></h3>
<p>‘Reactome_Pathways_2024’,’KEGG_2021_Human’,’WikiPathways_2024_Human’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Reactome_Pathways_2024&#39;</span><span class="p">,</span><span class="s1">&#39;KEGG_2021_Human&#39;</span><span class="p">,</span><span class="s1">&#39;WikiPathways_2024_Human&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Transport of Small Mol...</td>
      <td>-0.824514</td>
      <td>-2.264825</td>
      <td>0.0</td>
      <td>0.004731</td>
      <td>0.002</td>
      <td>8/12</td>
      <td>17.00%</td>
      <td>NEDD4L;ATP1B1;ABCA1;NPC1;RUNX1;ATP1A1;ATP13A4;...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Muscle Contraction</td>
      <td>-0.790402</td>
      <td>-1.986282</td>
      <td>0.032258</td>
      <td>0.059931</td>
      <td>0.058</td>
      <td>5/8</td>
      <td>19.60%</td>
      <td>ATP1B1;ANXA2;ATP1A1;VIM;AKAP9</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__GPCR Ligand Binding</td>
      <td>-0.805044</td>
      <td>-1.692623</td>
      <td>0.061538</td>
      <td>0.442123</td>
      <td>0.403</td>
      <td>2/6</td>
      <td>14.40%</td>
      <td>GPR39;CD55</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Antigen Processing Ubi...</td>
      <td>-0.891271</td>
      <td>-1.645598</td>
      <td>0.053097</td>
      <td>0.466831</td>
      <td>0.492</td>
      <td>4/5</td>
      <td>11.00%</td>
      <td>NEDD4L;LMO7;UBE2H;WSB1</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>WikiPathways_2024_Human__Pancreatic Cancer Sub...</td>
      <td>-0.728852</td>
      <td>-1.527968</td>
      <td>0.115385</td>
      <td>0.714756</td>
      <td>0.695</td>
      <td>2/6</td>
      <td>17.40%</td>
      <td>CEACAM6;KRT7</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>452</th>
      <td>prerank</td>
      <td>KEGG_2021_Human__Human papillomavirus infection</td>
      <td>0.351674</td>
      <td>0.607308</td>
      <td>0.982086</td>
      <td>0.993838</td>
      <td>1.0</td>
      <td>3/6</td>
      <td>49.00%</td>
      <td>RBL1;TCF7L1;ITGA2</td>
    </tr>
    <tr>
      <th>453</th>
      <td>prerank</td>
      <td>KEGG_2021_Human__Epstein-Barr virus infection</td>
      <td>0.305433</td>
      <td>0.558103</td>
      <td>0.987616</td>
      <td>0.999914</td>
      <td>1.0</td>
      <td>2/8</td>
      <td>36.40%</td>
      <td>E2F3;BCL2</td>
    </tr>
    <tr>
      <th>454</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Epigenetic Regulation ...</td>
      <td>0.30449</td>
      <td>0.542434</td>
      <td>0.993724</td>
      <td>0.999294</td>
      <td>1.0</td>
      <td>2/7</td>
      <td>48.80%</td>
      <td>H2AZ2;PSIP1</td>
    </tr>
    <tr>
      <th>455</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Transcriptional Activi...</td>
      <td>0.293124</td>
      <td>0.510242</td>
      <td>0.996774</td>
      <td>0.999444</td>
      <td>1.0</td>
      <td>3/6</td>
      <td>47.40%</td>
      <td>RBL1;PARP1;TFDP1</td>
    </tr>
    <tr>
      <th>456</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Bacterial Infection Pa...</td>
      <td>0.225096</td>
      <td>0.389116</td>
      <td>1.0</td>
      <td>0.99995</td>
      <td>1.0</td>
      <td>1/6</td>
      <td>17.20%</td>
      <td>KPNB1</td>
    </tr>
  </tbody>
</table>
<p>457 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Pathway Analysis - Proliferating AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prolifAT2_10&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Pathway Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_prolifAT2_10.png" alt="gsea_prolifAT2_10" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Pathway Analysis Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_prolifAT2_10.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prlolifAT2_11&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Pathway Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_prlolifAT2_11.png" alt="gsea_prlolifAT2_11" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Pathway Analysis Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_prlolifAT2_11.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id16">
<h3>GSEA - Pathway Analysis [Top 500]<a class="headerlink" href="#id16" title="Link to this heading">#</a></h3>
<p>‘Reactome_Pathways_2024’,’KEGG_2021_Human’,’WikiPathways_2024_Human’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list_top</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Reactome_Pathways_2024&#39;</span><span class="p">,</span><span class="s1">&#39;KEGG_2021_Human&#39;</span><span class="p">,</span><span class="s1">&#39;WikiPathways_2024_Human&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Apoptotic Execution Phase</td>
      <td>0.830957</td>
      <td>2.632532</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>9/9</td>
      <td>18.40%</td>
      <td>H1-4;LMNB1;CLSPN;HMGB1;HMGB2;H1-5;KPNB1;H1-2;H1-3</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>WikiPathways_2024_Human__Retinoblastoma Gene I...</td>
      <td>0.451619</td>
      <td>2.458866</td>
      <td>0.0</td>
      <td>0.0036</td>
      <td>0.003</td>
      <td>25/41</td>
      <td>29.00%</td>
      <td>TOP2A;RRM2;POLA1;PRKDC;RFC3;ANLN;RRM1;DNMT1;SM...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Metabolism of RNA</td>
      <td>-0.272074</td>
      <td>-2.419223</td>
      <td>0.0</td>
      <td>0.015613</td>
      <td>0.014</td>
      <td>54/57</td>
      <td>69.60%</td>
      <td>WDR43;DHX15;SNRNP200;NDC1;IGF2BP3;CSTF3;SF1;PP...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Apoptosis Induced DNA ...</td>
      <td>0.827586</td>
      <td>2.388774</td>
      <td>0.0</td>
      <td>0.004115</td>
      <td>0.008</td>
      <td>7/7</td>
      <td>18.40%</td>
      <td>H1-4;HMGB1;HMGB2;H1-5;KPNB1;H1-2;H1-3</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Apoptosis</td>
      <td>0.693247</td>
      <td>2.382348</td>
      <td>0.0</td>
      <td>0.003343</td>
      <td>0.009</td>
      <td>9/11</td>
      <td>18.40%</td>
      <td>H1-4;LMNB1;CLSPN;HMGB1;HMGB2;H1-5;KPNB1;H1-2;H1-3</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>456</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Regulation of Endogeno...</td>
      <td>0.158935</td>
      <td>0.487942</td>
      <td>0.987821</td>
      <td>0.992973</td>
      <td>1.0</td>
      <td>5/8</td>
      <td>53.20%</td>
      <td>CBX5;H2AZ2;RBBP7;RBBP4;ARID1A</td>
    </tr>
    <tr>
      <th>457</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__SARS-CoV Infections</td>
      <td>0.100827</td>
      <td>0.480998</td>
      <td>0.970964</td>
      <td>0.992366</td>
      <td>1.0</td>
      <td>2/28</td>
      <td>3.20%</td>
      <td>TUBB;GPC6</td>
    </tr>
    <tr>
      <th>458</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Metabolism</td>
      <td>0.078187</td>
      <td>0.455181</td>
      <td>0.97754</td>
      <td>0.995875</td>
      <td>1.0</td>
      <td>3/55</td>
      <td>3.20%</td>
      <td>BRIP1;RRM2;GPC6</td>
    </tr>
    <tr>
      <th>459</th>
      <td>prerank</td>
      <td>WikiPathways_2024_Human__lncRNA In Canonical W...</td>
      <td>0.164153</td>
      <td>0.45056</td>
      <td>0.992888</td>
      <td>0.993899</td>
      <td>1.0</td>
      <td>1/6</td>
      <td>8.40%</td>
      <td>EZH2</td>
    </tr>
    <tr>
      <th>460</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Axon Guidance</td>
      <td>0.118666</td>
      <td>0.439801</td>
      <td>0.992453</td>
      <td>0.992943</td>
      <td>1.0</td>
      <td>4/14</td>
      <td>26.40%</td>
      <td>TUBA1B;KIF4A;TUBB4B;MYH10</td>
    </tr>
  </tbody>
</table>
<p>461 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Pathway Analysis - Proliferating AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prolifAT2_12&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Pathway Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_prolifAT2_12.png" alt="gsea_prolifAT2_12" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Pathway Analysis Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_prolifAT2_12.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prlolifAT2_13&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Pathway Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_prlolifAT2_13.png" alt="gsea_prlolifAT2_13" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Pathway Analysis Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_prlolifAT2_13.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id17">
<h3>GSEA - Proteomic Analysis [Top 250 &amp; Down 250]<a class="headerlink" href="#id17" title="Link to this heading">#</a></h3>
<p>‘ProteomicsDB_2020’,’Tissue_Protein_Expression_from_Human_Proteome_Map’,’Tissue_Protein_Expression_from_ProteomicsDB’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;ProteomicsDB_2020&#39;</span><span class="p">,</span><span class="s1">&#39;Tissue_Protein_Expression_from_Human_Proteome_Map&#39;</span><span class="p">,</span><span class="s1">&#39;Tissue_Protein_Expression_from_ProteomicsDB&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Cecum LS1034 PDB:200120 LS1...</td>
      <td>-0.591037</td>
      <td>-2.038605</td>
      <td>0.0</td>
      <td>0.112314</td>
      <td>0.053106</td>
      <td>11/16</td>
      <td>14.60%</td>
      <td>LMO7;GLS;CDH1;GPRC5A;ATP1B1;CTNNA1;EVI5;TRIM2;...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Cecum C84 PDB:200069 C84 CRC65</td>
      <td>-0.594225</td>
      <td>-1.882992</td>
      <td>0.0</td>
      <td>0.222105</td>
      <td>0.172345</td>
      <td>9/14</td>
      <td>12.20%</td>
      <td>LMO7;CDH1;MAGI3;TACC2;ATP1B1;CEACAM6;CTNNA1;TR...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Colon HDC-111 PDB:201056 HD...</td>
      <td>-0.446606</td>
      <td>-1.8592</td>
      <td>0.0</td>
      <td>0.175412</td>
      <td>0.191383</td>
      <td>10/18</td>
      <td>12.80%</td>
      <td>LMO7;CDH1;MAGI3;TACC2;GOLGA4;ATP1B1;CTNNA1;IFT...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_ProteomicsDB__t...</td>
      <td>-0.895</td>
      <td>-1.792181</td>
      <td>0.015873</td>
      <td>0.210116</td>
      <td>0.270541</td>
      <td>4/7</td>
      <td>10.00%</td>
      <td>SFTPB;LMO7;LIMCH1;UNC13B</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Colon HRA-19 PDB:201059 HRA...</td>
      <td>-0.587018</td>
      <td>-1.758903</td>
      <td>0.0</td>
      <td>0.194846</td>
      <td>0.311623</td>
      <td>12/15</td>
      <td>15.80%</td>
      <td>LMO7;CDH1;MAGI3;TACC2;ATP1B1;CTNNA1;TRIM2;PAM;...</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>991</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Brain SF-295 BTO:0004213 P0...</td>
      <td>0.269892</td>
      <td>0.52545</td>
      <td>0.997976</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>5/12</td>
      <td>49.00%</td>
      <td>KIF4A;BARD1;PCNA;NUP155;ITGA2</td>
    </tr>
    <tr>
      <th>992</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Prostate DU-145 BTO:0001332...</td>
      <td>0.258413</td>
      <td>0.520867</td>
      <td>0.997994</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>5/17</td>
      <td>48.60%</td>
      <td>BRCA2;BARD1;CEP152;NUP155;ANP32E</td>
    </tr>
    <tr>
      <th>993</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Lung HOP-92 BTO:0004085 P00...</td>
      <td>0.258485</td>
      <td>0.520628</td>
      <td>0.998997</td>
      <td>0.999254</td>
      <td>1.0</td>
      <td>6/17</td>
      <td>49.00%</td>
      <td>PRKDC;BARD1;HSPD1;NUP155;MTHFD1L;ITGA2</td>
    </tr>
    <tr>
      <th>994</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Hepatoma Hep-G2 BTO:0000599...</td>
      <td>0.259668</td>
      <td>0.519947</td>
      <td>1.0</td>
      <td>0.998261</td>
      <td>1.0</td>
      <td>3/15</td>
      <td>22.00%</td>
      <td>HMGA2;BARD1;RTTN</td>
    </tr>
    <tr>
      <th>995</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Hepatoma Hep-G2 BTO:0000599...</td>
      <td>0.245602</td>
      <td>0.477834</td>
      <td>0.997976</td>
      <td>0.998991</td>
      <td>1.0</td>
      <td>2/12</td>
      <td>22.00%</td>
      <td>BARD1;RTTN</td>
    </tr>
  </tbody>
</table>
<p>996 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Proteomic Analysis - Proliferating AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prolifAT2_14&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Proteomic Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">ValueError</span><span class="g g-Whitespace">                                </span>Traceback (most recent call last)
<span class="n">Cell</span> <span class="n">In</span><span class="p">[</span><span class="mi">167</span><span class="p">],</span> <span class="n">line</span> <span class="mi">2</span>
<span class="g g-Whitespace">      </span><span class="mi">1</span> <span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="ne">----&gt; </span><span class="mi">2</span> <span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">3</span>                  <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">4</span>                  <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Proteomic Analysis - Proliferating AT2 Like Cells&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">5</span>                  <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">6</span>                  <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">7</span>                  <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
<span class="g g-Whitespace">      </span><span class="mi">8</span>                  <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">9</span> <span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="g g-Whitespace">     </span><span class="mi">10</span> <span class="n">sc_plot_and_embed</span><span class="p">(</span>
<span class="g g-Whitespace">     </span><span class="mi">11</span>     <span class="n">ax_dot</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">12</span>     <span class="s2">&quot;gsea_prolifAT2_14&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">13</span>     <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Proteomic Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">14</span>     <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="g g-Whitespace">     </span><span class="mi">15</span> <span class="p">)</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:1225,</span> in <span class="ni">dotplot</span><span class="nt">(df, column, x, y, x_order, y_order, title, cutoff, top_term, size, ax, figsize, cmap, ofname, xticklabels_rot, yticklabels_rot, marker, show_ring, **kwargs)</span>
<span class="g g-Whitespace">   </span><span class="mi">1222</span>     <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span><span class="s2">&quot;group is deprecated; use x instead&quot;</span><span class="p">,</span> <span class="ne">DeprecationWarning</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1223</span>     <span class="k">return</span>
<span class="ne">-&gt; </span><span class="mi">1225</span> <span class="n">dot</span> <span class="o">=</span> <span class="n">DotPlot</span><span class="p">(</span>
<span class="g g-Whitespace">   </span><span class="mi">1226</span>     <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1227</span>     <span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1228</span>     <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1229</span>     <span class="n">x_order</span><span class="o">=</span><span class="n">x_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1230</span>     <span class="n">y_order</span><span class="o">=</span><span class="n">y_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1231</span>     <span class="n">hue</span><span class="o">=</span><span class="n">column</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1232</span>     <span class="n">title</span><span class="o">=</span><span class="n">title</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1233</span>     <span class="n">thresh</span><span class="o">=</span><span class="n">cutoff</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1234</span>     <span class="n">n_terms</span><span class="o">=</span><span class="nb">int</span><span class="p">(</span><span class="n">top_term</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1235</span>     <span class="n">dot_scale</span><span class="o">=</span><span class="n">size</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1236</span>     <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1237</span>     <span class="n">figsize</span><span class="o">=</span><span class="n">figsize</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1238</span>     <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1239</span>     <span class="n">ofname</span><span class="o">=</span><span class="n">ofname</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1240</span>     <span class="n">marker</span><span class="o">=</span><span class="n">marker</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1241</span> <span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1242</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dot</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">outer_ring</span><span class="o">=</span><span class="n">show_ring</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1244</span> <span class="k">if</span> <span class="n">xticklabels_rot</span><span class="p">:</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:692,</span> in <span class="ni">DotPlot.__init__</span><span class="nt">(self, df, x, y, hue, dot_scale, x_order, y_order, thresh, n_terms, title, ax, figsize, cmap, ofname, marker, **kwargs)</span>
<span class="g g-Whitespace">    </span><span class="mi">690</span> <span class="bp">self</span><span class="o">.</span><span class="n">n_terms</span> <span class="o">=</span> <span class="n">n_terms</span>
<span class="g g-Whitespace">    </span><span class="mi">691</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span> <span class="o">=</span> <span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">692</span> <span class="bp">self</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">693</span> <span class="n">plt</span><span class="o">.</span><span class="n">rcParams</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="s2">&quot;pdf.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">,</span> <span class="s2">&quot;ps.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">})</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:717,</span> in <span class="ni">DotPlot.process</span><span class="nt">(self, df)</span>
<span class="g g-Whitespace">    </span><span class="mi">715</span> <span class="k">if</span> <span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">716</span>     <span class="n">msg</span> <span class="o">=</span> <span class="s2">&quot;Warning: No enrich terms when cutoff = </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">717</span>     <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="n">msg</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">718</span> <span class="bp">self</span><span class="o">.</span><span class="n">cbar_title</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">colname</span>
<span class="g g-Whitespace">    </span><span class="mi">719</span> <span class="c1"># clip GSEA lower bounds</span>
<span class="g g-Whitespace">    </span><span class="mi">720</span> <span class="c1"># if self.colname in [&quot;NOM p-val&quot;, &quot;FDR q-val&quot;]:</span>
<span class="g g-Whitespace">    </span><span class="mi">721</span> <span class="c1">#     df[self.colname].clip(1e-5, 1.0, inplace=True)</span>
<span class="g g-Whitespace">    </span><span class="mi">722</span> <span class="c1"># sorting the dataframe for better visualization</span>

<span class="ne">ValueError</span>: Warning: No enrich terms when cutoff = 0.05
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prlolifAT2_15&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Proteomic Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_prlolifAT2_15.png" alt="gsea_prlolifAT2_15" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Proteomic Analysis Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_prlolifAT2_15.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id18">
<h3>GSEA - Proteomic Analysis [Top 500]<a class="headerlink" href="#id18" title="Link to this heading">#</a></h3>
<p>‘ProteomicsDB_2020’,’Tissue_Protein_Expression_from_Human_Proteome_Map’,’Tissue_Protein_Expression_from_ProteomicsDB’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list_top</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;ProteomicsDB_2020&#39;</span><span class="p">,</span><span class="s1">&#39;Tissue_Protein_Expression_from_Human_Proteome_Map&#39;</span><span class="p">,</span><span class="s1">&#39;Tissue_Protein_Expression_from_ProteomicsDB&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Ovary SKOV-3 BTO:0000948 P0...</td>
      <td>-0.579755</td>
      <td>-2.472834</td>
      <td>0.0</td>
      <td>0.067616</td>
      <td>0.043</td>
      <td>10/11</td>
      <td>31.00%</td>
      <td>ALDH7A1;HSPA9;SLC7A1;RTN3;LDHA;PTPRG;FAR1;XRCC...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Ovary SKOV-3 BTO:0000948 P0...</td>
      <td>-0.579652</td>
      <td>-2.392973</td>
      <td>0.0</td>
      <td>0.05501</td>
      <td>0.064</td>
      <td>10/11</td>
      <td>31.00%</td>
      <td>ALDH7A1;HSPA9;IVNS1ABP;RTN3;LDHA;PTPRG;FAR1;KH...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_ProteomicsDB__U...</td>
      <td>0.607542</td>
      <td>2.387737</td>
      <td>0.0</td>
      <td>0.030599</td>
      <td>0.02</td>
      <td>13/16</td>
      <td>29.40%</td>
      <td>CENPP;SMC4;FANCI;NCAPG2;RFC3;MMS22L;SMC2;MELK;...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Hematopoietic CCRF-CEM BTO:...</td>
      <td>0.411117</td>
      <td>2.263177</td>
      <td>0.001099</td>
      <td>0.080323</td>
      <td>0.081</td>
      <td>27/46</td>
      <td>29.60%</td>
      <td>RRM2;TUBB;FANCI;TMPO;KIF11;DNMT1;SMC2;HMGB1;HM...</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Cervix HeLa BTO:0000567 Lys...</td>
      <td>0.321261</td>
      <td>2.147951</td>
      <td>0.0</td>
      <td>0.170845</td>
      <td>0.195</td>
      <td>59/105</td>
      <td>38.80%</td>
      <td>DIAPH3;SMC4;BUB1B;CENPF;POLA1;ASPM;NCAPG2;PRKD...</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>990</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Ovary OVCAR-8/ADR BTO:00041...</td>
      <td>0.113156</td>
      <td>0.466362</td>
      <td>0.990037</td>
      <td>0.991548</td>
      <td>1.0</td>
      <td>10/17</td>
      <td>54.60%</td>
      <td>TMPO;XRCC5;BARD1;PCNA;PTGES3;RAN;HSP90AA1;IPO5...</td>
    </tr>
    <tr>
      <th>991</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Colon SW-480 BTO:0000038 SW...</td>
      <td>0.128919</td>
      <td>0.457402</td>
      <td>0.994631</td>
      <td>0.992275</td>
      <td>1.0</td>
      <td>2/12</td>
      <td>11.00%</td>
      <td>HMGB1;HMGB2</td>
    </tr>
    <tr>
      <th>992</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Melanoma UACC-257 BTO:00042...</td>
      <td>0.090526</td>
      <td>0.435862</td>
      <td>0.989595</td>
      <td>0.995334</td>
      <td>1.0</td>
      <td>8/29</td>
      <td>26.40%</td>
      <td>PRKDC;WWOX;XRCC5;KPNB1;BARD1;MTHFD1;CEP152;MYH10</td>
    </tr>
    <tr>
      <th>993</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Renal HEK-293 BTO:0000007 H...</td>
      <td>0.076038</td>
      <td>0.396308</td>
      <td>0.998865</td>
      <td>0.998515</td>
      <td>1.0</td>
      <td>29/36</td>
      <td>76.80%</td>
      <td>TMPO;MMS22L;MCM4;USP1;ILF3;BARD1;CHAF1A;SAE1;D...</td>
    </tr>
    <tr>
      <th>994</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_ProteomicsDB__a...</td>
      <td>0.149495</td>
      <td>0.384558</td>
      <td>0.998523</td>
      <td>0.998048</td>
      <td>1.0</td>
      <td>5/5</td>
      <td>85.20%</td>
      <td>RIF1;PCNT;PNN;MDN1;BAZ1A</td>
    </tr>
  </tbody>
</table>
<p>995 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Proteomic Analysis - Proliferating AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prolifAT2_16&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Proteomic Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_prolifAT2_16.png" alt="gsea_prolifAT2_16" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Proteomic Analysis Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_prolifAT2_16.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prlolifAT2_17&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Proteomic Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_prlolifAT2_17.png" alt="gsea_prlolifAT2_17" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Proteomic Analysis Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_prlolifAT2_17.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id19">
<h3>GSEA - Disease Analysis [Top 250 &amp; Down 250]<a class="headerlink" href="#id19" title="Link to this heading">#</a></h3>
<p>‘DisGeNET’,’Disease_Perturbations_from_GEO_down’,’Disease_Perturbations_from_GEO_up’,’HDSigDB_Human_2021’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;DisGeNET&#39;</span><span class="p">,</span><span class="s1">&#39;Disease_Perturbations_from_GEO_down&#39;</span><span class="p">,</span><span class="s1">&#39;Disease_Perturbations_from_GEO_up&#39;</span><span class="p">,</span><span class="s1">&#39;HDSigDB_Human_2021&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Up-Regulated In HDAC...</td>
      <td>-0.732041</td>
      <td>-3.246905</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>8/19</td>
      <td>20.20%</td>
      <td>CTNNA1;ANXA2;UNC13B;NPC1;RRBP1;SAMD4A;AHNAK;PPARG</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_down__Asthma, a...</td>
      <td>-0.842459</td>
      <td>-2.59758</td>
      <td>0.0</td>
      <td>0.00222</td>
      <td>0.001</td>
      <td>11/14</td>
      <td>16.00%</td>
      <td>SFTPB;LMO7;SPTBN1;TPT1;LIMCH1;MACF1;EMP2;EEF1A...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_down__pulmonary...</td>
      <td>-0.775974</td>
      <td>-2.590141</td>
      <td>0.0</td>
      <td>0.00148</td>
      <td>0.001</td>
      <td>6/15</td>
      <td>16.00%</td>
      <td>NEAT1;NPC1;ICAM1;SCD;GDF15;LPCAT1</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Down-Regulated In HD...</td>
      <td>-0.800289</td>
      <td>-2.589882</td>
      <td>0.0</td>
      <td>0.00111</td>
      <td>0.001</td>
      <td>5/13</td>
      <td>11.20%</td>
      <td>MYOF;CDH1;ALCAM;DSP;RBMS1</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Mouse Astrocyte Markers (B...</td>
      <td>-0.840192</td>
      <td>-2.557499</td>
      <td>0.0</td>
      <td>0.000888</td>
      <td>0.001</td>
      <td>5/10</td>
      <td>13.40%</td>
      <td>GPC5;ABCA1;NPC1;KIAA1217;SHROOM3</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>4078</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_up__Papillary C...</td>
      <td>0.205094</td>
      <td>0.41691</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>3/18</td>
      <td>49.00%</td>
      <td>CENPK;TYMS;ITGA2</td>
    </tr>
    <tr>
      <th>4079</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_down__Adenoma C...</td>
      <td>0.205683</td>
      <td>0.405591</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>4/14</td>
      <td>48.00%</td>
      <td>PTMA;TUBB4B;RBBP7;HSP90AA1</td>
    </tr>
    <tr>
      <th>4080</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__AD Up-Regulated Genes In C...</td>
      <td>0.187329</td>
      <td>0.402769</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>7/36</td>
      <td>47.80%</td>
      <td>NSD2;HMGA1;CEP152;RGS3;IPO9;SF3B3;SPAG5</td>
    </tr>
    <tr>
      <th>4081</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Changed In Inferior ...</td>
      <td>0.195488</td>
      <td>0.402502</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>4/21</td>
      <td>48.00%</td>
      <td>STMN1;KAZN;FUS;HSP90AA1</td>
    </tr>
    <tr>
      <th>4082</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Up-Regulated In Frie...</td>
      <td>0.162491</td>
      <td>0.363383</td>
      <td>1.0</td>
      <td>0.999999</td>
      <td>1.0</td>
      <td>21/88</td>
      <td>49.60%</td>
      <td>SMC4;PRC1;HMGA1;HMGA2;KAZN;BRCA1;TYMS;IQGAP3;S...</td>
    </tr>
  </tbody>
</table>
<p>4083 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Disease Analysis - Proliferating AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prolifAT2_18&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Disease Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_prolifAT2_18.png" alt="gsea_prolifAT2_18" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Disease Analysis Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_prolifAT2_18.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prlolifAT2_19&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Disease Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_prlolifAT2_19.png" alt="gsea_prlolifAT2_19" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Disease Analysis Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_prlolifAT2_19.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id20">
<h3>GSEA - Disease Analysis [Top 500]<a class="headerlink" href="#id20" title="Link to this heading">#</a></h3>
<p>‘DisGeNET’,’Disease_Perturbations_from_GEO_down’,’Disease_Perturbations_from_GEO_up’,’HDSigDB_Human_2021’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list_top</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;DisGeNET&#39;</span><span class="p">,</span><span class="s1">&#39;Disease_Perturbations_from_GEO_down&#39;</span><span class="p">,</span><span class="s1">&#39;Disease_Perturbations_from_GEO_up&#39;</span><span class="p">,</span><span class="s1">&#39;HDSigDB_Human_2021&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_down__multiple ...</td>
      <td>0.494507</td>
      <td>3.291794</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>59/98</td>
      <td>29.00%</td>
      <td>TOP2A;MKI67;HELLS;BRIP1;ATAD2;RRM2;SMC4;BUB1B;...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_up__lung squamo...</td>
      <td>0.582531</td>
      <td>3.242865</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>28/43</td>
      <td>20.60%</td>
      <td>MKI67;HELLS;ATAD2;RRM2;BUB1B;CENPF;FAM111B;POL...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_up__lung large ...</td>
      <td>0.584294</td>
      <td>3.198427</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>26/40</td>
      <td>20.60%</td>
      <td>MKI67;ATAD2;RRM2;BUB1B;CENPF;FAM111B;POLQ;ASPM...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Up-Regulated In Expr...</td>
      <td>0.430943</td>
      <td>3.091636</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>82/191</td>
      <td>23.40%</td>
      <td>TOP2A;MKI67;DIAPH3;HELLS;BRIP1;ATAD2;RRM2;SMC4...</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_down__prostate ...</td>
      <td>0.460149</td>
      <td>3.075349</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>59/104</td>
      <td>29.00%</td>
      <td>TOP2A;MKI67;DIAPH3;HELLS;BRIP1;ATAD2;RRM2;SMC4...</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>3680</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Changed In Vehicle T...</td>
      <td>0.101195</td>
      <td>0.354441</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>2/12</td>
      <td>14.60%</td>
      <td>DEK;USP1</td>
    </tr>
    <tr>
      <th>3681</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Changed In HDACi 106...</td>
      <td>0.049494</td>
      <td>0.350635</td>
      <td>0.992879</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>85/145</td>
      <td>60.20%</td>
      <td>TOP2A;BUB1B;CENPF;ASPM;DEK;RRM1;LMNB1;EZH2;PRC...</td>
    </tr>
    <tr>
      <th>3682</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Down-Regulated In Mo...</td>
      <td>0.061458</td>
      <td>0.349562</td>
      <td>0.998922</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>11/50</td>
      <td>23.80%</td>
      <td>GPC6;LINC01572;RAD51B;STMN1;SCLT1;SYNE2;ILF3;H...</td>
    </tr>
    <tr>
      <th>3683</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Changed In Endotheli...</td>
      <td>0.051093</td>
      <td>0.347973</td>
      <td>0.996967</td>
      <td>0.999698</td>
      <td>1.0</td>
      <td>13/117</td>
      <td>12.20%</td>
      <td>GPC6;POLA1;FANCI;PRKDC;NASP;DEK;NSD2;DNMT1;RAD...</td>
    </tr>
    <tr>
      <th>3684</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Differentially Expressed G...</td>
      <td>0.047344</td>
      <td>0.336188</td>
      <td>0.995992</td>
      <td>0.999564</td>
      <td>1.0</td>
      <td>10/171</td>
      <td>6.00%</td>
      <td>MKI67;DIAPH3;ATAD2;SMC4;TUBA1B;PRKDC;DEK;NSD2;...</td>
    </tr>
  </tbody>
</table>
<p>3685 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s1">&#39;GSEA Disease Analysis - Proliferating AT2 Like Cells&#39;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prolifAT2_20&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Disease Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_prolifAT2_20.png" alt="gsea_prolifAT2_20" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Disease Analysis Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_prolifAT2_20.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_prlolifAT2_21&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Disease Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_prlolifAT2_21.png" alt="gsea_prlolifAT2_21" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Disease Analysis Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_prlolifAT2_21.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
</section>
<section id="ascl1-at2-like-cells">
<h2>‘ASCL1+’ -AT2 Like Cells<a class="headerlink" href="#ascl1-at2-like-cells" title="Link to this heading">#</a></h2>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">numpy</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">np</span>

<span class="n">deg_results</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({</span>
    <span class="s1">&#39;names&#39;</span><span class="p">:</span> <span class="n">at2_h_lp_filt</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;rank_genes_groups&#39;</span><span class="p">][</span><span class="s1">&#39;names&#39;</span><span class="p">][</span><span class="s2">&quot;&#39;ASCL1+&#39; AT2 Like Cells&quot;</span> <span class="p">],</span>    <span class="c1"># Gene names</span>
    <span class="s1">&#39;logfoldchanges&#39;</span><span class="p">:</span> <span class="n">at2_h_lp_filt</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;rank_genes_groups&#39;</span><span class="p">][</span><span class="s1">&#39;logfoldchanges&#39;</span><span class="p">][</span><span class="s2">&quot;&#39;ASCL1+&#39; AT2 Like Cells&quot;</span> <span class="p">],</span>     <span class="c1"># Log fold changes</span>
    <span class="s1">&#39;pvals&#39;</span><span class="p">:</span> <span class="n">at2_h_lp_filt</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;rank_genes_groups&#39;</span><span class="p">][</span><span class="s1">&#39;pvals&#39;</span><span class="p">][</span><span class="s2">&quot;&#39;ASCL1+&#39; AT2 Like Cells&quot;</span> <span class="p">],</span>          <span class="c1"># p-values</span>
    <span class="s1">&#39;pvals_adj&#39;</span><span class="p">:</span> <span class="n">at2_h_lp_filt</span><span class="o">.</span><span class="n">uns</span><span class="p">[</span><span class="s1">&#39;rank_genes_groups&#39;</span><span class="p">][</span><span class="s1">&#39;pvals_adj&#39;</span><span class="p">][</span><span class="s2">&quot;&#39;ASCL1+&#39; AT2 Like Cells&quot;</span> <span class="p">]</span> <span class="c1"># Adjusted p-values (FDR corrected)</span>
<span class="p">})</span>

<span class="c1"># Step 1: Set thresholds for logFC and p-value (adjusted)</span>
<span class="n">logfc_threshold</span> <span class="o">=</span> <span class="mf">0.01</span>   <span class="c1"># Example: filter for logFC &gt; 1 (or &lt; -1 for downregulated)</span>
<span class="n">pval_adj_threshold</span> <span class="o">=</span> <span class="mf">0.05</span>  <span class="c1"># Example: adjusted p-value &lt; 0.05</span>

<span class="c1"># Step 2: Apply the filtering based on logFC and adjusted p-value</span>
<span class="n">filtered_genes</span> <span class="o">=</span> <span class="n">deg_results</span><span class="p">[</span>
    <span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">abs</span><span class="p">(</span><span class="n">deg_results</span><span class="p">[</span><span class="s1">&#39;logfoldchanges&#39;</span><span class="p">])</span> <span class="o">&gt;=</span> <span class="n">logfc_threshold</span><span class="p">)</span> <span class="o">&amp;</span>  <span class="c1"># Absolute logFC above threshold</span>
    <span class="p">(</span><span class="n">deg_results</span><span class="p">[</span><span class="s1">&#39;pvals_adj&#39;</span><span class="p">]</span> <span class="o">&lt;</span> <span class="n">pval_adj_threshold</span><span class="p">)</span>               <span class="c1"># Adjusted p-value below threshold</span>
<span class="p">]</span>

<span class="c1"># Step 3: Extract the list of filtered gene names</span>
<span class="n">filtered_gene_names</span> <span class="o">=</span> <span class="n">filtered_genes</span><span class="p">[</span><span class="s1">&#39;names&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
<span class="c1">#filtered_gene_names[:50]</span>

<span class="c1"># Run g:Profiler</span>
<span class="n">go_results</span> <span class="o">=</span> <span class="n">gp</span><span class="o">.</span><span class="n">profile</span><span class="p">(</span><span class="n">organism</span><span class="o">=</span><span class="s1">&#39;hsapiens&#39;</span><span class="p">,</span> <span class="n">query</span><span class="o">=</span><span class="n">filtered_gene_names</span><span class="p">[:</span><span class="mi">250</span><span class="p">])</span>

<span class="c1"># Define the sources you want to loop over</span>
<span class="n">sources</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;GO:BP&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:MF&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:CC&#39;</span><span class="p">,</span> <span class="s1">&#39;KEGG&#39;</span><span class="p">,</span> <span class="s1">&#39;REAC&#39;</span><span class="p">,</span> <span class="s1">&#39;HPA&#39;</span><span class="p">,</span> <span class="s1">&#39;WP&#39;</span><span class="p">]</span>

<span class="c1"># Create a stacked figure with one row per source</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">nrows</span><span class="o">=</span><span class="nb">len</span><span class="p">(</span><span class="n">sources</span><span class="p">),</span> <span class="n">ncols</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">5.5</span><span class="p">,</span> <span class="mf">2.5</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">sources</span><span class="p">)))</span>
<span class="n">axes</span> <span class="o">=</span> <span class="n">axes</span><span class="o">.</span><span class="n">flatten</span><span class="p">()</span>

<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">source</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">sources</span><span class="p">):</span>
    <span class="c1"># Filter results for this source</span>
    <span class="n">specific_results</span> <span class="o">=</span> <span class="n">go_results</span><span class="p">[</span><span class="n">go_results</span><span class="p">[</span><span class="s1">&#39;source&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">source</span><span class="p">]</span>
    
    <span class="k">if</span> <span class="n">specific_results</span><span class="o">.</span><span class="n">empty</span><span class="p">:</span>
        <span class="k">continue</span>  <span class="c1"># skip if no results for this source</span>
    
    <span class="c1"># Select top 8–10 terms</span>
    <span class="n">go_results_top</span> <span class="o">=</span> <span class="p">(</span>
        <span class="n">specific_results</span><span class="p">[[</span><span class="s1">&#39;name&#39;</span><span class="p">,</span> <span class="s1">&#39;p_value&#39;</span><span class="p">]]</span>
        <span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;p_value&#39;</span><span class="p">)</span>
        <span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">8</span><span class="p">)</span>
    <span class="p">)</span>

    <span class="n">labels</span> <span class="o">=</span> <span class="n">go_results_top</span><span class="p">[</span><span class="s1">&#39;name&#39;</span><span class="p">]</span>
    <span class="n">values</span> <span class="o">=</span> <span class="o">-</span><span class="n">np</span><span class="o">.</span><span class="n">log10</span><span class="p">(</span><span class="n">go_results_top</span><span class="p">[</span><span class="s1">&#39;p_value&#39;</span><span class="p">])</span>

    <span class="n">ax</span> <span class="o">=</span> <span class="n">axes</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">barh</span><span class="p">(</span><span class="n">labels</span><span class="p">,</span> <span class="n">values</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s2">&quot;#1f77b4&quot;</span><span class="p">)</span>
    
    <span class="c1"># Aesthetics</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">source</span><span class="si">}</span><span class="s2"> enrichment&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span> <span class="n">loc</span><span class="o">=</span><span class="s2">&quot;left&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;-log10(P-value)&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">invert_yaxis</span><span class="p">()</span>  <span class="c1"># highest term on top</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="s2">&quot;y&quot;</span><span class="p">,</span> <span class="n">labelsize</span><span class="o">=</span><span class="mi">9</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="s2">&quot;x&quot;</span><span class="p">,</span> <span class="n">labelsize</span><span class="o">=</span><span class="mi">9</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s2">&quot;top&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="kc">False</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s2">&quot;right&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="kc">False</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">grid</span><span class="p">(</span><span class="kc">False</span><span class="p">)</span>
<span class="c1"># Adjust layout</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>

<span class="c1"># Save if needed</span>
<span class="c1"># plt.savefig(&quot;GO_enrichment_all_sources.png&quot;, dpi=600, bbox_inches=&quot;tight&quot;)</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_1&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GProfiler Analysis of top 250 differentially expressed genes from &#39;ASCL1+&#39; AT2 Like Cells.&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_ascl1AT2_1.png" alt="gsea_ascl1AT2_1" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GProfiler Analysis of top 250 differentially expressed genes from 'ASCL1+' AT2 Like Cells.</em><br>
        <a href="_static/Figures/gsea_ascl1AT2_1.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>

<span class="n">group</span> <span class="o">=</span> <span class="s2">&quot;&#39;ASCL1+&#39; AT2 Like Cells&quot;</span>  <span class="c1"># change for each group</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">get</span><span class="o">.</span><span class="n">rank_genes_groups_df</span><span class="p">(</span><span class="n">at2_h_lp_filt</span><span class="p">,</span> <span class="n">group</span><span class="o">=</span><span class="n">group</span><span class="p">)</span>

<span class="c1"># GSEAPY prerank needs index=gene, score=stat/logFC</span>
<span class="n">ranked_list</span> <span class="o">=</span> <span class="n">df</span><span class="p">[[</span><span class="s1">&#39;names&#39;</span><span class="p">,</span> <span class="s1">&#39;scores&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;names&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">squeeze</span><span class="p">()</span>

<span class="n">df_ranked_list_top</span> <span class="o">=</span> <span class="n">ranked_list</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">500</span><span class="p">]</span>

<span class="n">ranked_list_pos</span> <span class="o">=</span> <span class="n">ranked_list</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">250</span><span class="p">]</span>
<span class="n">ranked_list_neg</span> <span class="o">=</span> <span class="n">ranked_list</span><span class="p">[(</span><span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">)</span><span class="o">-</span><span class="mi">250</span><span class="p">):</span><span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">)]</span>

<span class="n">ranked_lists</span> <span class="o">=</span> <span class="p">[</span><span class="n">ranked_list_pos</span><span class="p">,</span><span class="n">ranked_list_neg</span><span class="p">]</span>

<span class="n">df_ranked_list</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">ranked_lists</span><span class="p">)</span>
<span class="c1"># Save to .rnk if needed</span>
<span class="c1">#ranked_list.to_csv(&quot;lateAT2.rnk&quot;, sep=&quot;\t&quot;)</span>
<span class="n">df_ranked_list</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>names
LINC00511    21.420732
DSP          20.795889
NKAIN2       19.929850
DACH1        19.489086
IGFBP5       18.652613
               ...    
ROS1        -21.196671
GPC5        -21.495214
ATP11A      -21.637205
SFTPC       -24.224373
DLC1        -27.102341
Name: scores, Length: 500, dtype: float32
</pre></div>
</div>
</div>
</div>
<section id="id21">
<h3>GSEA - Cell Type [Top 250 &amp; Down 250]<a class="headerlink" href="#id21" title="Link to this heading">#</a></h3>
<p>‘Azimuth_2023’,’CellMarker_2024’,’Descartes_Cell_Types_and_Tissue_2021’,’Human_Gene_Atlas’,’Human_Phenotype_Ontology’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Azimuth_2023&#39;</span><span class="p">,</span><span class="s1">&#39;CellMarker_2024&#39;</span><span class="p">,</span><span class="s1">&#39;Descartes_Cell_Types_and_Tissue_2021&#39;</span><span class="p">,</span><span class="s1">&#39;Human_Gene_Atlas&#39;</span><span class="p">,</span><span class="s1">&#39;Human_Phenotype_Ontology&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>Descartes_Cell_Types_and_Tissue_2021__Bronchio...</td>
      <td>-0.729611</td>
      <td>-2.507278</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>8/12</td>
      <td>8.60%</td>
      <td>SFTPC;ATP11A;ROS1;SLC34A2;SFTPB;ACOXL;CPM;LAMP3</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>Human_Gene_Atlas__Lung</td>
      <td>-0.529746</td>
      <td>-2.3837</td>
      <td>0.0</td>
      <td>0.001948</td>
      <td>0.004</td>
      <td>20/26</td>
      <td>35.20%</td>
      <td>DLC1;SFTPC;SLC34A2;SFTPB;ABCA3;SPTBN1;LAMP3;SL...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>Azimuth_2023__Lung v1-L1-Alveolar Epithelial T...</td>
      <td>-0.845097</td>
      <td>-2.359168</td>
      <td>0.0</td>
      <td>0.001299</td>
      <td>0.004</td>
      <td>6/7</td>
      <td>8.60%</td>
      <td>SFTPC;LPCAT1;SLC34A2;SFTPB;ABCA3;LAMP3</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>CellMarker_2024__Type II Pneumocyte Lung Mouse</td>
      <td>-0.683673</td>
      <td>-2.173454</td>
      <td>0.0</td>
      <td>0.011446</td>
      <td>0.043</td>
      <td>10/10</td>
      <td>33.20%</td>
      <td>SFTPC;SFTPA1;SLC34A2;SFTPB;RAB27A;LAMP3;FASN;C...</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>Human_Gene_Atlas__SmoothMuscle</td>
      <td>-0.4741</td>
      <td>-2.169738</td>
      <td>0.0</td>
      <td>0.009157</td>
      <td>0.043</td>
      <td>22/29</td>
      <td>38.80%</td>
      <td>DLC1;MYOF;EXT1;PID1;OSMR;SEC24D;ICAM1;FNDC3B;P...</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>287</th>
      <td>prerank</td>
      <td>CellMarker_2024__Monocyte Spleen Mouse</td>
      <td>-0.184584</td>
      <td>-0.518195</td>
      <td>0.972509</td>
      <td>0.983606</td>
      <td>1.0</td>
      <td>7/7</td>
      <td>82.00%</td>
      <td>CUX1;SMC4;NPC1;VIM;CTNNB1;ADD3;APP</td>
    </tr>
    <tr>
      <th>288</th>
      <td>prerank</td>
      <td>Human_Phenotype_Ontology__Developmental regres...</td>
      <td>0.184211</td>
      <td>0.517956</td>
      <td>0.990783</td>
      <td>0.988614</td>
      <td>1.0</td>
      <td>6/6</td>
      <td>81.80%</td>
      <td>APP;PSAP;ZBTB20;CHD2;ACTB;ST3GAL5</td>
    </tr>
    <tr>
      <th>289</th>
      <td>prerank</td>
      <td>Human_Phenotype_Ontology__Hypertelorism (HP:00...</td>
      <td>-0.146932</td>
      <td>-0.506215</td>
      <td>0.976744</td>
      <td>0.982628</td>
      <td>1.0</td>
      <td>1/12</td>
      <td>3.20%</td>
      <td>ADK</td>
    </tr>
    <tr>
      <th>290</th>
      <td>prerank</td>
      <td>CellMarker_2024__Neuron Undefined Human</td>
      <td>-0.196356</td>
      <td>-0.504283</td>
      <td>0.981387</td>
      <td>0.977556</td>
      <td>1.0</td>
      <td>6/6</td>
      <td>80.80%</td>
      <td>NDNF;CD47;VIM;CD24;NCAM1;BCL2</td>
    </tr>
    <tr>
      <th>291</th>
      <td>prerank</td>
      <td>Human_Gene_Atlas__CD71+ EarlyErythroid</td>
      <td>0.122449</td>
      <td>0.431815</td>
      <td>1.0</td>
      <td>0.997343</td>
      <td>1.0</td>
      <td>10/10</td>
      <td>88.00%</td>
      <td>TSPAN5;NFIA;FRYL;SLC2A1;SSBP3;ATAD2;MKI67;SMC4...</td>
    </tr>
  </tbody>
</table>
<p>292 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s2">&quot;GSEA Cell Type - &#39;ASCL1+&#39; AT2 Like Cells&quot;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_2&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Cell Type Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_ascl1AT2_2.png" alt="gsea_ascl1AT2_2" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Cell Type Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_ascl1AT2_2.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_3&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Cell Type Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_ascl1AT2_3.png" alt="gsea_ascl1AT2_3" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Cell Type Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_ascl1AT2_3.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id22">
<h3>GSEA - Cell Type [Top 500]<a class="headerlink" href="#id22" title="Link to this heading">#</a></h3>
<p>‘Azimuth_2023’,’CellMarker_2024’,’Descartes_Cell_Types_and_Tissue_2021’,’Human_Gene_Atlas’,’Human_Phenotype_Ontology’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list_top</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Azimuth_2023&#39;</span><span class="p">,</span><span class="s1">&#39;CellMarker_2024&#39;</span><span class="p">,</span><span class="s1">&#39;Descartes_Cell_Types_and_Tissue_2021&#39;</span><span class="p">,</span><span class="s1">&#39;Human_Gene_Atlas&#39;</span><span class="p">,</span><span class="s1">&#39;Human_Phenotype_Ontology&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>CellMarker_2024__Meiotic Prophase Fetal Germ C...</td>
      <td>-0.32322</td>
      <td>-2.325014</td>
      <td>0.0</td>
      <td>0.018553</td>
      <td>0.025</td>
      <td>26/30</td>
      <td>49.80%</td>
      <td>ATOSA;KDM5B;PHYHIPL;AGBL4;RABGAP1;KIF1A;KIF5B;...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>Azimuth_2023__Fetal Development-L2-Lung-Neuroe...</td>
      <td>0.775303</td>
      <td>1.931397</td>
      <td>0.002604</td>
      <td>0.300941</td>
      <td>0.218</td>
      <td>3/6</td>
      <td>1.40%</td>
      <td>NKAIN2;IGFBP5;SEC11C</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>CellMarker_2024__Type IC Spiral Ganglion Neuro...</td>
      <td>0.613614</td>
      <td>1.918755</td>
      <td>0.004678</td>
      <td>0.174032</td>
      <td>0.251</td>
      <td>6/13</td>
      <td>9.60%</td>
      <td>DACH1;ROBO1;MGAT4C;NCALD;NEGR1;GALNT13</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>Descartes_Cell_Types_and_Tissue_2021__Stromal ...</td>
      <td>0.56324</td>
      <td>1.86041</td>
      <td>0.001117</td>
      <td>0.22169</td>
      <td>0.423</td>
      <td>12/15</td>
      <td>35.80%</td>
      <td>IGFBP5;GALNT13;ADAMTS19;APCDD1;CTNNA2;GREB1;LI...</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>Descartes_Cell_Types_and_Tissue_2021__Megakary...</td>
      <td>-0.597166</td>
      <td>-1.816302</td>
      <td>0.024793</td>
      <td>0.238097</td>
      <td>0.452</td>
      <td>6/6</td>
      <td>41.20%</td>
      <td>SH3BGRL2;VCL;SLC37A1;SLC24A3;DSCAML1;MEIS1</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>239</th>
      <td>prerank</td>
      <td>Human_Phenotype_Ontology__Paresthesia (HP:0003...</td>
      <td>-0.183838</td>
      <td>-0.484539</td>
      <td>0.984733</td>
      <td>0.994082</td>
      <td>1.0</td>
      <td>5/5</td>
      <td>82.00%</td>
      <td>TET2;LYST;CACNA1A;SLC2A1;APP</td>
    </tr>
    <tr>
      <th>240</th>
      <td>prerank</td>
      <td>CellMarker_2024__Gonadal Mitotic Phase Fetal G...</td>
      <td>0.18467</td>
      <td>0.482551</td>
      <td>0.981156</td>
      <td>0.993698</td>
      <td>1.0</td>
      <td>1/7</td>
      <td>8.80%</td>
      <td>RAB11FIP1</td>
    </tr>
    <tr>
      <th>241</th>
      <td>prerank</td>
      <td>Human_Phenotype_Ontology__Constipation (HP:000...</td>
      <td>0.169489</td>
      <td>0.469805</td>
      <td>0.988693</td>
      <td>0.992973</td>
      <td>1.0</td>
      <td>5/9</td>
      <td>52.00%</td>
      <td>NRXN1;ASCL1;EPCAM;TCF4;PRDM16</td>
    </tr>
    <tr>
      <th>242</th>
      <td>prerank</td>
      <td>CellMarker_2024__Ciliated Cell Trachea Mouse</td>
      <td>0.128969</td>
      <td>0.464639</td>
      <td>0.982997</td>
      <td>0.989542</td>
      <td>1.0</td>
      <td>10/22</td>
      <td>46.40%</td>
      <td>SLC2A12;CALM1;CLIC6;FAM149A;TXNIP;DTNA;KREMEN1...</td>
    </tr>
    <tr>
      <th>243</th>
      <td>prerank</td>
      <td>Descartes_Cell_Types_and_Tissue_2021__Mesothel...</td>
      <td>0.123966</td>
      <td>0.405211</td>
      <td>0.995449</td>
      <td>0.995595</td>
      <td>1.0</td>
      <td>8/15</td>
      <td>54.20%</td>
      <td>NEBL;NREP;LINGO2;SEMA3C;KRT8;TUFT1;B3GALT1;MTMR7</td>
    </tr>
  </tbody>
</table>
<p>244 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s2">&quot;GSEA Cell Type - &#39;ASCL1+&#39; AT2 Like Cells&quot;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_4&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Cell Type Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_ascl1AT2_4.png" alt="gsea_ascl1AT2_4" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Cell Type Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_ascl1AT2_4.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_5&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Cell Type Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_ascl1AT2_5.png" alt="gsea_ascl1AT2_5" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Cell Type Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_ascl1AT2_5.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id23">
<h3>GSEA - Gene Ontology [Top 250 &amp; Down 250]<a class="headerlink" href="#id23" title="Link to this heading">#</a></h3>
<p>‘GO_Biological_Process_2025’,’GO_Cellular_Component_2025’,’GO_Molecular_Function_2025’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;GO_Biological_Process_2025&#39;</span><span class="p">,</span><span class="s1">&#39;GO_Cellular_Component_2025&#39;</span><span class="p">,</span><span class="s1">&#39;GO_Molecular_Function_2025&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>GO_Cellular_Component_2025__Multivesicular Bod...</td>
      <td>-0.712185</td>
      <td>-2.089974</td>
      <td>0.0</td>
      <td>0.265586</td>
      <td>0.168</td>
      <td>5/8</td>
      <td>7.40%</td>
      <td>SFTPC;SFTPA1;SFTPB;LRRK2;RAB27A</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>GO_Cellular_Component_2025__Lamellar Body (GO:...</td>
      <td>-0.767206</td>
      <td>-2.025926</td>
      <td>0.0</td>
      <td>0.234654</td>
      <td>0.281</td>
      <td>6/6</td>
      <td>24.40%</td>
      <td>SFTPC;SFTPA1;SFTPB;ABCA3;NAPSA;CTSH</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>GO_Cellular_Component_2025__Alveolar Lamellar ...</td>
      <td>-0.765657</td>
      <td>-1.863103</td>
      <td>0.00349</td>
      <td>0.625389</td>
      <td>0.695</td>
      <td>5/5</td>
      <td>24.40%</td>
      <td>SFTPC;SFTPB;LAMP3;NAPSA;CTSH</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>GO_Cellular_Component_2025__Endocytic Vesicle ...</td>
      <td>-0.533005</td>
      <td>-1.818669</td>
      <td>0.011725</td>
      <td>0.673831</td>
      <td>0.81</td>
      <td>4/12</td>
      <td>3.60%</td>
      <td>DLC1;SFTPC;SFTPA1;SFTPB</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Notch Signaling Pa...</td>
      <td>0.643725</td>
      <td>1.803392</td>
      <td>0.015873</td>
      <td>1.0</td>
      <td>0.839</td>
      <td>6/6</td>
      <td>36.40%</td>
      <td>GMDS;ASCL1;MAML2;SORBS2;MAML3;RBPJ</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>489</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Positive Regulatio...</td>
      <td>-0.113084</td>
      <td>-0.45328</td>
      <td>0.987842</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>9/19</td>
      <td>42.20%</td>
      <td>PTN;NLGN1;BMP1;SOX5;NFATC3;SMURF2;ZBTB16;MCU;P...</td>
    </tr>
    <tr>
      <th>490</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Response to Amyloi...</td>
      <td>-0.171665</td>
      <td>-0.451343</td>
      <td>0.994633</td>
      <td>0.998518</td>
      <td>1.0</td>
      <td>2/6</td>
      <td>25.00%</td>
      <td>ICAM1;CD36</td>
    </tr>
    <tr>
      <th>491</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Cellular Response ...</td>
      <td>-0.171665</td>
      <td>-0.451343</td>
      <td>0.994633</td>
      <td>0.998518</td>
      <td>1.0</td>
      <td>2/6</td>
      <td>25.00%</td>
      <td>ICAM1;CD36</td>
    </tr>
    <tr>
      <th>492</th>
      <td>prerank</td>
      <td>GO_Molecular_Function_2025__Protein Kinase Bin...</td>
      <td>-0.10677</td>
      <td>-0.440933</td>
      <td>0.989164</td>
      <td>0.996996</td>
      <td>1.0</td>
      <td>9/21</td>
      <td>39.80%</td>
      <td>PTN;UTRN;RHOBTB2;PRKCA;SPRED1;TCF7L1;IQGAP1;MS...</td>
    </tr>
    <tr>
      <th>493</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Proteasomal Protei...</td>
      <td>-0.145695</td>
      <td>-0.427685</td>
      <td>0.998243</td>
      <td>0.995463</td>
      <td>1.0</td>
      <td>3/8</td>
      <td>32.20%</td>
      <td>RNF145;KLHL5;SMURF2</td>
    </tr>
  </tbody>
</table>
<p>494 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s2">&quot;GSEA Gene Ontology - &#39;ASCL1+&#39; AT2 Like Cells&quot;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_6&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Gene Ontology Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">ValueError</span><span class="g g-Whitespace">                                </span>Traceback (most recent call last)
<span class="n">Cell</span> <span class="n">In</span><span class="p">[</span><span class="mi">204</span><span class="p">],</span> <span class="n">line</span> <span class="mi">2</span>
<span class="g g-Whitespace">      </span><span class="mi">1</span> <span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="ne">----&gt; </span><span class="mi">2</span> <span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">3</span>                  <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">4</span>                  <span class="n">title</span><span class="o">=</span><span class="s2">&quot;GSEA Gene Ontology - &#39;ASCL1+&#39; AT2 Like Cells&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">5</span>                  <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">6</span>                  <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">7</span>                  <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
<span class="g g-Whitespace">      </span><span class="mi">8</span>                  <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">9</span> <span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="g g-Whitespace">     </span><span class="mi">10</span> <span class="n">sc_plot_and_embed</span><span class="p">(</span>
<span class="g g-Whitespace">     </span><span class="mi">11</span>     <span class="n">ax_dot</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">12</span>     <span class="s2">&quot;gsea_ascl1AT2_6&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">13</span>     <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Gene Ontology Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">14</span>     <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="g g-Whitespace">     </span><span class="mi">15</span> <span class="p">)</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:1225,</span> in <span class="ni">dotplot</span><span class="nt">(df, column, x, y, x_order, y_order, title, cutoff, top_term, size, ax, figsize, cmap, ofname, xticklabels_rot, yticklabels_rot, marker, show_ring, **kwargs)</span>
<span class="g g-Whitespace">   </span><span class="mi">1222</span>     <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span><span class="s2">&quot;group is deprecated; use x instead&quot;</span><span class="p">,</span> <span class="ne">DeprecationWarning</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1223</span>     <span class="k">return</span>
<span class="ne">-&gt; </span><span class="mi">1225</span> <span class="n">dot</span> <span class="o">=</span> <span class="n">DotPlot</span><span class="p">(</span>
<span class="g g-Whitespace">   </span><span class="mi">1226</span>     <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1227</span>     <span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1228</span>     <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1229</span>     <span class="n">x_order</span><span class="o">=</span><span class="n">x_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1230</span>     <span class="n">y_order</span><span class="o">=</span><span class="n">y_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1231</span>     <span class="n">hue</span><span class="o">=</span><span class="n">column</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1232</span>     <span class="n">title</span><span class="o">=</span><span class="n">title</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1233</span>     <span class="n">thresh</span><span class="o">=</span><span class="n">cutoff</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1234</span>     <span class="n">n_terms</span><span class="o">=</span><span class="nb">int</span><span class="p">(</span><span class="n">top_term</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1235</span>     <span class="n">dot_scale</span><span class="o">=</span><span class="n">size</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1236</span>     <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1237</span>     <span class="n">figsize</span><span class="o">=</span><span class="n">figsize</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1238</span>     <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1239</span>     <span class="n">ofname</span><span class="o">=</span><span class="n">ofname</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1240</span>     <span class="n">marker</span><span class="o">=</span><span class="n">marker</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1241</span> <span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1242</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dot</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">outer_ring</span><span class="o">=</span><span class="n">show_ring</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1244</span> <span class="k">if</span> <span class="n">xticklabels_rot</span><span class="p">:</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:692,</span> in <span class="ni">DotPlot.__init__</span><span class="nt">(self, df, x, y, hue, dot_scale, x_order, y_order, thresh, n_terms, title, ax, figsize, cmap, ofname, marker, **kwargs)</span>
<span class="g g-Whitespace">    </span><span class="mi">690</span> <span class="bp">self</span><span class="o">.</span><span class="n">n_terms</span> <span class="o">=</span> <span class="n">n_terms</span>
<span class="g g-Whitespace">    </span><span class="mi">691</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span> <span class="o">=</span> <span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">692</span> <span class="bp">self</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">693</span> <span class="n">plt</span><span class="o">.</span><span class="n">rcParams</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="s2">&quot;pdf.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">,</span> <span class="s2">&quot;ps.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">})</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:717,</span> in <span class="ni">DotPlot.process</span><span class="nt">(self, df)</span>
<span class="g g-Whitespace">    </span><span class="mi">715</span> <span class="k">if</span> <span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">716</span>     <span class="n">msg</span> <span class="o">=</span> <span class="s2">&quot;Warning: No enrich terms when cutoff = </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">717</span>     <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="n">msg</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">718</span> <span class="bp">self</span><span class="o">.</span><span class="n">cbar_title</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">colname</span>
<span class="g g-Whitespace">    </span><span class="mi">719</span> <span class="c1"># clip GSEA lower bounds</span>
<span class="g g-Whitespace">    </span><span class="mi">720</span> <span class="c1"># if self.colname in [&quot;NOM p-val&quot;, &quot;FDR q-val&quot;]:</span>
<span class="g g-Whitespace">    </span><span class="mi">721</span> <span class="c1">#     df[self.colname].clip(1e-5, 1.0, inplace=True)</span>
<span class="g g-Whitespace">    </span><span class="mi">722</span> <span class="c1"># sorting the dataframe for better visualization</span>

<span class="ne">ValueError</span>: Warning: No enrich terms when cutoff = 0.05
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_7&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Gene Ontology Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_ascl1AT2_7.png" alt="gsea_ascl1AT2_7" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Gene Ontology Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_ascl1AT2_7.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id24">
<h3>GSEA - Gene Ontology [Top 500]<a class="headerlink" href="#id24" title="Link to this heading">#</a></h3>
<p>‘GO_Biological_Process_2025’,’GO_Cellular_Component_2025’,’GO_Molecular_Function_2025’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list_top</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;GO_Biological_Process_2025&#39;</span><span class="p">,</span><span class="s1">&#39;GO_Cellular_Component_2025&#39;</span><span class="p">,</span><span class="s1">&#39;GO_Molecular_Function_2025&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Intracellular Prot...</td>
      <td>-0.684211</td>
      <td>-2.01718</td>
      <td>0.004049</td>
      <td>0.30289</td>
      <td>0.273</td>
      <td>6/6</td>
      <td>32.60%</td>
      <td>ASPSCR1;PDCD6;ARL4A;VPS13C;SNX9;STXBP1</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Synapse Organizati...</td>
      <td>0.597037</td>
      <td>1.939288</td>
      <td>0.0</td>
      <td>0.426264</td>
      <td>0.256</td>
      <td>9/14</td>
      <td>20.60%</td>
      <td>CTNND2;PDZRN3;NRXN1;LRRC4C;NRCAM;SDK1;APBB2;AP...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Neuron Projection ...</td>
      <td>0.621718</td>
      <td>1.874162</td>
      <td>0.005875</td>
      <td>0.438133</td>
      <td>0.44</td>
      <td>8/11</td>
      <td>23.20%</td>
      <td>CTNND2;MAP1B;NRXN1;NRCAM;KLF7;APP;CTNNA2;CTNNB1</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__DNA Metabolic Proc...</td>
      <td>-0.664646</td>
      <td>-1.847757</td>
      <td>0.014981</td>
      <td>0.418523</td>
      <td>0.562</td>
      <td>5/5</td>
      <td>34.40%</td>
      <td>AXIN2;XRCC5;GADD45A;PSME4;HMGB3</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Cell Surface Recep...</td>
      <td>0.542095</td>
      <td>1.844007</td>
      <td>0.002176</td>
      <td>0.400461</td>
      <td>0.546</td>
      <td>9/17</td>
      <td>18.20%</td>
      <td>PLCB1;MYO1E;DOCK3;ERBB4;DOK6;MTSS1;FYN;KALRN;E...</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>404</th>
      <td>prerank</td>
      <td>GO_Molecular_Function_2025__Ubiquitin-Like Pro...</td>
      <td>0.206061</td>
      <td>0.475418</td>
      <td>0.983584</td>
      <td>0.991745</td>
      <td>1.0</td>
      <td>5/5</td>
      <td>79.60%</td>
      <td>MARCHF6;PRKN;RNF19A;ZMYM2;UBE3D</td>
    </tr>
    <tr>
      <th>405</th>
      <td>prerank</td>
      <td>GO_Molecular_Function_2025__Ubiquitin Conjugat...</td>
      <td>0.206061</td>
      <td>0.475418</td>
      <td>0.983584</td>
      <td>0.991745</td>
      <td>1.0</td>
      <td>5/5</td>
      <td>79.60%</td>
      <td>MARCHF6;PRKN;RNF19A;ZMYM2;UBE3D</td>
    </tr>
    <tr>
      <th>406</th>
      <td>prerank</td>
      <td>GO_Molecular_Function_2025__Microtubule Bindin...</td>
      <td>0.148243</td>
      <td>0.440656</td>
      <td>0.986014</td>
      <td>0.996982</td>
      <td>1.0</td>
      <td>1/11</td>
      <td>5.40%</td>
      <td>MAP1B</td>
    </tr>
    <tr>
      <th>407</th>
      <td>prerank</td>
      <td>GO_Biological_Process_2025__Modification-Depen...</td>
      <td>0.169581</td>
      <td>0.436452</td>
      <td>0.993687</td>
      <td>0.994779</td>
      <td>1.0</td>
      <td>4/7</td>
      <td>51.80%</td>
      <td>HECW2;RNF43;PRKN;OS9</td>
    </tr>
    <tr>
      <th>408</th>
      <td>prerank</td>
      <td>GO_Cellular_Component_2025__Microtubule (GO:00...</td>
      <td>0.127911</td>
      <td>0.390822</td>
      <td>0.99537</td>
      <td>0.997359</td>
      <td>1.0</td>
      <td>1/12</td>
      <td>5.40%</td>
      <td>MAP1B</td>
    </tr>
  </tbody>
</table>
<p>409 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s2">&quot;GSEA Gene Ontology - &#39;ASCL1+&#39; AT2 Like Cells&quot;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_8&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Gene Ontology Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">ValueError</span><span class="g g-Whitespace">                                </span>Traceback (most recent call last)
<span class="n">Cell</span> <span class="n">In</span><span class="p">[</span><span class="mi">217</span><span class="p">],</span> <span class="n">line</span> <span class="mi">2</span>
<span class="g g-Whitespace">      </span><span class="mi">1</span> <span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="ne">----&gt; </span><span class="mi">2</span> <span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">3</span>                  <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">4</span>                  <span class="n">title</span><span class="o">=</span><span class="s2">&quot;GSEA Gene Ontology - &#39;ASCL1+&#39; AT2 Like Cells&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">5</span>                  <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">6</span>                  <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">7</span>                  <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
<span class="g g-Whitespace">      </span><span class="mi">8</span>                  <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">9</span> <span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="g g-Whitespace">     </span><span class="mi">10</span> <span class="n">sc_plot_and_embed</span><span class="p">(</span>
<span class="g g-Whitespace">     </span><span class="mi">11</span>     <span class="n">ax_dot</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">12</span>     <span class="s2">&quot;gsea_ascl1AT2_8&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">13</span>     <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Gene Ontology Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">14</span>     <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="g g-Whitespace">     </span><span class="mi">15</span> <span class="p">)</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:1225,</span> in <span class="ni">dotplot</span><span class="nt">(df, column, x, y, x_order, y_order, title, cutoff, top_term, size, ax, figsize, cmap, ofname, xticklabels_rot, yticklabels_rot, marker, show_ring, **kwargs)</span>
<span class="g g-Whitespace">   </span><span class="mi">1222</span>     <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span><span class="s2">&quot;group is deprecated; use x instead&quot;</span><span class="p">,</span> <span class="ne">DeprecationWarning</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1223</span>     <span class="k">return</span>
<span class="ne">-&gt; </span><span class="mi">1225</span> <span class="n">dot</span> <span class="o">=</span> <span class="n">DotPlot</span><span class="p">(</span>
<span class="g g-Whitespace">   </span><span class="mi">1226</span>     <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1227</span>     <span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1228</span>     <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1229</span>     <span class="n">x_order</span><span class="o">=</span><span class="n">x_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1230</span>     <span class="n">y_order</span><span class="o">=</span><span class="n">y_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1231</span>     <span class="n">hue</span><span class="o">=</span><span class="n">column</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1232</span>     <span class="n">title</span><span class="o">=</span><span class="n">title</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1233</span>     <span class="n">thresh</span><span class="o">=</span><span class="n">cutoff</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1234</span>     <span class="n">n_terms</span><span class="o">=</span><span class="nb">int</span><span class="p">(</span><span class="n">top_term</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1235</span>     <span class="n">dot_scale</span><span class="o">=</span><span class="n">size</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1236</span>     <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1237</span>     <span class="n">figsize</span><span class="o">=</span><span class="n">figsize</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1238</span>     <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1239</span>     <span class="n">ofname</span><span class="o">=</span><span class="n">ofname</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1240</span>     <span class="n">marker</span><span class="o">=</span><span class="n">marker</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1241</span> <span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1242</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dot</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">outer_ring</span><span class="o">=</span><span class="n">show_ring</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1244</span> <span class="k">if</span> <span class="n">xticklabels_rot</span><span class="p">:</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:692,</span> in <span class="ni">DotPlot.__init__</span><span class="nt">(self, df, x, y, hue, dot_scale, x_order, y_order, thresh, n_terms, title, ax, figsize, cmap, ofname, marker, **kwargs)</span>
<span class="g g-Whitespace">    </span><span class="mi">690</span> <span class="bp">self</span><span class="o">.</span><span class="n">n_terms</span> <span class="o">=</span> <span class="n">n_terms</span>
<span class="g g-Whitespace">    </span><span class="mi">691</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span> <span class="o">=</span> <span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">692</span> <span class="bp">self</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">693</span> <span class="n">plt</span><span class="o">.</span><span class="n">rcParams</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="s2">&quot;pdf.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">,</span> <span class="s2">&quot;ps.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">})</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:717,</span> in <span class="ni">DotPlot.process</span><span class="nt">(self, df)</span>
<span class="g g-Whitespace">    </span><span class="mi">715</span> <span class="k">if</span> <span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">716</span>     <span class="n">msg</span> <span class="o">=</span> <span class="s2">&quot;Warning: No enrich terms when cutoff = </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">717</span>     <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="n">msg</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">718</span> <span class="bp">self</span><span class="o">.</span><span class="n">cbar_title</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">colname</span>
<span class="g g-Whitespace">    </span><span class="mi">719</span> <span class="c1"># clip GSEA lower bounds</span>
<span class="g g-Whitespace">    </span><span class="mi">720</span> <span class="c1"># if self.colname in [&quot;NOM p-val&quot;, &quot;FDR q-val&quot;]:</span>
<span class="g g-Whitespace">    </span><span class="mi">721</span> <span class="c1">#     df[self.colname].clip(1e-5, 1.0, inplace=True)</span>
<span class="g g-Whitespace">    </span><span class="mi">722</span> <span class="c1"># sorting the dataframe for better visualization</span>

<span class="ne">ValueError</span>: Warning: No enrich terms when cutoff = 0.05
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_9&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Gene Ontology Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_ascl1AT2_9.png" alt="gsea_ascl1AT2_9" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Gene Ontology Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_ascl1AT2_9.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id25">
<h3>GSEA - Pathway Analysis [Top 250 &amp; Down 250]<a class="headerlink" href="#id25" title="Link to this heading">#</a></h3>
<p>‘Reactome_Pathways_2024’,’KEGG_2021_Human’,’WikiPathways_2024_Human’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Reactome_Pathways_2024&#39;</span><span class="p">,</span><span class="s1">&#39;KEGG_2021_Human&#39;</span><span class="p">,</span><span class="s1">&#39;WikiPathways_2024_Human&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Diseases of Metabolism</td>
      <td>-0.648115</td>
      <td>-2.633465</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>11/19</td>
      <td>13.60%</td>
      <td>SFTPC;GPC5;ADAMTSL1;SFTPA1;SLC34A2;SFTPB;ABCA3...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Surfactant Metabolism</td>
      <td>-0.77712</td>
      <td>-2.408708</td>
      <td>0.0</td>
      <td>0.000965</td>
      <td>0.002</td>
      <td>6/9</td>
      <td>5.20%</td>
      <td>SFTPC;SFTPA1;ADGRF5;SLC34A2;SFTPB;ABCA3</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Diseases Associated Wi...</td>
      <td>-0.959596</td>
      <td>-2.360726</td>
      <td>0.0</td>
      <td>0.001286</td>
      <td>0.003</td>
      <td>5/5</td>
      <td>5.20%</td>
      <td>SFTPC;SFTPA1;SLC34A2;SFTPB;ABCA3</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>WikiPathways_2024_Human__Complement System WP2806</td>
      <td>-0.722672</td>
      <td>-1.864206</td>
      <td>0.006981</td>
      <td>0.342976</td>
      <td>0.538</td>
      <td>6/6</td>
      <td>28.80%</td>
      <td>SFTPA1;CD55;ICAM1;ITGA2;PRKCA;PROS1</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>WikiPathways_2024_Human__Spinal Cord Injury WP...</td>
      <td>-0.601117</td>
      <td>-1.812836</td>
      <td>0.009868</td>
      <td>0.421798</td>
      <td>0.661</td>
      <td>7/9</td>
      <td>26.80%</td>
      <td>ROS1;EGFR;SLIT2;ICAM1;ANXA1;PRKCA;CD47</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>437</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Death Receptor Signaling</td>
      <td>0.163968</td>
      <td>0.4618</td>
      <td>0.995181</td>
      <td>0.994091</td>
      <td>1.0</td>
      <td>6/6</td>
      <td>83.80%</td>
      <td>KALRN;MCF2L;NET1;TAX1BP1;ARHGEF26;AKAP13</td>
    </tr>
    <tr>
      <th>438</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Cell Death Signalling ...</td>
      <td>-0.184694</td>
      <td>-0.448537</td>
      <td>0.99651</td>
      <td>0.992825</td>
      <td>1.0</td>
      <td>2/5</td>
      <td>30.60%</td>
      <td>AKAP13;ARHGEF26</td>
    </tr>
    <tr>
      <th>439</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__G Alpha (12 13) Signal...</td>
      <td>-0.184694</td>
      <td>-0.448537</td>
      <td>0.99651</td>
      <td>0.992825</td>
      <td>1.0</td>
      <td>2/5</td>
      <td>30.60%</td>
      <td>AKAP13;ARHGEF26</td>
    </tr>
    <tr>
      <th>440</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__P75 NTR Receptor-Media...</td>
      <td>-0.184694</td>
      <td>-0.448537</td>
      <td>0.99651</td>
      <td>0.992825</td>
      <td>1.0</td>
      <td>2/5</td>
      <td>30.60%</td>
      <td>AKAP13;ARHGEF26</td>
    </tr>
    <tr>
      <th>441</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__NRAGE Signals Death Th...</td>
      <td>-0.184694</td>
      <td>-0.448537</td>
      <td>0.99651</td>
      <td>0.992825</td>
      <td>1.0</td>
      <td>2/5</td>
      <td>30.60%</td>
      <td>AKAP13;ARHGEF26</td>
    </tr>
  </tbody>
</table>
<p>442 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s2">&quot;GSEA Pathway Analysis - &#39;ASCL1+&#39; AT2 Like Cells&quot;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_10&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Pathway Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_ascl1AT2_10.png" alt="gsea_ascl1AT2_10" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Pathway Analysis Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_ascl1AT2_10.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_11&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Pathway Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_ascl1AT2_11.png" alt="gsea_ascl1AT2_11" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Pathway Analysis Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_ascl1AT2_11.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id26">
<h3>GSEA - Pathway Analysis [Top 500]<a class="headerlink" href="#id26" title="Link to this heading">#</a></h3>
<p>‘Reactome_Pathways_2024’,’KEGG_2021_Human’,’WikiPathways_2024_Human’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list_top</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Reactome_Pathways_2024&#39;</span><span class="p">,</span><span class="s1">&#39;KEGG_2021_Human&#39;</span><span class="p">,</span><span class="s1">&#39;WikiPathways_2024_Human&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>KEGG_2021_Human__FoxO signaling pathway</td>
      <td>-0.557695</td>
      <td>-1.970179</td>
      <td>0.004484</td>
      <td>0.569352</td>
      <td>0.302</td>
      <td>7/8</td>
      <td>30.40%</td>
      <td>BNIP3;CCND1;GADD45A;PRKAA2;BCL2L11;CCND2;PIK3R3</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>WikiPathways_2024_Human__Fragile X Syndrome WP...</td>
      <td>0.76598</td>
      <td>1.857154</td>
      <td>0.001299</td>
      <td>0.990223</td>
      <td>0.411</td>
      <td>5/6</td>
      <td>18.40%</td>
      <td>PLCB1;ARHGAP32;MAP1B;FYN;APP</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>KEGG_2021_Human__Estrogen signaling pathway</td>
      <td>0.670573</td>
      <td>1.753561</td>
      <td>0.007692</td>
      <td>1.0</td>
      <td>0.696</td>
      <td>5/7</td>
      <td>19.60%</td>
      <td>PLCB1;ITPR2;PLCB4;CALM1;BCL2</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__G Alpha (Q) Signalling...</td>
      <td>0.646287</td>
      <td>1.738643</td>
      <td>0.012706</td>
      <td>1.0</td>
      <td>0.732</td>
      <td>6/8</td>
      <td>21.80%</td>
      <td>PLCB1;ITPR2;KALRN;PLCB4;APP;PRKCE</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Opioid Signalling</td>
      <td>0.750314</td>
      <td>1.738546</td>
      <td>0.002692</td>
      <td>0.514851</td>
      <td>0.732</td>
      <td>4/5</td>
      <td>17.60%</td>
      <td>PLCB1;ITPR2;PLCB4;CALM1</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>397</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__Metabolism</td>
      <td>0.116844</td>
      <td>0.480577</td>
      <td>0.9652</td>
      <td>0.994687</td>
      <td>1.0</td>
      <td>2/40</td>
      <td>2.80%</td>
      <td>PLCB1;ITPR2</td>
    </tr>
    <tr>
      <th>398</th>
      <td>prerank</td>
      <td>Reactome_Pathways_2024__PIP3 Activates AKT Sig...</td>
      <td>0.161819</td>
      <td>0.476272</td>
      <td>0.979452</td>
      <td>0.992768</td>
      <td>1.0</td>
      <td>2/11</td>
      <td>16.00%</td>
      <td>ERBB4;FYN</td>
    </tr>
    <tr>
      <th>399</th>
      <td>prerank</td>
      <td>KEGG_2021_Human__T cell receptor signaling pat...</td>
      <td>0.182309</td>
      <td>0.452928</td>
      <td>0.992395</td>
      <td>0.995673</td>
      <td>1.0</td>
      <td>5/6</td>
      <td>70.00%</td>
      <td>FYN;GSK3B;JUN;MALT1;PIK3R3</td>
    </tr>
    <tr>
      <th>400</th>
      <td>prerank</td>
      <td>KEGG_2021_Human__cAMP signaling pathway</td>
      <td>0.181917</td>
      <td>0.447862</td>
      <td>0.993473</td>
      <td>0.993489</td>
      <td>1.0</td>
      <td>5/6</td>
      <td>70.00%</td>
      <td>CALM1;ATP2A3;JUN;RAPGEF4;PIK3R3</td>
    </tr>
    <tr>
      <th>401</th>
      <td>prerank</td>
      <td>KEGG_2021_Human__Endocytosis</td>
      <td>0.166684</td>
      <td>0.446863</td>
      <td>0.993857</td>
      <td>0.9905</td>
      <td>1.0</td>
      <td>1/8</td>
      <td>8.80%</td>
      <td>RAB11FIP1</td>
    </tr>
  </tbody>
</table>
<p>402 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s2">&quot;GSEA Pathway Analysis - &#39;ASCL1+&#39; AT2 Like Cells&quot;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_12&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Pathway Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">ValueError</span><span class="g g-Whitespace">                                </span>Traceback (most recent call last)
<span class="n">Cell</span> <span class="n">In</span><span class="p">[</span><span class="mi">223</span><span class="p">],</span> <span class="n">line</span> <span class="mi">2</span>
<span class="g g-Whitespace">      </span><span class="mi">1</span> <span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="ne">----&gt; </span><span class="mi">2</span> <span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">3</span>                  <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">4</span>                  <span class="n">title</span><span class="o">=</span><span class="s2">&quot;GSEA Pathway Analysis - &#39;ASCL1+&#39; AT2 Like Cells&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">5</span>                  <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">6</span>                  <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">7</span>                  <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
<span class="g g-Whitespace">      </span><span class="mi">8</span>                  <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">9</span> <span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="g g-Whitespace">     </span><span class="mi">10</span> <span class="n">sc_plot_and_embed</span><span class="p">(</span>
<span class="g g-Whitespace">     </span><span class="mi">11</span>     <span class="n">ax_dot</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">12</span>     <span class="s2">&quot;gsea_ascl1AT2_12&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">13</span>     <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Pathway Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">14</span>     <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="g g-Whitespace">     </span><span class="mi">15</span> <span class="p">)</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:1225,</span> in <span class="ni">dotplot</span><span class="nt">(df, column, x, y, x_order, y_order, title, cutoff, top_term, size, ax, figsize, cmap, ofname, xticklabels_rot, yticklabels_rot, marker, show_ring, **kwargs)</span>
<span class="g g-Whitespace">   </span><span class="mi">1222</span>     <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span><span class="s2">&quot;group is deprecated; use x instead&quot;</span><span class="p">,</span> <span class="ne">DeprecationWarning</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1223</span>     <span class="k">return</span>
<span class="ne">-&gt; </span><span class="mi">1225</span> <span class="n">dot</span> <span class="o">=</span> <span class="n">DotPlot</span><span class="p">(</span>
<span class="g g-Whitespace">   </span><span class="mi">1226</span>     <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1227</span>     <span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1228</span>     <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1229</span>     <span class="n">x_order</span><span class="o">=</span><span class="n">x_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1230</span>     <span class="n">y_order</span><span class="o">=</span><span class="n">y_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1231</span>     <span class="n">hue</span><span class="o">=</span><span class="n">column</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1232</span>     <span class="n">title</span><span class="o">=</span><span class="n">title</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1233</span>     <span class="n">thresh</span><span class="o">=</span><span class="n">cutoff</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1234</span>     <span class="n">n_terms</span><span class="o">=</span><span class="nb">int</span><span class="p">(</span><span class="n">top_term</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1235</span>     <span class="n">dot_scale</span><span class="o">=</span><span class="n">size</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1236</span>     <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1237</span>     <span class="n">figsize</span><span class="o">=</span><span class="n">figsize</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1238</span>     <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1239</span>     <span class="n">ofname</span><span class="o">=</span><span class="n">ofname</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1240</span>     <span class="n">marker</span><span class="o">=</span><span class="n">marker</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1241</span> <span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1242</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dot</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">outer_ring</span><span class="o">=</span><span class="n">show_ring</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1244</span> <span class="k">if</span> <span class="n">xticklabels_rot</span><span class="p">:</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:692,</span> in <span class="ni">DotPlot.__init__</span><span class="nt">(self, df, x, y, hue, dot_scale, x_order, y_order, thresh, n_terms, title, ax, figsize, cmap, ofname, marker, **kwargs)</span>
<span class="g g-Whitespace">    </span><span class="mi">690</span> <span class="bp">self</span><span class="o">.</span><span class="n">n_terms</span> <span class="o">=</span> <span class="n">n_terms</span>
<span class="g g-Whitespace">    </span><span class="mi">691</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span> <span class="o">=</span> <span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">692</span> <span class="bp">self</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">693</span> <span class="n">plt</span><span class="o">.</span><span class="n">rcParams</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="s2">&quot;pdf.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">,</span> <span class="s2">&quot;ps.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">})</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:717,</span> in <span class="ni">DotPlot.process</span><span class="nt">(self, df)</span>
<span class="g g-Whitespace">    </span><span class="mi">715</span> <span class="k">if</span> <span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">716</span>     <span class="n">msg</span> <span class="o">=</span> <span class="s2">&quot;Warning: No enrich terms when cutoff = </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">717</span>     <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="n">msg</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">718</span> <span class="bp">self</span><span class="o">.</span><span class="n">cbar_title</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">colname</span>
<span class="g g-Whitespace">    </span><span class="mi">719</span> <span class="c1"># clip GSEA lower bounds</span>
<span class="g g-Whitespace">    </span><span class="mi">720</span> <span class="c1"># if self.colname in [&quot;NOM p-val&quot;, &quot;FDR q-val&quot;]:</span>
<span class="g g-Whitespace">    </span><span class="mi">721</span> <span class="c1">#     df[self.colname].clip(1e-5, 1.0, inplace=True)</span>
<span class="g g-Whitespace">    </span><span class="mi">722</span> <span class="c1"># sorting the dataframe for better visualization</span>

<span class="ne">ValueError</span>: Warning: No enrich terms when cutoff = 0.05
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_13&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Pathway Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_ascl1AT2_13.png" alt="gsea_ascl1AT2_13" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Pathway Analysis Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_ascl1AT2_13.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id27">
<h3>GSEA - Proteomic Analysis [Top 250 &amp; Down 250]<a class="headerlink" href="#id27" title="Link to this heading">#</a></h3>
<p>‘ProteomicsDB_2020’,’Tissue_Protein_Expression_from_Human_Proteome_Map’,’Tissue_Protein_Expression_from_ProteomicsDB’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;ProteomicsDB_2020&#39;</span><span class="p">,</span><span class="s1">&#39;Tissue_Protein_Expression_from_Human_Proteome_Map&#39;</span><span class="p">,</span><span class="s1">&#39;Tissue_Protein_Expression_from_ProteomicsDB&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_ProteomicsDB__K...</td>
      <td>0.576244</td>
      <td>2.513038</td>
      <td>0.0</td>
      <td>0.014596</td>
      <td>0.008</td>
      <td>8/16</td>
      <td>9.20%</td>
      <td>DACH1;ROBO1;NKD1;SYT1;ARHGAP32;NOL4;CMIP;MACC1</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Breast HMT-3522 BTO:0003884...</td>
      <td>0.594929</td>
      <td>2.170591</td>
      <td>0.0</td>
      <td>0.185857</td>
      <td>0.178</td>
      <td>8/11</td>
      <td>23.60%</td>
      <td>DSP;IGFBP5;CBFA2T2;NEBL;CHD6;NAV2;CTNNA2;NFIA</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Breast MCF-7 BTO:0000093 HM...</td>
      <td>0.594929</td>
      <td>2.170591</td>
      <td>0.0</td>
      <td>0.185857</td>
      <td>0.178</td>
      <td>8/11</td>
      <td>23.60%</td>
      <td>DSP;IGFBP5;CBFA2T2;NEBL;CHD6;NAV2;CTNNA2;NFIA</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_Human_Proteome_...</td>
      <td>-0.581839</td>
      <td>-2.083726</td>
      <td>0.00159</td>
      <td>0.420847</td>
      <td>0.238</td>
      <td>9/14</td>
      <td>22.20%</td>
      <td>SFTPC;SFTPA1;SLC34A2;SFTPB;LAMP3;ATXN1;CLDN18;...</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_Human_Proteome_...</td>
      <td>0.560039</td>
      <td>2.07171</td>
      <td>0.010582</td>
      <td>0.306518</td>
      <td>0.332</td>
      <td>7/11</td>
      <td>15.00%</td>
      <td>PTPRN2;NKD1;CTNND2;MACC1;NRXN1;SMOC2;APBB2</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>1002</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_ProteomicsDB__M...</td>
      <td>-0.172754</td>
      <td>-0.450108</td>
      <td>0.994983</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>4/6</td>
      <td>53.00%</td>
      <td>EXT1;WWTR1;RBMS1;SKIL</td>
    </tr>
    <tr>
      <th>1003</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Melanoma A-375 BTO:0002806 ...</td>
      <td>-0.124473</td>
      <td>-0.438758</td>
      <td>0.987013</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>6/12</td>
      <td>42.80%</td>
      <td>CD55;TKT;GPRC5A;IQGAP1;ACTB;RBMS1</td>
    </tr>
    <tr>
      <th>1004</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Cervix HeLa BTO:0000567 HeL...</td>
      <td>-0.128946</td>
      <td>-0.426546</td>
      <td>0.996656</td>
      <td>0.999684</td>
      <td>1.0</td>
      <td>7/11</td>
      <td>51.60%</td>
      <td>CD55;SLC39A8;PON2;MCU;TSPAN13;PAPSS2;DDX5</td>
    </tr>
    <tr>
      <th>1005</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Mixed PDB:200045 gcpool scx...</td>
      <td>-0.102134</td>
      <td>-0.376183</td>
      <td>1.0</td>
      <td>0.999041</td>
      <td>1.0</td>
      <td>1/14</td>
      <td>4.20%</td>
      <td>LAMA3</td>
    </tr>
    <tr>
      <th>1006</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Gastric HFE-145 PDB:200047 ...</td>
      <td>-0.102134</td>
      <td>-0.376183</td>
      <td>1.0</td>
      <td>0.999041</td>
      <td>1.0</td>
      <td>1/14</td>
      <td>4.20%</td>
      <td>LAMA3</td>
    </tr>
  </tbody>
</table>
<p>1007 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s2">&quot;GSEA Proteomic Analysis - &#39;ASCL1+&#39; AT2 Like Cells&quot;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_14&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Proteomic Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_ascl1AT2_14.png" alt="gsea_ascl1AT2_14" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Proteomic Analysis Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_ascl1AT2_14.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_15&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Proteomic Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_ascl1AT2_15.png" alt="gsea_ascl1AT2_15" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Proteomic Analysis Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_ascl1AT2_15.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id28">
<h3>GSEA - Proteomic Analysis [Top 500]<a class="headerlink" href="#id28" title="Link to this heading">#</a></h3>
<p>‘ProteomicsDB_2020’,’Tissue_Protein_Expression_from_Human_Proteome_Map’,’Tissue_Protein_Expression_from_ProteomicsDB’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list_top</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;ProteomicsDB_2020&#39;</span><span class="p">,</span><span class="s1">&#39;Tissue_Protein_Expression_from_Human_Proteome_Map&#39;</span><span class="p">,</span><span class="s1">&#39;Tissue_Protein_Expression_from_ProteomicsDB&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_ProteomicsDB__L...</td>
      <td>-0.568228</td>
      <td>-2.11312</td>
      <td>0.0</td>
      <td>0.178104</td>
      <td>0.152</td>
      <td>9/9</td>
      <td>44.40%</td>
      <td>PCMTD2;RGS12;SLC37A1;MAF;BCL2L11;ANKH;RFX3;STX...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_Human_Proteome_...</td>
      <td>0.669842</td>
      <td>2.034479</td>
      <td>0.0</td>
      <td>0.485567</td>
      <td>0.148</td>
      <td>7/11</td>
      <td>15.00%</td>
      <td>PTPRN2;NKD1;CTNND2;MACC1;NRXN1;SMOC2;APBB2</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_Human_Proteome_...</td>
      <td>-0.509165</td>
      <td>-1.933243</td>
      <td>0.009615</td>
      <td>0.243986</td>
      <td>0.345</td>
      <td>9/9</td>
      <td>50.20%</td>
      <td>BNIP3;NOVA1;DISC1;SCN3A;LRRN1;RNF19A;SMYD2;CAS...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_ProteomicsDB__K...</td>
      <td>0.502071</td>
      <td>1.846742</td>
      <td>0.00325</td>
      <td>1.0</td>
      <td>0.655</td>
      <td>8/23</td>
      <td>9.20%</td>
      <td>DACH1;ROBO1;NKD1;SYT1;ARHGAP32;NOL4;CMIP;MACC1</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>Tissue_Protein_Expression_from_ProteomicsDB__S...</td>
      <td>-0.482688</td>
      <td>-1.731073</td>
      <td>0.016484</td>
      <td>0.463532</td>
      <td>0.683</td>
      <td>9/9</td>
      <td>52.80%</td>
      <td>BNIP3;AGPAT4;ID4;LRRC8D;GLDC;DNAH14;KSR1;DENND...</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>1112</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Embryonic stem hES H14 PDB:...</td>
      <td>0.103454</td>
      <td>0.359292</td>
      <td>0.998899</td>
      <td>0.999401</td>
      <td>1.0</td>
      <td>1/20</td>
      <td>3.20%</td>
      <td>CBFA2T2</td>
    </tr>
    <tr>
      <th>1113</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Human iPS iPS DF19.7 PDB:20...</td>
      <td>0.103454</td>
      <td>0.359292</td>
      <td>0.998899</td>
      <td>0.999401</td>
      <td>1.0</td>
      <td>1/20</td>
      <td>3.20%</td>
      <td>CBFA2T2</td>
    </tr>
    <tr>
      <th>1114</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Human iPS iPS DF6.9 PDB:200...</td>
      <td>0.103454</td>
      <td>0.359292</td>
      <td>0.998899</td>
      <td>0.999401</td>
      <td>1.0</td>
      <td>1/20</td>
      <td>3.20%</td>
      <td>CBFA2T2</td>
    </tr>
    <tr>
      <th>1115</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Human iPS iPS DF19.11 PDB:2...</td>
      <td>0.103454</td>
      <td>0.359292</td>
      <td>0.998899</td>
      <td>0.999401</td>
      <td>1.0</td>
      <td>1/20</td>
      <td>3.20%</td>
      <td>CBFA2T2</td>
    </tr>
    <tr>
      <th>1116</th>
      <td>prerank</td>
      <td>ProteomicsDB_2020__Colon SW837 PDB:200164 SW83...</td>
      <td>0.09179</td>
      <td>0.331056</td>
      <td>1.0</td>
      <td>0.999469</td>
      <td>1.0</td>
      <td>1/22</td>
      <td>3.20%</td>
      <td>CBFA2T2</td>
    </tr>
  </tbody>
</table>
<p>1117 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s2">&quot;GSEA Proteomic Analysis - &#39;ASCL1+&#39; AT2 Like Cells&quot;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_16&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Proteomic Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">ValueError</span><span class="g g-Whitespace">                                </span>Traceback (most recent call last)
<span class="n">Cell</span> <span class="n">In</span><span class="p">[</span><span class="mi">232</span><span class="p">],</span> <span class="n">line</span> <span class="mi">2</span>
<span class="g g-Whitespace">      </span><span class="mi">1</span> <span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="ne">----&gt; </span><span class="mi">2</span> <span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">3</span>                  <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">4</span>                  <span class="n">title</span><span class="o">=</span><span class="s2">&quot;GSEA Proteomic Analysis - &#39;ASCL1+&#39; AT2 Like Cells&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">5</span>                  <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">6</span>                  <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">7</span>                  <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
<span class="g g-Whitespace">      </span><span class="mi">8</span>                  <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">9</span> <span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="g g-Whitespace">     </span><span class="mi">10</span> <span class="n">sc_plot_and_embed</span><span class="p">(</span>
<span class="g g-Whitespace">     </span><span class="mi">11</span>     <span class="n">ax_dot</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">12</span>     <span class="s2">&quot;gsea_ascl1AT2_16&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">13</span>     <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Proteomic Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">14</span>     <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="g g-Whitespace">     </span><span class="mi">15</span> <span class="p">)</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:1225,</span> in <span class="ni">dotplot</span><span class="nt">(df, column, x, y, x_order, y_order, title, cutoff, top_term, size, ax, figsize, cmap, ofname, xticklabels_rot, yticklabels_rot, marker, show_ring, **kwargs)</span>
<span class="g g-Whitespace">   </span><span class="mi">1222</span>     <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span><span class="s2">&quot;group is deprecated; use x instead&quot;</span><span class="p">,</span> <span class="ne">DeprecationWarning</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1223</span>     <span class="k">return</span>
<span class="ne">-&gt; </span><span class="mi">1225</span> <span class="n">dot</span> <span class="o">=</span> <span class="n">DotPlot</span><span class="p">(</span>
<span class="g g-Whitespace">   </span><span class="mi">1226</span>     <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1227</span>     <span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1228</span>     <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1229</span>     <span class="n">x_order</span><span class="o">=</span><span class="n">x_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1230</span>     <span class="n">y_order</span><span class="o">=</span><span class="n">y_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1231</span>     <span class="n">hue</span><span class="o">=</span><span class="n">column</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1232</span>     <span class="n">title</span><span class="o">=</span><span class="n">title</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1233</span>     <span class="n">thresh</span><span class="o">=</span><span class="n">cutoff</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1234</span>     <span class="n">n_terms</span><span class="o">=</span><span class="nb">int</span><span class="p">(</span><span class="n">top_term</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1235</span>     <span class="n">dot_scale</span><span class="o">=</span><span class="n">size</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1236</span>     <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1237</span>     <span class="n">figsize</span><span class="o">=</span><span class="n">figsize</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1238</span>     <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1239</span>     <span class="n">ofname</span><span class="o">=</span><span class="n">ofname</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1240</span>     <span class="n">marker</span><span class="o">=</span><span class="n">marker</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1241</span> <span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1242</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dot</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">outer_ring</span><span class="o">=</span><span class="n">show_ring</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1244</span> <span class="k">if</span> <span class="n">xticklabels_rot</span><span class="p">:</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:692,</span> in <span class="ni">DotPlot.__init__</span><span class="nt">(self, df, x, y, hue, dot_scale, x_order, y_order, thresh, n_terms, title, ax, figsize, cmap, ofname, marker, **kwargs)</span>
<span class="g g-Whitespace">    </span><span class="mi">690</span> <span class="bp">self</span><span class="o">.</span><span class="n">n_terms</span> <span class="o">=</span> <span class="n">n_terms</span>
<span class="g g-Whitespace">    </span><span class="mi">691</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span> <span class="o">=</span> <span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">692</span> <span class="bp">self</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">693</span> <span class="n">plt</span><span class="o">.</span><span class="n">rcParams</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="s2">&quot;pdf.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">,</span> <span class="s2">&quot;ps.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">})</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:717,</span> in <span class="ni">DotPlot.process</span><span class="nt">(self, df)</span>
<span class="g g-Whitespace">    </span><span class="mi">715</span> <span class="k">if</span> <span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">716</span>     <span class="n">msg</span> <span class="o">=</span> <span class="s2">&quot;Warning: No enrich terms when cutoff = </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">717</span>     <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="n">msg</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">718</span> <span class="bp">self</span><span class="o">.</span><span class="n">cbar_title</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">colname</span>
<span class="g g-Whitespace">    </span><span class="mi">719</span> <span class="c1"># clip GSEA lower bounds</span>
<span class="g g-Whitespace">    </span><span class="mi">720</span> <span class="c1"># if self.colname in [&quot;NOM p-val&quot;, &quot;FDR q-val&quot;]:</span>
<span class="g g-Whitespace">    </span><span class="mi">721</span> <span class="c1">#     df[self.colname].clip(1e-5, 1.0, inplace=True)</span>
<span class="g g-Whitespace">    </span><span class="mi">722</span> <span class="c1"># sorting the dataframe for better visualization</span>

<span class="ne">ValueError</span>: Warning: No enrich terms when cutoff = 0.05
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_17&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Proteomic Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_ascl1AT2_17.png" alt="gsea_ascl1AT2_17" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Proteomic Analysis Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_ascl1AT2_17.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id29">
<h3>GSEA - Disease Analysis [Top 250 &amp; Down 250]<a class="headerlink" href="#id29" title="Link to this heading">#</a></h3>
<p>‘DisGeNET’,’Disease_Perturbations_from_GEO_down’,’Disease_Perturbations_from_GEO_up’,’HDSigDB_Human_2021’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;DisGeNET&#39;</span><span class="p">,</span><span class="s1">&#39;Disease_Perturbations_from_GEO_down&#39;</span><span class="p">,</span><span class="s1">&#39;Disease_Perturbations_from_GEO_up&#39;</span><span class="p">,</span><span class="s1">&#39;HDSigDB_Human_2021&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_up__large cell ...</td>
      <td>0.642512</td>
      <td>2.813549</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>15/17</td>
      <td>31.60%</td>
      <td>NKAIN2;SEC11C;PROX1;PTPRN2;SYT1;NOL4;KCNH8;PCS...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_up__Spinal Musc...</td>
      <td>0.640107</td>
      <td>2.796659</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>16/18</td>
      <td>31.60%</td>
      <td>NKAIN2;PTPRN2;SYT1;CTNND2;DOCK3;GALNT13;NRXN1;...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_down__Squamous ...</td>
      <td>-0.517252</td>
      <td>-2.775469</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>41/47</td>
      <td>43.60%</td>
      <td>SFTPC;LPCAT1;SLC34A2;SFTPB;ABCA3;CD55;LAMP3;PI...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Up-Regulated In Frie...</td>
      <td>-0.375058</td>
      <td>-2.692336</td>
      <td>0.0</td>
      <td>0.000969</td>
      <td>0.002</td>
      <td>140/221</td>
      <td>46.20%</td>
      <td>DLC1;SFTPC;GPC5;ROS1;LPCAT1;COL8A1;SFTPA1;ADGR...</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Top Neuron-Enriched Genes ...</td>
      <td>0.338401</td>
      <td>2.54557</td>
      <td>0.0</td>
      <td>0.016551</td>
      <td>0.042</td>
      <td>49/81</td>
      <td>32.60%</td>
      <td>IGFBP5;PROX1;ROBO1;PTPRN2;PLCB1;TOX3;SYT1;MGAT...</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>4304</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Expression In Striatum Fro...</td>
      <td>0.062965</td>
      <td>0.362156</td>
      <td>1.0</td>
      <td>0.999837</td>
      <td>1.0</td>
      <td>2/35</td>
      <td>4.00%</td>
      <td>PLCB1;SYT1</td>
    </tr>
    <tr>
      <th>4305</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_up__Type 1 diab...</td>
      <td>-0.115802</td>
      <td>-0.353248</td>
      <td>0.998342</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>4/9</td>
      <td>39.20%</td>
      <td>MBNL1;NEK7;ZBTB16;SSR3</td>
    </tr>
    <tr>
      <th>4306</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_down__systemic ...</td>
      <td>-0.134894</td>
      <td>-0.350541</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>1/6</td>
      <td>10.80%</td>
      <td>PID1</td>
    </tr>
    <tr>
      <th>4307</th>
      <td>prerank</td>
      <td>DisGeNET__Adenocarcinoma, Endometrioid</td>
      <td>-0.115215</td>
      <td>-0.349203</td>
      <td>1.0</td>
      <td>0.999975</td>
      <td>1.0</td>
      <td>5/9</td>
      <td>48.00%</td>
      <td>EGFR;ETV5;AHR;MKI67;SPARC</td>
    </tr>
    <tr>
      <th>4308</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_down__ulcerativ...</td>
      <td>-0.099259</td>
      <td>-0.315424</td>
      <td>1.0</td>
      <td>0.999916</td>
      <td>1.0</td>
      <td>5/10</td>
      <td>45.80%</td>
      <td>MAGI3;SOX5;EPHA6;LDLR;MUC1</td>
    </tr>
  </tbody>
</table>
<p>4309 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s2">&quot;GSEA Disease Analysis - &#39;ASCL1+&#39; AT2 Like Cells&quot;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_18&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Disease Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_ascl1AT2_18.png" alt="gsea_ascl1AT2_18" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA DotPlot of Disease Analysis Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_ascl1AT2_18.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_19&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Disease Analysis Datasets [Top 250 &amp; Down 250 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_ascl1AT2_19.png" alt="gsea_ascl1AT2_19" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Disease Analysis Datasets [Top 250 & Down 250 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_ascl1AT2_19.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
<section id="id30">
<h3>GSEA - Disease Analysis [Top 500]<a class="headerlink" href="#id30" title="Link to this heading">#</a></h3>
<p>‘DisGeNET’,’Disease_Perturbations_from_GEO_down’,’Disease_Perturbations_from_GEO_up’,’HDSigDB_Human_2021’</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">pre_res</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">prerank</span><span class="p">(</span>
                        <span class="n">rnk</span><span class="o">=</span><span class="n">df_ranked_list_top</span><span class="p">,</span>
                        <span class="n">gene_sets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;DisGeNET&#39;</span><span class="p">,</span><span class="s1">&#39;Disease_Perturbations_from_GEO_down&#39;</span><span class="p">,</span><span class="s1">&#39;Disease_Perturbations_from_GEO_up&#39;</span><span class="p">,</span><span class="s1">&#39;HDSigDB_Human_2021&#39;</span><span class="p">],</span>
                        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;gsea_lateAT2&#39;</span><span class="p">,</span>
                        <span class="n">min_size</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                        <span class="n">max_size</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span>
                        <span class="n">permutation_num</span><span class="o">=</span><span class="mi">1000</span>
<span class="p">)</span>
<span class="n">res_df</span> <span class="o">=</span> <span class="n">pre_res</span><span class="o">.</span><span class="n">res2d</span>  <span class="c1"># your enriched pathways data</span>
<span class="n">res_df</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Term</th>
      <th>ES</th>
      <th>NES</th>
      <th>NOM p-val</th>
      <th>FDR q-val</th>
      <th>FWER p-val</th>
      <th>Tag %</th>
      <th>Gene %</th>
      <th>Lead_genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Changed In OPCs Of z...</td>
      <td>0.591014</td>
      <td>2.148172</td>
      <td>0.0</td>
      <td>0.235671</td>
      <td>0.176</td>
      <td>16/23</td>
      <td>28.20%</td>
      <td>NKAIN2;ROBO1;ABHD2;SYT1;DENND1A;RERE;ERBB4;NRX...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Changed In Oligodend...</td>
      <td>0.562616</td>
      <td>2.082987</td>
      <td>0.001056</td>
      <td>0.302361</td>
      <td>0.36</td>
      <td>13/24</td>
      <td>18.60%</td>
      <td>NKAIN2;ROBO1;SYT1;MGAT4C;CTNND2;ANK2;PLCL1;NEG...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Changed In Microglia...</td>
      <td>0.650456</td>
      <td>2.082437</td>
      <td>0.0</td>
      <td>0.151933</td>
      <td>0.361</td>
      <td>8/14</td>
      <td>12.60%</td>
      <td>NKAIN2;ROBO1;SYT1;CTNND2;ANK2;NEGR1;NRXN1;LRRC4C</td>
    </tr>
    <tr>
      <th>3</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Up-Regulated In Micr...</td>
      <td>0.650456</td>
      <td>2.082437</td>
      <td>0.0</td>
      <td>0.151933</td>
      <td>0.361</td>
      <td>8/14</td>
      <td>12.60%</td>
      <td>NKAIN2;ROBO1;SYT1;CTNND2;ANK2;NEGR1;NRXN1;LRRC4C</td>
    </tr>
    <tr>
      <th>4</th>
      <td>prerank</td>
      <td>DisGeNET__Narcolepsy</td>
      <td>0.640691</td>
      <td>2.078273</td>
      <td>0.0</td>
      <td>0.131775</td>
      <td>0.389</td>
      <td>11/15</td>
      <td>25.40%</td>
      <td>NKAIN2;CTNND2;NOL4;NRXN1;LRRC4C;SDK1;KALRN;LIN...</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>4002</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Up-Regulated In HD G...</td>
      <td>0.085914</td>
      <td>0.35167</td>
      <td>0.993769</td>
      <td>0.999954</td>
      <td>1.0</td>
      <td>2/37</td>
      <td>4.60%</td>
      <td>MSI2;NEBL</td>
    </tr>
    <tr>
      <th>4003</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Neuron Gene Module M11 Fro...</td>
      <td>0.111347</td>
      <td>0.341474</td>
      <td>0.998832</td>
      <td>1.0</td>
      <td>1.0</td>
      <td>7/12</td>
      <td>58.20%</td>
      <td>NRXN1;CALM1;GGH;CACNA2D1;CCSER2;CHGB;TPD52</td>
    </tr>
    <tr>
      <th>4004</th>
      <td>prerank</td>
      <td>Disease_Perturbations_from_GEO_up__Dehydration...</td>
      <td>0.108531</td>
      <td>0.335075</td>
      <td>0.998841</td>
      <td>0.99987</td>
      <td>1.0</td>
      <td>2/12</td>
      <td>17.60%</td>
      <td>TSPAN5;CALM1</td>
    </tr>
    <tr>
      <th>4005</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Genes Changed In HDACi 4B ...</td>
      <td>0.083878</td>
      <td>0.324961</td>
      <td>0.998959</td>
      <td>0.999786</td>
      <td>1.0</td>
      <td>16/31</td>
      <td>56.60%</td>
      <td>ARHGAP32;GSE1;FYN;ID2;MYO10;SLC1A5;SIPA1L2;NAV...</td>
    </tr>
    <tr>
      <th>4006</th>
      <td>prerank</td>
      <td>HDSigDB_Human_2021__Mbf1 Suppressor Deletion I...</td>
      <td>0.082359</td>
      <td>0.282033</td>
      <td>1.0</td>
      <td>0.9999</td>
      <td>1.0</td>
      <td>6/18</td>
      <td>38.80%</td>
      <td>NCALD;SLC7A11;PRKCE;SLC2A1;GSK3B;TTC39C</td>
    </tr>
  </tbody>
</table>
<p>4007 rows × 10 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
                 <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
                 <span class="n">title</span><span class="o">=</span><span class="s2">&quot;GSEA Disease Analysis - &#39;ASCL1+&#39; AT2 Like Cells&quot;</span><span class="p">,</span> 
                 <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
                 <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
                 <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">ax_dot</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_20&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Disease Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">ValueError</span><span class="g g-Whitespace">                                </span>Traceback (most recent call last)
<span class="n">Cell</span> <span class="n">In</span><span class="p">[</span><span class="mi">239</span><span class="p">],</span> <span class="n">line</span> <span class="mi">2</span>
<span class="g g-Whitespace">      </span><span class="mi">1</span> <span class="c1"># Dotplot: Size = significance, color = FDR</span>
<span class="ne">----&gt; </span><span class="mi">2</span> <span class="n">ax_dot</span> <span class="o">=</span> <span class="n">gseapy</span><span class="o">.</span><span class="n">dotplot</span><span class="p">(</span><span class="n">res_df</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">3</span>                  <span class="n">column</span><span class="o">=</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">4</span>                  <span class="n">title</span><span class="o">=</span><span class="s2">&quot;GSEA Disease Analysis - &#39;ASCL1+&#39; AT2 Like Cells&quot;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">5</span>                  <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;turbo&#39;</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">6</span>                  <span class="n">size</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> 
<span class="g g-Whitespace">      </span><span class="mi">7</span>                  <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> 
<span class="g g-Whitespace">      </span><span class="mi">8</span>                  <span class="n">cutoff</span><span class="o">=</span><span class="mf">0.05</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">9</span> <span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="g g-Whitespace">     </span><span class="mi">10</span> <span class="n">sc_plot_and_embed</span><span class="p">(</span>
<span class="g g-Whitespace">     </span><span class="mi">11</span>     <span class="n">ax_dot</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">12</span>     <span class="s2">&quot;gsea_ascl1AT2_20&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">13</span>     <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA DotPlot of Disease Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
<span class="g g-Whitespace">     </span><span class="mi">14</span>     <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="g g-Whitespace">     </span><span class="mi">15</span> <span class="p">)</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:1225,</span> in <span class="ni">dotplot</span><span class="nt">(df, column, x, y, x_order, y_order, title, cutoff, top_term, size, ax, figsize, cmap, ofname, xticklabels_rot, yticklabels_rot, marker, show_ring, **kwargs)</span>
<span class="g g-Whitespace">   </span><span class="mi">1222</span>     <span class="n">warnings</span><span class="o">.</span><span class="n">warn</span><span class="p">(</span><span class="s2">&quot;group is deprecated; use x instead&quot;</span><span class="p">,</span> <span class="ne">DeprecationWarning</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1223</span>     <span class="k">return</span>
<span class="ne">-&gt; </span><span class="mi">1225</span> <span class="n">dot</span> <span class="o">=</span> <span class="n">DotPlot</span><span class="p">(</span>
<span class="g g-Whitespace">   </span><span class="mi">1226</span>     <span class="n">df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1227</span>     <span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1228</span>     <span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1229</span>     <span class="n">x_order</span><span class="o">=</span><span class="n">x_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1230</span>     <span class="n">y_order</span><span class="o">=</span><span class="n">y_order</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1231</span>     <span class="n">hue</span><span class="o">=</span><span class="n">column</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1232</span>     <span class="n">title</span><span class="o">=</span><span class="n">title</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1233</span>     <span class="n">thresh</span><span class="o">=</span><span class="n">cutoff</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1234</span>     <span class="n">n_terms</span><span class="o">=</span><span class="nb">int</span><span class="p">(</span><span class="n">top_term</span><span class="p">),</span>
<span class="g g-Whitespace">   </span><span class="mi">1235</span>     <span class="n">dot_scale</span><span class="o">=</span><span class="n">size</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1236</span>     <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1237</span>     <span class="n">figsize</span><span class="o">=</span><span class="n">figsize</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1238</span>     <span class="n">cmap</span><span class="o">=</span><span class="n">cmap</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1239</span>     <span class="n">ofname</span><span class="o">=</span><span class="n">ofname</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1240</span>     <span class="n">marker</span><span class="o">=</span><span class="n">marker</span><span class="p">,</span>
<span class="g g-Whitespace">   </span><span class="mi">1241</span> <span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1242</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">dot</span><span class="o">.</span><span class="n">scatter</span><span class="p">(</span><span class="n">outer_ring</span><span class="o">=</span><span class="n">show_ring</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">1244</span> <span class="k">if</span> <span class="n">xticklabels_rot</span><span class="p">:</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:692,</span> in <span class="ni">DotPlot.__init__</span><span class="nt">(self, df, x, y, hue, dot_scale, x_order, y_order, thresh, n_terms, title, ax, figsize, cmap, ofname, marker, **kwargs)</span>
<span class="g g-Whitespace">    </span><span class="mi">690</span> <span class="bp">self</span><span class="o">.</span><span class="n">n_terms</span> <span class="o">=</span> <span class="n">n_terms</span>
<span class="g g-Whitespace">    </span><span class="mi">691</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span> <span class="o">=</span> <span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">692</span> <span class="bp">self</span><span class="o">.</span><span class="n">data</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">process</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">693</span> <span class="n">plt</span><span class="o">.</span><span class="n">rcParams</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="s2">&quot;pdf.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">,</span> <span class="s2">&quot;ps.fonttype&quot;</span><span class="p">:</span> <span class="mi">42</span><span class="p">})</span>

<span class="nn">File ~/miniconda3/envs/scRNASeqHannan/lib/python3.13/site-packages/gseapy/plot.py:717,</span> in <span class="ni">DotPlot.process</span><span class="nt">(self, df)</span>
<span class="g g-Whitespace">    </span><span class="mi">715</span> <span class="k">if</span> <span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
<span class="g g-Whitespace">    </span><span class="mi">716</span>     <span class="n">msg</span> <span class="o">=</span> <span class="s2">&quot;Warning: No enrich terms when cutoff = </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">thresh</span>
<span class="ne">--&gt; </span><span class="mi">717</span>     <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="n">msg</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">718</span> <span class="bp">self</span><span class="o">.</span><span class="n">cbar_title</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">colname</span>
<span class="g g-Whitespace">    </span><span class="mi">719</span> <span class="c1"># clip GSEA lower bounds</span>
<span class="g g-Whitespace">    </span><span class="mi">720</span> <span class="c1"># if self.colname in [&quot;NOM p-val&quot;, &quot;FDR q-val&quot;]:</span>
<span class="g g-Whitespace">    </span><span class="mi">721</span> <span class="c1">#     df[self.colname].clip(1e-5, 1.0, inplace=True)</span>
<span class="g g-Whitespace">    </span><span class="mi">722</span> <span class="c1"># sorting the dataframe for better visualization</span>

<span class="ne">ValueError</span>: Warning: No enrich terms when cutoff = 0.05
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<details class="admonition hide above-input">
<summary aria-label="Toggle hidden content">
<p class="collapsed admonition-title">Show code cell source</p>
<p class="expanded admonition-title">Hide code cell source</p>
</summary>
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#| hide-input</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">gseapy.plot</span><span class="w"> </span><span class="kn">import</span> <span class="n">gseaplot</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.image</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">mpimg</span>

<span class="n">top_n</span> <span class="o">=</span> <span class="mi">5</span>
<span class="n">top_terms</span> <span class="o">=</span> <span class="n">res_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;FDR q-val&quot;</span><span class="p">)[</span><span class="s1">&#39;Term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="n">top_n</span><span class="p">]</span>

<span class="c1"># Step 1: Generate and save plots for each top term</span>
<span class="n">plot_files</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">top_terms</span><span class="p">):</span>
    <span class="n">fname</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;gsea_</span><span class="si">{</span><span class="n">i</span><span class="si">}</span><span class="s2">.png&quot;</span>
    <span class="n">gseaplot</span><span class="p">(</span><span class="n">rank_metric</span><span class="o">=</span><span class="n">pre_res</span><span class="o">.</span><span class="n">ranking</span><span class="p">,</span>
             <span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span>
             <span class="o">**</span><span class="n">pre_res</span><span class="o">.</span><span class="n">results</span><span class="p">[</span><span class="n">term</span><span class="p">],</span>
             <span class="n">ofname</span><span class="o">=</span><span class="n">fname</span><span class="p">)</span>   <span class="c1"># save each plot as PNG</span>
    <span class="n">plot_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>

<span class="c1"># Step 2: Load images into Matplotlib and arrange horizontally</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">top_n</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="o">*</span><span class="n">top_n</span><span class="p">,</span> <span class="mi">4</span><span class="p">))</span>

<span class="k">if</span> <span class="n">top_n</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
    <span class="n">axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">axes</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">f</span><span class="p">,</span> <span class="n">term</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">plot_files</span><span class="p">,</span> <span class="n">top_terms</span><span class="p">):</span>
    <span class="n">img</span> <span class="o">=</span> <span class="n">mpimg</span><span class="o">.</span><span class="n">imread</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">sc_plot_and_embed</span><span class="p">(</span>
    <span class="n">fig</span><span class="p">,</span>
    <span class="s2">&quot;gsea_ascl1AT2_21&quot;</span><span class="p">,</span>
    <span class="n">caption</span><span class="o">=</span><span class="s2">&quot;GSEA Plots of Disease Analysis Datasets [Top 500 Differentially Expressed Genes].&quot;</span><span class="p">,</span>
    <span class="n">kind</span><span class="o">=</span><span class="s2">&quot;matplotlib&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</details>
<div class="cell_output docutils container">
<div class="output text_html">
    <div class="figure-wrapper" style="text-align:center; margin:20px 0;">
      <img src="_static/Figures/gsea_ascl1AT2_21.png" alt="gsea_ascl1AT2_21" style="max-width:100%; height:auto;" />
      <p style="font-size:0.9em; color:#555;">
        <em>GSEA Plots of Disease Analysis Datasets [Top 500 Differentially Expressed Genes].</em><br>
        <a href="_static/Figures/gsea_ascl1AT2_21.tiff">Download TIFF (600 dpi)</a>
      </p>
    </div>
    </div></div>
</div>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "scrnaseqhannan"
        },
        kernelOptions: {
            name: "scrnaseqhannan",
            path: "./content"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'scrnaseqhannan'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="Final_annotation.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Final Annotation</p>
      </div>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#late-at2-like-cells">Late AT2 Like Cells</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gsea-cell-type-top-250-down-250">GSEA - Cell Type [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gsea-cell-type-top-500">GSEA - Cell Type [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gsea-gene-ontology-top-250-down-250">GSEA - Gene Ontology [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gsea-gene-ontology-top-500">GSEA - Gene Ontology [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gsea-pathway-analysis-top-250-down-250">GSEA - Pathway Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gsea-pathway-analysis-top-500">GSEA - Pathway Analysis [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gsea-proteomic-analysis-top-250-down-250">GSEA - Proteomic Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gsea-proteomic-analysis-top-500">GSEA - Proteomic Analysis [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gsea-disease-analysis-top-250-down-250">GSEA - Disease Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gsea-disease-analysis-top-500">GSEA - Disease Analysis [Top 500]</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#early-at2-like-cells">Early AT2 Like Cells</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">GSEA - Cell Type [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">GSEA - Cell Type [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id3">GSEA - Gene Ontology [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id4">GSEA - Gene Ontology [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id5">GSEA - Pathway Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id6">GSEA - Pathway Analysis [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id7">GSEA - Proteomic Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id8">GSEA - Proteomic Analysis [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id9">GSEA - Disease Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id10">GSEA - Disease Analysis [Top 500]</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#proliferating-at2-like-cells">Proliferating AT2 Like Cells</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id11">GSEA - Cell Type [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id12">GSEA - Cell Type [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id13">GSEA - Gene Ontology [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id14">GSEA - Gene Ontology [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id15">GSEA - Pathway Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id16">GSEA - Pathway Analysis [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id17">GSEA - Proteomic Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id18">GSEA - Proteomic Analysis [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id19">GSEA - Disease Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id20">GSEA - Disease Analysis [Top 500]</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#ascl1-at2-like-cells">‘ASCL1+’ -AT2 Like Cells</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id21">GSEA - Cell Type [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id22">GSEA - Cell Type [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id23">GSEA - Gene Ontology [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id24">GSEA - Gene Ontology [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id25">GSEA - Pathway Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id26">GSEA - Pathway Analysis [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id27">GSEA - Proteomic Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id28">GSEA - Proteomic Analysis [Top 500]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id29">GSEA - Disease Analysis [Top 250 &amp; Down 250]</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id30">GSEA - Disease Analysis [Top 500]</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Carlos B. Sainz Zuñiga
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>